<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Cell propotion analysis – FTLD Analysis Documentation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">FTLD Analysis Documentation</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-introduction" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Introduction</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-introduction">    
        <li>
    <a class="dropdown-item" href="../Introduction/Introduction.html">
 <span class="dropdown-text">Introduction</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Introduction/Background.html">
 <span class="dropdown-text">Background</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Introduction/Cohort.html">
 <span class="dropdown-text">Cohort</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-methods" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Methods</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-methods">    
        <li>
    <a class="dropdown-item" href="../Methods/Bulk_dea.html">
 <span class="dropdown-text">Bulk DEA</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Methods/Bayesprism.html">
 <span class="dropdown-text">Bayesian Deconvolution</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Methods/Cell_prop.html">
 <span class="dropdown-text">Cell proportions analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Methods/Sc_dea.html">
 <span class="dropdown-text">Single-Cell Analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Methods/Hdwgcna.html">
 <span class="dropdown-text">Co-expression Networks analysis</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-results" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Results</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-results">    
        <li>
    <a class="dropdown-item" href="../Results/Bulk_dea_results.html">
 <span class="dropdown-text">Bulk Differential Expression</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Cell_composition.html">
 <span class="dropdown-text">Cell Composition Differences</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Cell_correlation.html">
 <span class="dropdown-text">Cell Composition × Neuropathology</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Sc_dea_results_C9.html">
 <span class="dropdown-text">Single-Cell Differential Expression (C9) 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Sc_dea_results_C9_2.html">
 <span class="dropdown-text">Single-Cell Differential Expression (C9) 2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Sc_dea_results_TDP.html">
 <span class="dropdown-text">Single-Cell Differential Expression (TDP) 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Sc_dea_results_TDP_2.html">
 <span class="dropdown-text">Single-Cell Differential Expression (TDP) 2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Genes_FTLD_summary.html">
 <span class="dropdown-text">Single-Cell FTLD-related genes DE</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Sc_gene_correlation.html">
 <span class="dropdown-text">Single-Cell Gene Correlation</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Hdwgcna_differences.html">
 <span class="dropdown-text">hdWGCNA Differences</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Hdwgcna_C9_correlation.html">
 <span class="dropdown-text">C9 hdWGCNA Correlation</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Hdwgcna_TDP_correlation.html">
 <span class="dropdown-text">TDP hdWGCNA Correlation</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-validation" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Validation</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-validation">    
        <li>
    <a class="dropdown-item" href="../Validation/Bulk_validation.html">
 <span class="dropdown-text">Bulk Analysis Validation</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Validation/Sc_validation.html">
 <span class="dropdown-text">Single-Cell Validation</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Validation/Hdwgcna_validation.html">
 <span class="dropdown-text">hdWGCNA Validation</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-discussion" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Discussion</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-discussion">    
        <li>
    <a class="dropdown-item" href="../Discussion/Discussion.html">
 <span class="dropdown-text">Discussion</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-references" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">References</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-references">    
        <li>
    <a class="dropdown-item" href="../References/References.html">
 <span class="dropdown-text">References</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#differences" id="toc-differences" class="nav-link active" data-scroll-target="#differences">Differences</a>
  <ul class="collapse">
  <li><a href="#correlation" id="toc-correlation" class="nav-link" data-scroll-target="#correlation">Correlation</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Cell propotion analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="differences" class="level2">
<h2 class="anchored" data-anchor-id="differences">Differences</h2>
<p>Compare inferred cell-type proportions across diagnosis groups using multiple statistical tests and data transformations.</p>
<p>To assess alterations in inferred cell state proportions among FTLD cases and cognitively healthy controls, we performed non-parametric statistical comparisons using the Kruskal–Wallis test. Deconvolved cell state proportions were obtained from Bayesian modeling of bulk RNA-seq data, and all samples were included in the analysis.</p>
<p>For each cell state, we compared estimated proportions between disease groups (FTLD-C9 or FTLD-TDP) and healthy controls. Prior to statistical testing, all zero values were replaced with a small constant (1e−10) to avoid division-by-zero errors in downstream calculations. Group comparisons were limited to conditions with at least two observations per group to ensure statistical robustness.</p>
<p>Cell-wise comparisons were conducted using the Kruskal–Wallis rank-sum test, implemented in R. Tests were performed independently for each cell state, with disease groups compared separately against controls. In parallel, fold change and log2 fold change values were computed based on group-wise means, providing complementary effect size measures alongside statistical significance.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"openxlsx"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"coin"</span>)  </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># FILE PATHS</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>frequency_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/BAYESPRISM/FTLD/TDP/theta.state_original.csv"</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>metadata_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/FTLD_BULK/METADATA/decoder_DeSeq2_FTD_FINAL.xlsx"</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>output_path <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/FTLD/TDP/"</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(output_path)) {</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(output_path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>))) {</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(metadata_file)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>freq_cell_types <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(frequency_file)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"long"</span>, <span class="st">""</span>, x))</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"X"</span>, <span class="st">""</span>, x))</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> <span class="fu">merge</span>(data, freq_cell_types, <span class="at">by.x =</span> <span class="st">"sample.ID"</span>, <span class="at">by.y =</span> <span class="st">"X"</span>)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>merged_data[merged_data <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="fl">1e-10</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate needed objects</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>m_conditions <span class="ot">&lt;-</span> <span class="fu">unique</span>(merged_data<span class="sc">$</span>group.ID)</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>m_conditions_nh <span class="ot">&lt;-</span> m_conditions[m_conditions <span class="sc">!=</span> <span class="st">"Healthy"</span>]</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>files_to_iterate <span class="ot">&lt;-</span> <span class="fu">colnames</span>(merged_data)[<span class="dv">4</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(merged_data))]</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>m_cond <span class="ot">&lt;-</span> m_conditions_nh[<span class="dv">1</span>]</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>significatives <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">nrow =</span> <span class="fu">length</span>(files_to_iterate), <span class="at">ncol =</span> <span class="dv">3</span>))</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives)<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>)</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (column <span class="cf">in</span> <span class="fu">colnames</span>(freq_cell_types)[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(freq_cell_types))]){</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Kruskal_"</span>, column, <span class="st">"_"</span>, m_cond, <span class="st">".txt"</span>)</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>  subset_data <span class="ot">&lt;-</span> merged_data[merged_data<span class="sc">$</span>group.ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond), </span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">c</span>(<span class="st">"group.ID"</span>, column)]</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure group.ID is treated as a factor</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>  subset_data<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(subset_data<span class="sc">$</span>group.ID, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond))</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate group means</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>  mean_healthy <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"Healthy"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>  mean_condition <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> m_cond], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate fold change</span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a>  fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mean_healthy <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">NA</span>, mean_condition <span class="sc">/</span> mean_healthy)  <span class="co"># Avoid division by zero</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Log-transform the fold change (optional)</span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>  log2_fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(fold_change), <span class="cn">NA</span>, <span class="fu">log2</span>(fold_change))</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both groups (Healthy and m_cond) have at least 2 observations</span></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>  save <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>(<span class="fu">paste0</span>(output_path, filename))}</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(filename)</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Model A: Kruskal-Wallis Test (without covariates)</span></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>    kruskal_A <span class="ot">&lt;-</span> <span class="fu">kruskal.test</span>(subset_data[[column]], subset_data<span class="sc">$</span>group.ID)</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"Kruskal-Wallis Test (General model)"</span>)</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(kruskal_A)</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>()}</span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Error in processing column "</span>, column, <span class="st">": "</span>, e<span class="sc">$</span>message)</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store the p-values, fold change, and means in the significatives data frame</span></span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">1</span>] <span class="ot">&lt;-</span> column</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">3</span>] <span class="ot">&lt;-</span> kruskal_A<span class="sc">$</span>p.value  <span class="co"># Group (from Kruskal-Wallis test)</span></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">4</span>] <span class="ot">&lt;-</span> fold_change  <span class="co"># Fold change</span></span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">5</span>] <span class="ot">&lt;-</span> log2_fold_change  <span class="co"># Log2 fold change</span></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">6</span>] <span class="ot">&lt;-</span> mean_healthy  <span class="co"># Mean of the Healthy group</span></span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">7</span>] <span class="ot">&lt;-</span> mean_condition  <span class="co"># Mean of the specific condition group</span></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="co"># Update column names of the significatives data frame</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>, <span class="st">"fold_change"</span>, </span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"log2_fold_change"</span>, <span class="st">"mean_healthy"</span>, <span class="st">"mean_condition"</span>)</span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the significatives data frame as a TSV file</span></span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>tsv_output_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_path, <span class="st">"significatives_nonparametric_with_means_cs.tsv"</span>)</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(significatives, </span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> tsv_output_path, </span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,          </span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>,   </span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>,    </span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a><span class="do">### FTLD TDP</span></span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a><span class="do">#### ArcSin</span></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"openxlsx"</span>)</span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"coin"</span>)  </span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a><span class="co"># FILE PATHS</span></span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>frequency_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/TRANSFORMATED_PROPORTIONS/ArcSin/FTLD/TDP/Arcsin_Cell_state_proportions.csv"</span></span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>metadata_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/FTLD_BULK/METADATA/decoder_DeSeq2_FTD_FINAL.xlsx"</span></span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a>output_path <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/TRANSFORMED_RESULTS/FTLD/TDP/ArcSin/"</span></span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(output_path)) {</span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(output_path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>))) {</span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(metadata_file)</span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a>freq_cell_types <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(frequency_file)</span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"long"</span>, <span class="st">""</span>, x))</span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"X"</span>, <span class="st">""</span>, x))</span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> <span class="fu">merge</span>(data, freq_cell_types, <span class="at">by.x =</span> <span class="st">"sample.ID"</span>, <span class="at">by.y =</span> <span class="st">"X"</span>)</span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a>merged_data[merged_data <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="fl">1e-10</span></span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate needed objects</span></span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a>m_conditions <span class="ot">&lt;-</span> <span class="fu">unique</span>(merged_data<span class="sc">$</span>group.ID)</span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a>m_conditions_nh <span class="ot">&lt;-</span> m_conditions[m_conditions <span class="sc">!=</span> <span class="st">"Healthy"</span>]</span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a>files_to_iterate <span class="ot">&lt;-</span> <span class="fu">colnames</span>(merged_data)[<span class="dv">4</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(merged_data))]</span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a>m_cond <span class="ot">&lt;-</span> m_conditions_nh[<span class="dv">1</span>]</span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a>significatives <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">nrow =</span> <span class="fu">length</span>(files_to_iterate), <span class="at">ncol =</span> <span class="dv">3</span>))</span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives)<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>)</span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (column <span class="cf">in</span> <span class="fu">colnames</span>(freq_cell_types)[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(freq_cell_types))]){</span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Kruskal_"</span>, column, <span class="st">"_"</span>, m_cond, <span class="st">".txt"</span>)</span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a>  subset_data <span class="ot">&lt;-</span> merged_data[merged_data<span class="sc">$</span>group.ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond), </span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">c</span>(<span class="st">"group.ID"</span>, column)]</span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure group.ID is treated as a factor</span></span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a>  subset_data<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(subset_data<span class="sc">$</span>group.ID, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond))</span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate group means</span></span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a>  mean_healthy <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"Healthy"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a>  mean_condition <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> m_cond], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate fold change</span></span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a>  fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mean_healthy <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">NA</span>, mean_condition <span class="sc">/</span> mean_healthy)  <span class="co"># Avoid division by zero</span></span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Log-transform the fold change (optional)</span></span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a>  log2_fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(fold_change), <span class="cn">NA</span>, <span class="fu">log2</span>(fold_change))</span>
<span id="cb1-159"><a href="#cb1-159" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-160"><a href="#cb1-160" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both groups (Healthy and m_cond) have at least 2 observations</span></span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a>  save <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb1-163"><a href="#cb1-163" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>(<span class="fu">paste0</span>(output_path, filename))}</span>
<span id="cb1-164"><a href="#cb1-164" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-165"><a href="#cb1-165" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(filename)</span>
<span id="cb1-166"><a href="#cb1-166" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-167"><a href="#cb1-167" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Model A: Kruskal-Wallis Test (without covariates)</span></span>
<span id="cb1-168"><a href="#cb1-168" aria-hidden="true" tabindex="-1"></a>    kruskal_A <span class="ot">&lt;-</span> <span class="fu">kruskal.test</span>(subset_data[[column]], subset_data<span class="sc">$</span>group.ID)</span>
<span id="cb1-169"><a href="#cb1-169" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"Kruskal-Wallis Test (General model)"</span>)</span>
<span id="cb1-170"><a href="#cb1-170" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(kruskal_A)</span>
<span id="cb1-171"><a href="#cb1-171" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>()}</span>
<span id="cb1-172"><a href="#cb1-172" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-173"><a href="#cb1-173" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb1-174"><a href="#cb1-174" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Error in processing column "</span>, column, <span class="st">": "</span>, e<span class="sc">$</span>message)</span>
<span id="cb1-175"><a href="#cb1-175" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb1-176"><a href="#cb1-176" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb1-177"><a href="#cb1-177" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-178"><a href="#cb1-178" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store the p-values, fold change, and means in the significatives data frame</span></span>
<span id="cb1-179"><a href="#cb1-179" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">1</span>] <span class="ot">&lt;-</span> column</span>
<span id="cb1-180"><a href="#cb1-180" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">3</span>] <span class="ot">&lt;-</span> kruskal_A<span class="sc">$</span>p.value  <span class="co"># Group (from Kruskal-Wallis test)</span></span>
<span id="cb1-181"><a href="#cb1-181" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">4</span>] <span class="ot">&lt;-</span> fold_change  <span class="co"># Fold change</span></span>
<span id="cb1-182"><a href="#cb1-182" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">5</span>] <span class="ot">&lt;-</span> log2_fold_change  <span class="co"># Log2 fold change</span></span>
<span id="cb1-183"><a href="#cb1-183" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">6</span>] <span class="ot">&lt;-</span> mean_healthy  <span class="co"># Mean of the Healthy group</span></span>
<span id="cb1-184"><a href="#cb1-184" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">7</span>] <span class="ot">&lt;-</span> mean_condition  <span class="co"># Mean of the specific condition group</span></span>
<span id="cb1-185"><a href="#cb1-185" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-186"><a href="#cb1-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-187"><a href="#cb1-187" aria-hidden="true" tabindex="-1"></a><span class="co"># Update column names of the significatives data frame</span></span>
<span id="cb1-188"><a href="#cb1-188" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>, <span class="st">"fold_change"</span>, </span>
<span id="cb1-189"><a href="#cb1-189" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"log2_fold_change"</span>, <span class="st">"mean_healthy"</span>, <span class="st">"mean_condition"</span>)</span>
<span id="cb1-190"><a href="#cb1-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-191"><a href="#cb1-191" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the significatives data frame as a TSV file</span></span>
<span id="cb1-192"><a href="#cb1-192" aria-hidden="true" tabindex="-1"></a>tsv_output_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_path, <span class="st">"significatives_nonparametric_with_means_cs.tsv"</span>)</span>
<span id="cb1-193"><a href="#cb1-193" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(significatives, </span>
<span id="cb1-194"><a href="#cb1-194" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> tsv_output_path, </span>
<span id="cb1-195"><a href="#cb1-195" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,          </span>
<span id="cb1-196"><a href="#cb1-196" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>,   </span>
<span id="cb1-197"><a href="#cb1-197" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>,    </span>
<span id="cb1-198"><a href="#cb1-198" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">### FTLD C9</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"openxlsx"</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"coin"</span>)  </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># FILE PATHS</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>frequency_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/BAYESPRISM/FTLD/C9/theta.state_cellstate.csv"</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>metadata_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/FTLD_BULK/METADATA/decoder_DeSeq2_FTD_FINAL.xlsx"</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>output_path <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/FTLD/c9/"</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(output_path)) {</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(output_path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>))) {</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(metadata_file)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>freq_cell_types <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(frequency_file)</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"long"</span>, <span class="st">""</span>, x))</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"X"</span>, <span class="st">""</span>, x))</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> <span class="fu">merge</span>(data, freq_cell_types, <span class="at">by.x =</span> <span class="st">"sample.ID"</span>, <span class="at">by.y =</span> <span class="st">"X"</span>)</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>merged_data[merged_data <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="fl">1e-10</span></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate needed objects</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>m_conditions <span class="ot">&lt;-</span> <span class="fu">unique</span>(merged_data<span class="sc">$</span>group.ID)</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>m_conditions_nh <span class="ot">&lt;-</span> m_conditions[m_conditions <span class="sc">!=</span> <span class="st">"Healthy"</span>]</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>files_to_iterate <span class="ot">&lt;-</span> <span class="fu">colnames</span>(merged_data)[<span class="dv">4</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(merged_data))]</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>m_cond <span class="ot">&lt;-</span> m_conditions_nh[<span class="dv">1</span>]</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>significatives <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">nrow =</span> <span class="fu">length</span>(files_to_iterate), <span class="at">ncol =</span> <span class="dv">3</span>))</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives)<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>)</span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (column <span class="cf">in</span> <span class="fu">colnames</span>(freq_cell_types)[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(freq_cell_types))]){</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Kruskal_"</span>, column, <span class="st">"_"</span>, m_cond, <span class="st">".txt"</span>)</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>  subset_data <span class="ot">&lt;-</span> merged_data[merged_data<span class="sc">$</span>group.ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond), </span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">c</span>(<span class="st">"group.ID"</span>, column)]</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure group.ID is treated as a factor</span></span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>  subset_data<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(subset_data<span class="sc">$</span>group.ID, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond))</span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate group means</span></span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>  mean_healthy <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"Healthy"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>  mean_condition <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> m_cond], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate fold change</span></span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>  fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mean_healthy <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">NA</span>, mean_condition <span class="sc">/</span> mean_healthy)  <span class="co"># Avoid division by zero</span></span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Log-transform the fold change (optional)</span></span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>  log2_fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(fold_change), <span class="cn">NA</span>, <span class="fu">log2</span>(fold_change))</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both groups (Healthy and m_cond) have at least 2 observations</span></span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>  save <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>(<span class="fu">paste0</span>(output_path, filename))}</span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(filename)</span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Model A: Kruskal-Wallis Test (without covariates)</span></span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a>    kruskal_A <span class="ot">&lt;-</span> <span class="fu">kruskal.test</span>(subset_data[[column]], subset_data<span class="sc">$</span>group.ID)</span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"Kruskal-Wallis Test (General model)"</span>)</span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(kruskal_A)</span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>()}</span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Error in processing column "</span>, column, <span class="st">": "</span>, e<span class="sc">$</span>message)</span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-79"><a href="#cb2-79" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store the p-values, fold change, and means in the significatives data frame</span></span>
<span id="cb2-80"><a href="#cb2-80" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">1</span>] <span class="ot">&lt;-</span> column</span>
<span id="cb2-81"><a href="#cb2-81" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">3</span>] <span class="ot">&lt;-</span> kruskal_A<span class="sc">$</span>p.value  <span class="co"># Group (from Kruskal-Wallis test)</span></span>
<span id="cb2-82"><a href="#cb2-82" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">4</span>] <span class="ot">&lt;-</span> fold_change  <span class="co"># Fold change</span></span>
<span id="cb2-83"><a href="#cb2-83" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">5</span>] <span class="ot">&lt;-</span> log2_fold_change  <span class="co"># Log2 fold change</span></span>
<span id="cb2-84"><a href="#cb2-84" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">6</span>] <span class="ot">&lt;-</span> mean_healthy  <span class="co"># Mean of the Healthy group</span></span>
<span id="cb2-85"><a href="#cb2-85" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">7</span>] <span class="ot">&lt;-</span> mean_condition  <span class="co"># Mean of the specific condition group</span></span>
<span id="cb2-86"><a href="#cb2-86" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-87"><a href="#cb2-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-88"><a href="#cb2-88" aria-hidden="true" tabindex="-1"></a><span class="co"># Update column names of the significatives data frame</span></span>
<span id="cb2-89"><a href="#cb2-89" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>, <span class="st">"fold_change"</span>, </span>
<span id="cb2-90"><a href="#cb2-90" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"log2_fold_change"</span>, <span class="st">"mean_healthy"</span>, <span class="st">"mean_condition"</span>)</span>
<span id="cb2-91"><a href="#cb2-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-92"><a href="#cb2-92" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the significatives data frame as a TSV file</span></span>
<span id="cb2-93"><a href="#cb2-93" aria-hidden="true" tabindex="-1"></a>tsv_output_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_path, <span class="st">"significatives_nonparametric_with_means_cs.tsv"</span>)</span>
<span id="cb2-94"><a href="#cb2-94" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(significatives, </span>
<span id="cb2-95"><a href="#cb2-95" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> tsv_output_path, </span>
<span id="cb2-96"><a href="#cb2-96" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,          </span>
<span id="cb2-97"><a href="#cb2-97" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>,   </span>
<span id="cb2-98"><a href="#cb2-98" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>,    </span>
<span id="cb2-99"><a href="#cb2-99" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-100"><a href="#cb2-100" aria-hidden="true" tabindex="-1"></a><span class="do">#### ArcSin</span></span>
<span id="cb2-101"><a href="#cb2-101" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb2-102"><a href="#cb2-102" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"openxlsx"</span>)</span>
<span id="cb2-103"><a href="#cb2-103" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb2-104"><a href="#cb2-104" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"coin"</span>)  </span>
<span id="cb2-105"><a href="#cb2-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-106"><a href="#cb2-106" aria-hidden="true" tabindex="-1"></a><span class="co"># FILE PATHS</span></span>
<span id="cb2-107"><a href="#cb2-107" aria-hidden="true" tabindex="-1"></a>frequency_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/TRANSFORMATED_PROPORTIONS/ArcSin/FTLD/C9/Arcsin_Cell_state_proportions.csv"</span></span>
<span id="cb2-108"><a href="#cb2-108" aria-hidden="true" tabindex="-1"></a>metadata_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/FTLD_BULK/METADATA/decoder_DeSeq2_FTD_FINAL.xlsx"</span></span>
<span id="cb2-109"><a href="#cb2-109" aria-hidden="true" tabindex="-1"></a>output_path <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/TRANSFORMED_RESULTS/FTLD/C9/ArcSin/"</span></span>
<span id="cb2-110"><a href="#cb2-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-111"><a href="#cb2-111" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(output_path)) {</span>
<span id="cb2-112"><a href="#cb2-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(output_path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-113"><a href="#cb2-113" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-114"><a href="#cb2-114" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>))) {</span>
<span id="cb2-115"><a href="#cb2-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-116"><a href="#cb2-116" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-117"><a href="#cb2-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-118"><a href="#cb2-118" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(metadata_file)</span>
<span id="cb2-119"><a href="#cb2-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-120"><a href="#cb2-120" aria-hidden="true" tabindex="-1"></a>freq_cell_types <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(frequency_file)</span>
<span id="cb2-121"><a href="#cb2-121" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"long"</span>, <span class="st">""</span>, x))</span>
<span id="cb2-122"><a href="#cb2-122" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"X"</span>, <span class="st">""</span>, x))</span>
<span id="cb2-123"><a href="#cb2-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-124"><a href="#cb2-124" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> <span class="fu">merge</span>(data, freq_cell_types, <span class="at">by.x =</span> <span class="st">"sample.ID"</span>, <span class="at">by.y =</span> <span class="st">"X"</span>)</span>
<span id="cb2-125"><a href="#cb2-125" aria-hidden="true" tabindex="-1"></a>merged_data[merged_data <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="fl">1e-10</span></span>
<span id="cb2-126"><a href="#cb2-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-127"><a href="#cb2-127" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate needed objects</span></span>
<span id="cb2-128"><a href="#cb2-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-129"><a href="#cb2-129" aria-hidden="true" tabindex="-1"></a>m_conditions <span class="ot">&lt;-</span> <span class="fu">unique</span>(merged_data<span class="sc">$</span>group.ID)</span>
<span id="cb2-130"><a href="#cb2-130" aria-hidden="true" tabindex="-1"></a>m_conditions_nh <span class="ot">&lt;-</span> m_conditions[m_conditions <span class="sc">!=</span> <span class="st">"Healthy"</span>]</span>
<span id="cb2-131"><a href="#cb2-131" aria-hidden="true" tabindex="-1"></a>files_to_iterate <span class="ot">&lt;-</span> <span class="fu">colnames</span>(merged_data)[<span class="dv">4</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(merged_data))]</span>
<span id="cb2-132"><a href="#cb2-132" aria-hidden="true" tabindex="-1"></a>m_cond <span class="ot">&lt;-</span> m_conditions_nh[<span class="dv">1</span>]</span>
<span id="cb2-133"><a href="#cb2-133" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb2-134"><a href="#cb2-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-135"><a href="#cb2-135" aria-hidden="true" tabindex="-1"></a>significatives <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">nrow =</span> <span class="fu">length</span>(files_to_iterate), <span class="at">ncol =</span> <span class="dv">3</span>))</span>
<span id="cb2-136"><a href="#cb2-136" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives)<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>)</span>
<span id="cb2-137"><a href="#cb2-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-138"><a href="#cb2-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-139"><a href="#cb2-139" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (column <span class="cf">in</span> <span class="fu">colnames</span>(freq_cell_types)[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(freq_cell_types))]){</span>
<span id="cb2-140"><a href="#cb2-140" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb2-141"><a href="#cb2-141" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Kruskal_"</span>, column, <span class="st">"_"</span>, m_cond, <span class="st">".txt"</span>)</span>
<span id="cb2-142"><a href="#cb2-142" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-143"><a href="#cb2-143" aria-hidden="true" tabindex="-1"></a>  subset_data <span class="ot">&lt;-</span> merged_data[merged_data<span class="sc">$</span>group.ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond), </span>
<span id="cb2-144"><a href="#cb2-144" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">c</span>(<span class="st">"group.ID"</span>, column)]</span>
<span id="cb2-145"><a href="#cb2-145" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-146"><a href="#cb2-146" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure group.ID is treated as a factor</span></span>
<span id="cb2-147"><a href="#cb2-147" aria-hidden="true" tabindex="-1"></a>  subset_data<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(subset_data<span class="sc">$</span>group.ID, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond))</span>
<span id="cb2-148"><a href="#cb2-148" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-149"><a href="#cb2-149" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate group means</span></span>
<span id="cb2-150"><a href="#cb2-150" aria-hidden="true" tabindex="-1"></a>  mean_healthy <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"Healthy"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-151"><a href="#cb2-151" aria-hidden="true" tabindex="-1"></a>  mean_condition <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> m_cond], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-152"><a href="#cb2-152" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-153"><a href="#cb2-153" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate fold change</span></span>
<span id="cb2-154"><a href="#cb2-154" aria-hidden="true" tabindex="-1"></a>  fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mean_healthy <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">NA</span>, mean_condition <span class="sc">/</span> mean_healthy)  <span class="co"># Avoid division by zero</span></span>
<span id="cb2-155"><a href="#cb2-155" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-156"><a href="#cb2-156" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Log-transform the fold change (optional)</span></span>
<span id="cb2-157"><a href="#cb2-157" aria-hidden="true" tabindex="-1"></a>  log2_fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(fold_change), <span class="cn">NA</span>, <span class="fu">log2</span>(fold_change))</span>
<span id="cb2-158"><a href="#cb2-158" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-159"><a href="#cb2-159" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both groups (Healthy and m_cond) have at least 2 observations</span></span>
<span id="cb2-160"><a href="#cb2-160" aria-hidden="true" tabindex="-1"></a>  save <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb2-161"><a href="#cb2-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb2-162"><a href="#cb2-162" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>(<span class="fu">paste0</span>(output_path, filename))}</span>
<span id="cb2-163"><a href="#cb2-163" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-164"><a href="#cb2-164" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(filename)</span>
<span id="cb2-165"><a href="#cb2-165" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-166"><a href="#cb2-166" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Model A: Kruskal-Wallis Test (without covariates)</span></span>
<span id="cb2-167"><a href="#cb2-167" aria-hidden="true" tabindex="-1"></a>    kruskal_A <span class="ot">&lt;-</span> <span class="fu">kruskal.test</span>(subset_data[[column]], subset_data<span class="sc">$</span>group.ID)</span>
<span id="cb2-168"><a href="#cb2-168" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"Kruskal-Wallis Test (General model)"</span>)</span>
<span id="cb2-169"><a href="#cb2-169" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(kruskal_A)</span>
<span id="cb2-170"><a href="#cb2-170" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>()}</span>
<span id="cb2-171"><a href="#cb2-171" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-172"><a href="#cb2-172" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb2-173"><a href="#cb2-173" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Error in processing column "</span>, column, <span class="st">": "</span>, e<span class="sc">$</span>message)</span>
<span id="cb2-174"><a href="#cb2-174" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb2-175"><a href="#cb2-175" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb2-176"><a href="#cb2-176" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-177"><a href="#cb2-177" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store the p-values, fold change, and means in the significatives data frame</span></span>
<span id="cb2-178"><a href="#cb2-178" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">1</span>] <span class="ot">&lt;-</span> column</span>
<span id="cb2-179"><a href="#cb2-179" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">3</span>] <span class="ot">&lt;-</span> kruskal_A<span class="sc">$</span>p.value  <span class="co"># Group (from Kruskal-Wallis test)</span></span>
<span id="cb2-180"><a href="#cb2-180" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">4</span>] <span class="ot">&lt;-</span> fold_change  <span class="co"># Fold change</span></span>
<span id="cb2-181"><a href="#cb2-181" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">5</span>] <span class="ot">&lt;-</span> log2_fold_change  <span class="co"># Log2 fold change</span></span>
<span id="cb2-182"><a href="#cb2-182" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">6</span>] <span class="ot">&lt;-</span> mean_healthy  <span class="co"># Mean of the Healthy group</span></span>
<span id="cb2-183"><a href="#cb2-183" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">7</span>] <span class="ot">&lt;-</span> mean_condition  <span class="co"># Mean of the specific condition group</span></span>
<span id="cb2-184"><a href="#cb2-184" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-185"><a href="#cb2-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-186"><a href="#cb2-186" aria-hidden="true" tabindex="-1"></a><span class="co"># Update column names of the significatives data frame</span></span>
<span id="cb2-187"><a href="#cb2-187" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>, <span class="st">"fold_change"</span>, </span>
<span id="cb2-188"><a href="#cb2-188" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"log2_fold_change"</span>, <span class="st">"mean_healthy"</span>, <span class="st">"mean_condition"</span>)</span>
<span id="cb2-189"><a href="#cb2-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-190"><a href="#cb2-190" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the significatives data frame as a TSV file</span></span>
<span id="cb2-191"><a href="#cb2-191" aria-hidden="true" tabindex="-1"></a>tsv_output_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_path, <span class="st">"significatives_nonparametric_with_means_cs.tsv"</span>)</span>
<span id="cb2-192"><a href="#cb2-192" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(significatives, </span>
<span id="cb2-193"><a href="#cb2-193" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> tsv_output_path, </span>
<span id="cb2-194"><a href="#cb2-194" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,          </span>
<span id="cb2-195"><a href="#cb2-195" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>,   </span>
<span id="cb2-196"><a href="#cb2-196" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>,    </span>
<span id="cb2-197"><a href="#cb2-197" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pottier</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"openxlsx"</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"coin"</span>)  </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># FILE PATHS</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>frequency_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/BAYESPRISM/NEW/theta.state_cellstate.csv"</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>metadata_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/NEW_BULK/METADATA/Sample_info.txt"</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>output_path <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/NEW/"</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(output_path)) {</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(output_path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>))) {</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(metadata_file)</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>freq_cell_types <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(frequency_file)</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">"_"</span>, x))</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>FCX_ID <span class="ot">&lt;-</span> <span class="fu">sapply</span>(data<span class="sc">$</span>FCX_ID, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"-"</span>, <span class="st">"_"</span>, x))</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> <span class="fu">merge</span>(data, freq_cell_types, <span class="at">by.x =</span> <span class="st">"FCX_ID"</span>, <span class="at">by.y =</span> <span class="st">"X"</span>)</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>merged_data[merged_data <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="fl">1e-10</span></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> merged_data[merged_data<span class="sc">$</span>GROUP <span class="sc">!=</span> <span class="st">"FTLD-TDP-C"</span>,]</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>merged_data<span class="sc">$</span>GROUP[merged_data<span class="sc">$</span>GROUP <span class="sc">!=</span> <span class="st">"Control"</span>]<span class="ot">&lt;-</span><span class="st">"FTLD"</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate needed objects</span></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>m_conditions <span class="ot">&lt;-</span> <span class="fu">unique</span>(merged_data<span class="sc">$</span>GROUP)</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>m_conditions_nh <span class="ot">&lt;-</span> m_conditions[m_conditions <span class="sc">!=</span> <span class="st">"Control"</span>]</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>files_to_iterate <span class="ot">&lt;-</span> <span class="fu">colnames</span>(merged_data)[<span class="dv">4</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(merged_data))]</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>m_cond <span class="ot">&lt;-</span> m_conditions_nh[<span class="dv">1</span>]</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>significatives <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">nrow =</span> <span class="fu">length</span>(files_to_iterate), <span class="at">ncol =</span> <span class="dv">3</span>))</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives)<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>)</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (column <span class="cf">in</span> <span class="fu">colnames</span>(freq_cell_types)[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(freq_cell_types))]){</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Kruskal_"</span>, column, <span class="st">"_"</span>, m_cond, <span class="st">".txt"</span>)</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>  subset_data <span class="ot">&lt;-</span> merged_data[merged_data<span class="sc">$</span>GROUP <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Control"</span>, m_cond), </span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">c</span>(<span class="st">"GROUP"</span>, column)]</span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure group.ID is treated as a factor</span></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>  subset_data<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(subset_data<span class="sc">$</span>GROUP, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Control"</span>, m_cond))</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate group means</span></span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>  mean_healthy <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>GROUP <span class="sc">==</span> <span class="st">"Control"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>  mean_condition <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>GROUP <span class="sc">==</span> m_cond], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate fold change</span></span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>  fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mean_healthy <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">NA</span>, mean_condition <span class="sc">/</span> mean_healthy)  <span class="co"># Avoid division by zero</span></span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Log-transform the fold change (optional)</span></span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>  log2_fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(fold_change), <span class="cn">NA</span>, <span class="fu">log2</span>(fold_change))</span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both groups (Healthy and m_cond) have at least 2 observations</span></span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>  save <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>(<span class="fu">paste0</span>(output_path, filename))}</span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(filename)</span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Model A: Kruskal-Wallis Test (without covariates)</span></span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a>    kruskal_A <span class="ot">&lt;-</span> <span class="fu">kruskal.test</span>(subset_data[[column]], subset_data<span class="sc">$</span>GROUP)</span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"Kruskal-Wallis Test (General model)"</span>)</span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(kruskal_A)</span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>()}</span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Error in processing column "</span>, column, <span class="st">": "</span>, e<span class="sc">$</span>message)</span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store the p-values, fold change, and means in the significatives data frame</span></span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">1</span>] <span class="ot">&lt;-</span> column</span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">3</span>] <span class="ot">&lt;-</span> kruskal_A<span class="sc">$</span>p.value  <span class="co"># Group (from Kruskal-Wallis test)</span></span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">4</span>] <span class="ot">&lt;-</span> fold_change  <span class="co"># Fold change</span></span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">5</span>] <span class="ot">&lt;-</span> log2_fold_change  <span class="co"># Log2 fold change</span></span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">6</span>] <span class="ot">&lt;-</span> mean_healthy  <span class="co"># Mean of the Healthy group</span></span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">7</span>] <span class="ot">&lt;-</span> mean_condition  <span class="co"># Mean of the specific condition group</span></span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a><span class="co"># Update column names of the significatives data frame</span></span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>, <span class="st">"fold_change"</span>, </span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"log2_fold_change"</span>, <span class="st">"mean_healthy"</span>, <span class="st">"mean_condition"</span>)</span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the significatives data frame as a TSV file</span></span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a>tsv_output_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_path, <span class="st">"significatives_nonparametric_with_means_cs.tsv"</span>)</span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(significatives, </span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> tsv_output_path, </span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,          </span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>,   </span>
<span id="cb3-101"><a href="#cb3-101" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>,    </span>
<span id="cb3-102"><a href="#cb3-102" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-103"><a href="#cb3-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-104"><a href="#cb3-104" aria-hidden="true" tabindex="-1"></a><span class="do">#### ArcSin</span></span>
<span id="cb3-105"><a href="#cb3-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-106"><a href="#cb3-106" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb3-107"><a href="#cb3-107" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"openxlsx"</span>)</span>
<span id="cb3-108"><a href="#cb3-108" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb3-109"><a href="#cb3-109" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"coin"</span>)  </span>
<span id="cb3-110"><a href="#cb3-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-111"><a href="#cb3-111" aria-hidden="true" tabindex="-1"></a><span class="co"># FILE PATHS</span></span>
<span id="cb3-112"><a href="#cb3-112" aria-hidden="true" tabindex="-1"></a>frequency_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/TRANSFORMATED_PROPORTIONS/ArcSin/Pottier/TDP/Log_Cell_state_proportions.csv"</span></span>
<span id="cb3-113"><a href="#cb3-113" aria-hidden="true" tabindex="-1"></a>metadata_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/NEW_BULK/METADATA/Sample_info.txt"</span></span>
<span id="cb3-114"><a href="#cb3-114" aria-hidden="true" tabindex="-1"></a>output_path <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/TRANSFORMED_RESULTS/Pottier/TDP/ArcSin/"</span></span>
<span id="cb3-115"><a href="#cb3-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-116"><a href="#cb3-116" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(output_path)) {</span>
<span id="cb3-117"><a href="#cb3-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(output_path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-118"><a href="#cb3-118" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-119"><a href="#cb3-119" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>))) {</span>
<span id="cb3-120"><a href="#cb3-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-121"><a href="#cb3-121" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-122"><a href="#cb3-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-123"><a href="#cb3-123" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(metadata_file)</span>
<span id="cb3-124"><a href="#cb3-124" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> data<span class="sc">$</span>GROUP</span>
<span id="cb3-125"><a href="#cb3-125" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>GROUP <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb3-126"><a href="#cb3-126" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[<span class="sc">!</span>data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"FTLD-TDP-C"</span>,]</span>
<span id="cb3-127"><a href="#cb3-127" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>group.ID[data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"Control"</span>] <span class="ot">&lt;-</span> <span class="st">"Healthy"</span></span>
<span id="cb3-128"><a href="#cb3-128" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>group.ID[<span class="sc">!</span>data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"Healthy"</span>] <span class="ot">&lt;-</span> <span class="st">"TDP"</span></span>
<span id="cb3-129"><a href="#cb3-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-130"><a href="#cb3-130" aria-hidden="true" tabindex="-1"></a>freq_cell_types <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(frequency_file)</span>
<span id="cb3-131"><a href="#cb3-131" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">"-"</span>, x))</span>
<span id="cb3-132"><a href="#cb3-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-133"><a href="#cb3-133" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> <span class="fu">merge</span>(data, freq_cell_types, <span class="at">by.x =</span> <span class="st">"FCX_ID"</span>, <span class="at">by.y =</span> <span class="st">"X"</span>)</span>
<span id="cb3-134"><a href="#cb3-134" aria-hidden="true" tabindex="-1"></a>merged_data[merged_data <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="fl">1e-10</span></span>
<span id="cb3-135"><a href="#cb3-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-136"><a href="#cb3-136" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate needed objects</span></span>
<span id="cb3-137"><a href="#cb3-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-138"><a href="#cb3-138" aria-hidden="true" tabindex="-1"></a>m_conditions <span class="ot">&lt;-</span> <span class="fu">unique</span>(merged_data<span class="sc">$</span>group.ID)</span>
<span id="cb3-139"><a href="#cb3-139" aria-hidden="true" tabindex="-1"></a>m_conditions_nh <span class="ot">&lt;-</span> m_conditions[m_conditions <span class="sc">!=</span> <span class="st">"Healthy"</span>]</span>
<span id="cb3-140"><a href="#cb3-140" aria-hidden="true" tabindex="-1"></a>files_to_iterate <span class="ot">&lt;-</span> <span class="fu">colnames</span>(merged_data)[<span class="dv">4</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(merged_data))]</span>
<span id="cb3-141"><a href="#cb3-141" aria-hidden="true" tabindex="-1"></a>m_cond <span class="ot">&lt;-</span> m_conditions_nh[<span class="dv">1</span>]</span>
<span id="cb3-142"><a href="#cb3-142" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb3-143"><a href="#cb3-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-144"><a href="#cb3-144" aria-hidden="true" tabindex="-1"></a>significatives <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">nrow =</span> <span class="fu">length</span>(files_to_iterate), <span class="at">ncol =</span> <span class="dv">3</span>))</span>
<span id="cb3-145"><a href="#cb3-145" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives)<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>)</span>
<span id="cb3-146"><a href="#cb3-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-147"><a href="#cb3-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-148"><a href="#cb3-148" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (column <span class="cf">in</span> <span class="fu">colnames</span>(freq_cell_types)[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(freq_cell_types))]){</span>
<span id="cb3-149"><a href="#cb3-149" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb3-150"><a href="#cb3-150" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Kruskal_"</span>, column, <span class="st">"_"</span>, m_cond, <span class="st">".txt"</span>)</span>
<span id="cb3-151"><a href="#cb3-151" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-152"><a href="#cb3-152" aria-hidden="true" tabindex="-1"></a>  subset_data <span class="ot">&lt;-</span> merged_data[merged_data<span class="sc">$</span>group.ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond), </span>
<span id="cb3-153"><a href="#cb3-153" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">c</span>(<span class="st">"group.ID"</span>, column)]</span>
<span id="cb3-154"><a href="#cb3-154" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-155"><a href="#cb3-155" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure group.ID is treated as a factor</span></span>
<span id="cb3-156"><a href="#cb3-156" aria-hidden="true" tabindex="-1"></a>  subset_data<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(subset_data<span class="sc">$</span>group.ID, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond))</span>
<span id="cb3-157"><a href="#cb3-157" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-158"><a href="#cb3-158" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate group means</span></span>
<span id="cb3-159"><a href="#cb3-159" aria-hidden="true" tabindex="-1"></a>  mean_healthy <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"Healthy"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-160"><a href="#cb3-160" aria-hidden="true" tabindex="-1"></a>  mean_condition <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> m_cond], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-161"><a href="#cb3-161" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-162"><a href="#cb3-162" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate fold change</span></span>
<span id="cb3-163"><a href="#cb3-163" aria-hidden="true" tabindex="-1"></a>  fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mean_healthy <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">NA</span>, mean_condition <span class="sc">/</span> mean_healthy)  <span class="co"># Avoid division by zero</span></span>
<span id="cb3-164"><a href="#cb3-164" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-165"><a href="#cb3-165" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Log-transform the fold change (optional)</span></span>
<span id="cb3-166"><a href="#cb3-166" aria-hidden="true" tabindex="-1"></a>  log2_fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(fold_change), <span class="cn">NA</span>, <span class="fu">log2</span>(fold_change))</span>
<span id="cb3-167"><a href="#cb3-167" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-168"><a href="#cb3-168" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both groups (Healthy and m_cond) have at least 2 observations</span></span>
<span id="cb3-169"><a href="#cb3-169" aria-hidden="true" tabindex="-1"></a>  save <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb3-170"><a href="#cb3-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb3-171"><a href="#cb3-171" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>(<span class="fu">paste0</span>(output_path, filename))}</span>
<span id="cb3-172"><a href="#cb3-172" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-173"><a href="#cb3-173" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(filename)</span>
<span id="cb3-174"><a href="#cb3-174" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-175"><a href="#cb3-175" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Model A: Kruskal-Wallis Test (without covariates)</span></span>
<span id="cb3-176"><a href="#cb3-176" aria-hidden="true" tabindex="-1"></a>    kruskal_A <span class="ot">&lt;-</span> <span class="fu">kruskal.test</span>(subset_data[[column]], subset_data<span class="sc">$</span>group.ID)</span>
<span id="cb3-177"><a href="#cb3-177" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"Kruskal-Wallis Test (General model)"</span>)</span>
<span id="cb3-178"><a href="#cb3-178" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(kruskal_A)</span>
<span id="cb3-179"><a href="#cb3-179" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>()}</span>
<span id="cb3-180"><a href="#cb3-180" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-181"><a href="#cb3-181" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb3-182"><a href="#cb3-182" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Error in processing column "</span>, column, <span class="st">": "</span>, e<span class="sc">$</span>message)</span>
<span id="cb3-183"><a href="#cb3-183" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb3-184"><a href="#cb3-184" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb3-185"><a href="#cb3-185" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-186"><a href="#cb3-186" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store the p-values, fold change, and means in the significatives data frame</span></span>
<span id="cb3-187"><a href="#cb3-187" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">1</span>] <span class="ot">&lt;-</span> column</span>
<span id="cb3-188"><a href="#cb3-188" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">3</span>] <span class="ot">&lt;-</span> kruskal_A<span class="sc">$</span>p.value  <span class="co"># Group (from Kruskal-Wallis test)</span></span>
<span id="cb3-189"><a href="#cb3-189" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">4</span>] <span class="ot">&lt;-</span> fold_change  <span class="co"># Fold change</span></span>
<span id="cb3-190"><a href="#cb3-190" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">5</span>] <span class="ot">&lt;-</span> log2_fold_change  <span class="co"># Log2 fold change</span></span>
<span id="cb3-191"><a href="#cb3-191" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">6</span>] <span class="ot">&lt;-</span> mean_healthy  <span class="co"># Mean of the Healthy group</span></span>
<span id="cb3-192"><a href="#cb3-192" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">7</span>] <span class="ot">&lt;-</span> mean_condition  <span class="co"># Mean of the specific condition group</span></span>
<span id="cb3-193"><a href="#cb3-193" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-194"><a href="#cb3-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-195"><a href="#cb3-195" aria-hidden="true" tabindex="-1"></a><span class="co"># Update column names of the significatives data frame</span></span>
<span id="cb3-196"><a href="#cb3-196" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>, <span class="st">"fold_change"</span>, </span>
<span id="cb3-197"><a href="#cb3-197" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"log2_fold_change"</span>, <span class="st">"mean_healthy"</span>, <span class="st">"mean_condition"</span>)</span>
<span id="cb3-198"><a href="#cb3-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-199"><a href="#cb3-199" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the significatives data frame as a TSV file</span></span>
<span id="cb3-200"><a href="#cb3-200" aria-hidden="true" tabindex="-1"></a>tsv_output_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_path, <span class="st">"significatives_nonparametric_with_means_cs.tsv"</span>)</span>
<span id="cb3-201"><a href="#cb3-201" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(significatives, </span>
<span id="cb3-202"><a href="#cb3-202" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> tsv_output_path, </span>
<span id="cb3-203"><a href="#cb3-203" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,          </span>
<span id="cb3-204"><a href="#cb3-204" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>,   </span>
<span id="cb3-205"><a href="#cb3-205" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>,    </span>
<span id="cb3-206"><a href="#cb3-206" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"openxlsx"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"coin"</span>)  </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co"># FILE PATHS</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>frequency_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/BAYESPRISM/RIMOD/theta.state_cellstate.csv"</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>metadata_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/RIMOD_BULK/DATA/rimod_ftd_dataset_table_v3.txt"</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>output_path <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/Rimod/"</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(output_path)) {</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(output_path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>))) {</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(metadata_file)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>freq_cell_types <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(frequency_file)</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> <span class="fu">merge</span>(data, freq_cell_types, <span class="at">by.x =</span> <span class="st">"RimodID"</span>, <span class="at">by.y =</span> <span class="st">"X"</span>)</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>merged_data[merged_data <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="fl">1e-10</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate needed objects</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>m_conditions <span class="ot">&lt;-</span> <span class="fu">unique</span>(merged_data<span class="sc">$</span>DiseaseCode)</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>m_conditions_nh <span class="ot">&lt;-</span> m_conditions[m_conditions <span class="sc">!=</span> <span class="st">"control"</span>]</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>files_to_iterate <span class="ot">&lt;-</span> <span class="fu">colnames</span>(merged_data)[<span class="dv">4</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(merged_data))]</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>m_cond <span class="ot">&lt;-</span> m_conditions_nh[<span class="dv">1</span>]</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>significatives <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">nrow =</span> <span class="fu">length</span>(files_to_iterate), <span class="at">ncol =</span> <span class="dv">3</span>))</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives)<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>)</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (column <span class="cf">in</span> <span class="fu">colnames</span>(freq_cell_types)[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(freq_cell_types))]){</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Kruskal_"</span>, column, <span class="st">"_"</span>, m_cond, <span class="st">".txt"</span>)</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>  subset_data <span class="ot">&lt;-</span> merged_data[merged_data<span class="sc">$</span>DiseaseCode <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"control"</span>, m_cond), </span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">c</span>(<span class="st">"DiseaseCode"</span>, column)]</span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure DiseaseCode is treated as a factor</span></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>  subset_data<span class="sc">$</span>DiseaseCode <span class="ot">&lt;-</span> <span class="fu">factor</span>(subset_data<span class="sc">$</span>DiseaseCode, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"control"</span>, m_cond))</span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate group means</span></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>  mean_healthy <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>DiseaseCode <span class="sc">==</span> <span class="st">"control"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>  mean_condition <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>DiseaseCode <span class="sc">==</span> m_cond], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate fold change</span></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>  fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mean_healthy <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">NA</span>, mean_condition <span class="sc">/</span> mean_healthy)  <span class="co"># Avoid division by zero</span></span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Log-transform the fold change (optional)</span></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>  log2_fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(fold_change), <span class="cn">NA</span>, <span class="fu">log2</span>(fold_change))</span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both groups (Healthy and m_cond) have at least 2 observations</span></span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>  save <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>(<span class="fu">paste0</span>(output_path, filename))}</span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(filename)</span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Model A: Kruskal-Wallis Test (without covariates)</span></span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a>    kruskal_A <span class="ot">&lt;-</span> <span class="fu">kruskal.test</span>(subset_data[[column]], subset_data<span class="sc">$</span>DiseaseCode)</span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"Kruskal-Wallis Test (General model)"</span>)</span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(kruskal_A)</span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>()}</span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Error in processing column "</span>, column, <span class="st">": "</span>, e<span class="sc">$</span>message)</span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store the p-values, fold change, and means in the significatives data frame</span></span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">1</span>] <span class="ot">&lt;-</span> column</span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">3</span>] <span class="ot">&lt;-</span> kruskal_A<span class="sc">$</span>p.value  <span class="co"># Group (from Kruskal-Wallis test)</span></span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">4</span>] <span class="ot">&lt;-</span> fold_change  <span class="co"># Fold change</span></span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">5</span>] <span class="ot">&lt;-</span> log2_fold_change  <span class="co"># Log2 fold change</span></span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">6</span>] <span class="ot">&lt;-</span> mean_healthy  <span class="co"># Mean of the Healthy group</span></span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">7</span>] <span class="ot">&lt;-</span> mean_condition  <span class="co"># Mean of the specific condition group</span></span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a><span class="co"># Update column names of the significatives data frame</span></span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>, <span class="st">"fold_change"</span>, </span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"log2_fold_change"</span>, <span class="st">"mean_healthy"</span>, <span class="st">"mean_condition"</span>)</span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-88"><a href="#cb4-88" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the significatives data frame as a TSV file</span></span>
<span id="cb4-89"><a href="#cb4-89" aria-hidden="true" tabindex="-1"></a>tsv_output_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_path, <span class="st">"significatives_nonparametric_with_means_cs.tsv"</span>)</span>
<span id="cb4-90"><a href="#cb4-90" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(significatives, </span>
<span id="cb4-91"><a href="#cb4-91" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> tsv_output_path, </span>
<span id="cb4-92"><a href="#cb4-92" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,          </span>
<span id="cb4-93"><a href="#cb4-93" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>,   </span>
<span id="cb4-94"><a href="#cb4-94" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>,    </span>
<span id="cb4-95"><a href="#cb4-95" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-96"><a href="#cb4-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-97"><a href="#cb4-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-98"><a href="#cb4-98" aria-hidden="true" tabindex="-1"></a><span class="do">#### ArcSin</span></span>
<span id="cb4-99"><a href="#cb4-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-100"><a href="#cb4-100" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb4-101"><a href="#cb4-101" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"openxlsx"</span>)</span>
<span id="cb4-102"><a href="#cb4-102" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb4-103"><a href="#cb4-103" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"coin"</span>)  </span>
<span id="cb4-104"><a href="#cb4-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-105"><a href="#cb4-105" aria-hidden="true" tabindex="-1"></a><span class="co"># FILE PATHS</span></span>
<span id="cb4-106"><a href="#cb4-106" aria-hidden="true" tabindex="-1"></a>frequency_file <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/TRANSFORMATED_PROPORTIONS/ArcSin/Rimod/C9/Arcsin_Cell_state_proportions.csv"</span></span>
<span id="cb4-107"><a href="#cb4-107" aria-hidden="true" tabindex="-1"></a>metadata_file <span class="ot">&lt;-</span> <span class="st">".../RiMod/Data/rimod_ftd_dataset_table_v3.txt"</span></span>
<span id="cb4-108"><a href="#cb4-108" aria-hidden="true" tabindex="-1"></a>output_path <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/TRANSFORMED_RESULTS/Rimod/C9/ArcSin/"</span></span>
<span id="cb4-109"><a href="#cb4-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-110"><a href="#cb4-110" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(output_path)) {</span>
<span id="cb4-111"><a href="#cb4-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(output_path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-112"><a href="#cb4-112" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-113"><a href="#cb4-113" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>))) {</span>
<span id="cb4-114"><a href="#cb4-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">file.path</span>(output_path,<span class="st">"Significative"</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-115"><a href="#cb4-115" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-116"><a href="#cb4-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-117"><a href="#cb4-117" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(metadata_file, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb4-118"><a href="#cb4-118" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> data<span class="sc">$</span>DiseaseCode</span>
<span id="cb4-119"><a href="#cb4-119" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>DiseaseCode <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb4-120"><a href="#cb4-120" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[<span class="sc">!</span>data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"FTD-GRN"</span>,]</span>
<span id="cb4-121"><a href="#cb4-121" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[<span class="sc">!</span>data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"FTD-MAPT"</span>,]</span>
<span id="cb4-122"><a href="#cb4-122" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>group.ID[data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"control"</span>] <span class="ot">&lt;-</span> <span class="st">"Healthy"</span></span>
<span id="cb4-123"><a href="#cb4-123" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>group.ID[data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"FTD-C9"</span>] <span class="ot">&lt;-</span> <span class="st">"C9orf72"</span></span>
<span id="cb4-124"><a href="#cb4-124" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>sample.ID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb4-125"><a href="#cb4-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-126"><a href="#cb4-126" aria-hidden="true" tabindex="-1"></a>freq_cell_types <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(frequency_file)</span>
<span id="cb4-127"><a href="#cb4-127" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"long"</span>, <span class="st">""</span>, x))</span>
<span id="cb4-128"><a href="#cb4-128" aria-hidden="true" tabindex="-1"></a>freq_cell_types<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">sapply</span>(freq_cell_types<span class="sc">$</span>X, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"X"</span>, <span class="st">""</span>, x))</span>
<span id="cb4-129"><a href="#cb4-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-130"><a href="#cb4-130" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> <span class="fu">merge</span>(data, freq_cell_types, <span class="at">by.x =</span> <span class="st">"sample.ID"</span>, <span class="at">by.y =</span> <span class="st">"X"</span>)</span>
<span id="cb4-131"><a href="#cb4-131" aria-hidden="true" tabindex="-1"></a>merged_data[merged_data <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="fl">1e-10</span></span>
<span id="cb4-132"><a href="#cb4-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-133"><a href="#cb4-133" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate needed objects</span></span>
<span id="cb4-134"><a href="#cb4-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-135"><a href="#cb4-135" aria-hidden="true" tabindex="-1"></a>m_conditions <span class="ot">&lt;-</span> <span class="fu">unique</span>(merged_data<span class="sc">$</span>group.ID)</span>
<span id="cb4-136"><a href="#cb4-136" aria-hidden="true" tabindex="-1"></a>m_conditions_nh <span class="ot">&lt;-</span> m_conditions[m_conditions <span class="sc">!=</span> <span class="st">"Healthy"</span>]</span>
<span id="cb4-137"><a href="#cb4-137" aria-hidden="true" tabindex="-1"></a>files_to_iterate <span class="ot">&lt;-</span> <span class="fu">colnames</span>(merged_data)[<span class="dv">4</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(merged_data))]</span>
<span id="cb4-138"><a href="#cb4-138" aria-hidden="true" tabindex="-1"></a>m_cond <span class="ot">&lt;-</span> m_conditions_nh[<span class="dv">1</span>]</span>
<span id="cb4-139"><a href="#cb4-139" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb4-140"><a href="#cb4-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-141"><a href="#cb4-141" aria-hidden="true" tabindex="-1"></a>significatives <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">nrow =</span> <span class="fu">length</span>(files_to_iterate), <span class="at">ncol =</span> <span class="dv">3</span>))</span>
<span id="cb4-142"><a href="#cb4-142" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives)<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>)</span>
<span id="cb4-143"><a href="#cb4-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-144"><a href="#cb4-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-145"><a href="#cb4-145" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (column <span class="cf">in</span> <span class="fu">colnames</span>(freq_cell_types)[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(freq_cell_types))]){</span>
<span id="cb4-146"><a href="#cb4-146" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb4-147"><a href="#cb4-147" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Kruskal_"</span>, column, <span class="st">"_"</span>, m_cond, <span class="st">".txt"</span>)</span>
<span id="cb4-148"><a href="#cb4-148" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-149"><a href="#cb4-149" aria-hidden="true" tabindex="-1"></a>  subset_data <span class="ot">&lt;-</span> merged_data[merged_data<span class="sc">$</span>group.ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond), </span>
<span id="cb4-150"><a href="#cb4-150" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">c</span>(<span class="st">"group.ID"</span>, column)]</span>
<span id="cb4-151"><a href="#cb4-151" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-152"><a href="#cb4-152" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure group.ID is treated as a factor</span></span>
<span id="cb4-153"><a href="#cb4-153" aria-hidden="true" tabindex="-1"></a>  subset_data<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(subset_data<span class="sc">$</span>group.ID, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Healthy"</span>, m_cond))</span>
<span id="cb4-154"><a href="#cb4-154" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-155"><a href="#cb4-155" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate group means</span></span>
<span id="cb4-156"><a href="#cb4-156" aria-hidden="true" tabindex="-1"></a>  mean_healthy <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"Healthy"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-157"><a href="#cb4-157" aria-hidden="true" tabindex="-1"></a>  mean_condition <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_data[[column]][subset_data<span class="sc">$</span>group.ID <span class="sc">==</span> m_cond], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-158"><a href="#cb4-158" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-159"><a href="#cb4-159" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate fold change</span></span>
<span id="cb4-160"><a href="#cb4-160" aria-hidden="true" tabindex="-1"></a>  fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mean_healthy <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">NA</span>, mean_condition <span class="sc">/</span> mean_healthy)  <span class="co"># Avoid division by zero</span></span>
<span id="cb4-161"><a href="#cb4-161" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-162"><a href="#cb4-162" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Log-transform the fold change (optional)</span></span>
<span id="cb4-163"><a href="#cb4-163" aria-hidden="true" tabindex="-1"></a>  log2_fold_change <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(fold_change), <span class="cn">NA</span>, <span class="fu">log2</span>(fold_change))</span>
<span id="cb4-164"><a href="#cb4-164" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-165"><a href="#cb4-165" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both groups (Healthy and m_cond) have at least 2 observations</span></span>
<span id="cb4-166"><a href="#cb4-166" aria-hidden="true" tabindex="-1"></a>  save <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb4-167"><a href="#cb4-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb4-168"><a href="#cb4-168" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>(<span class="fu">paste0</span>(output_path, filename))}</span>
<span id="cb4-169"><a href="#cb4-169" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-170"><a href="#cb4-170" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(filename)</span>
<span id="cb4-171"><a href="#cb4-171" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-172"><a href="#cb4-172" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Model A: Kruskal-Wallis Test (without covariates)</span></span>
<span id="cb4-173"><a href="#cb4-173" aria-hidden="true" tabindex="-1"></a>    kruskal_A <span class="ot">&lt;-</span> <span class="fu">kruskal.test</span>(subset_data[[column]], subset_data<span class="sc">$</span>group.ID)</span>
<span id="cb4-174"><a href="#cb4-174" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"Kruskal-Wallis Test (General model)"</span>)</span>
<span id="cb4-175"><a href="#cb4-175" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(kruskal_A)</span>
<span id="cb4-176"><a href="#cb4-176" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (save <span class="sc">==</span> <span class="cn">TRUE</span>){<span class="fu">sink</span>()}</span>
<span id="cb4-177"><a href="#cb4-177" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-178"><a href="#cb4-178" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb4-179"><a href="#cb4-179" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Error in processing column "</span>, column, <span class="st">": "</span>, e<span class="sc">$</span>message)</span>
<span id="cb4-180"><a href="#cb4-180" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb4-181"><a href="#cb4-181" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb4-182"><a href="#cb4-182" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-183"><a href="#cb4-183" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store the p-values, fold change, and means in the significatives data frame</span></span>
<span id="cb4-184"><a href="#cb4-184" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">1</span>] <span class="ot">&lt;-</span> column</span>
<span id="cb4-185"><a href="#cb4-185" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">3</span>] <span class="ot">&lt;-</span> kruskal_A<span class="sc">$</span>p.value  <span class="co"># Group (from Kruskal-Wallis test)</span></span>
<span id="cb4-186"><a href="#cb4-186" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">4</span>] <span class="ot">&lt;-</span> fold_change  <span class="co"># Fold change</span></span>
<span id="cb4-187"><a href="#cb4-187" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">5</span>] <span class="ot">&lt;-</span> log2_fold_change  <span class="co"># Log2 fold change</span></span>
<span id="cb4-188"><a href="#cb4-188" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">6</span>] <span class="ot">&lt;-</span> mean_healthy  <span class="co"># Mean of the Healthy group</span></span>
<span id="cb4-189"><a href="#cb4-189" aria-hidden="true" tabindex="-1"></a>  significatives[counter, <span class="dv">7</span>] <span class="ot">&lt;-</span> mean_condition  <span class="co"># Mean of the specific condition group</span></span>
<span id="cb4-190"><a href="#cb4-190" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-191"><a href="#cb4-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-192"><a href="#cb4-192" aria-hidden="true" tabindex="-1"></a><span class="co"># Update column names of the significatives data frame</span></span>
<span id="cb4-193"><a href="#cb4-193" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(significatives) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cell"</span>, <span class="st">"group+sv"</span>, <span class="st">"group"</span>, <span class="st">"fold_change"</span>, </span>
<span id="cb4-194"><a href="#cb4-194" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"log2_fold_change"</span>, <span class="st">"mean_healthy"</span>, <span class="st">"mean_condition"</span>)</span>
<span id="cb4-195"><a href="#cb4-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-196"><a href="#cb4-196" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the significatives data frame as a TSV file</span></span>
<span id="cb4-197"><a href="#cb4-197" aria-hidden="true" tabindex="-1"></a>tsv_output_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_path, <span class="st">"significatives_nonparametric_with_means_cs.tsv"</span>)</span>
<span id="cb4-198"><a href="#cb4-198" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(significatives, </span>
<span id="cb4-199"><a href="#cb4-199" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> tsv_output_path, </span>
<span id="cb4-200"><a href="#cb4-200" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,          </span>
<span id="cb4-201"><a href="#cb4-201" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>,   </span>
<span id="cb4-202"><a href="#cb4-202" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>,    </span>
<span id="cb4-203"><a href="#cb4-203" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="correlation" class="level3">
<h3 class="anchored" data-anchor-id="correlation">Correlation</h3>
<p>Correlate cell-type proportions (raw and transformed) with clinical and pathological variables for FTLD cohorts.</p>
<p>To assess the relationship between cellular composition and neuropathological severity, we performed Spearman correlation analyses between BayesPrism-inferred cell state proportions and neuropathological or molecular covariates derived from frontal cortex tissue from individuals with C9orf72 repeat expansion-associated FTLD (FTLD-C9). Specifically, cell state proportions from the BayesPrism deconvolution were correlated with quantitative pathology scores and molecular readouts of ACSL3, lncRNA, polyGA, polyGR, fociSENSE, fociANTI, pTDP43 and STMN2 across individuals using raw values and transformations to asses non linear correlations (log , log2, logit and arcsin) on both cell proportion and pathology values. Correlation coefficients and p-values were computed using the cor.test function.</p>
<p>In a complementary analysis, for each cell state, Spearman correlations were computed independently for each gene with each of the pathology markers indicated. These results provided information about the differences in the impact of these covariables among cell groups for disease-associated shifts in expression and suggested potential pathophysiological relevance for specific gene expression changes.</p>
<p>Finally, we evaluated the association between transcriptional co-expression modules and the described pathology markers. Module eigengenes (MEs) from each cell state-specific co-expression network were correlated with the corresponding pathology density using Spearman correlation. Modules were ranked by p-value and correlation strength to identify modules whose expression was most strongly associated with neuropathological burden. These correlations were calculated separately for each cell state, and significant modules were retained for downstream interpretation and enrichment analysis.</p>
<p>For the TDP43 samples the same analysis was performed using only pTDP43 and STMN2 pathology markers.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"Seurat"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"RColorBrewer"</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"unikn"</span>)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"cluster"</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"readxl"</span>)  <span class="co"># Asegurar que readxl está cargado</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Load Metadata</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">".../BAYESPRISM_12_3/FTLD_BULK/METADATA/decoder_DeSeq2_FTD_FINAL.xlsx"</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(metadata)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"X"</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> metadata<span class="sc">$</span>X[metadata<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"C9orf72"</span>]</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> metadata[metadata<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"C9orf72"</span>,]</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>CRscores <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">".../BAYESPRISM_12_3/CORRELATION/COVARIABLES/FTD_C9_neuropath_SOM.csv"</span>)</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>CRscores<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">"X"</span>, <span class="at">replacement =</span> <span class="st">""</span>, <span class="at">x =</span> CRscores<span class="sc">$</span>X)</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>CRscores <span class="ot">&lt;-</span> CRscores[CRscores<span class="sc">$</span>X <span class="sc">%in%</span> samples,]</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>proportions_directory <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/BAYESPRISM/FTLD/C9/theta.state_cellstate.csv"</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>proportions_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(proportions_directory, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(proportions_data) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^X"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(proportions_data))</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>common_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(proportions_data), CRscores<span class="sc">$</span>X)</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>proportions_data <span class="ot">&lt;-</span> proportions_data[common_samples, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>covariables <span class="ot">&lt;-</span> CRscores <span class="sc">%&gt;%</span> <span class="fu">filter</span>(X <span class="sc">%in%</span> common_samples) <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">"X"</span>)</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(proportions_data) <span class="sc">&gt;</span> <span class="dv">2</span> <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(covariables) <span class="sc">&gt;</span> <span class="dv">2</span>) {</span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>  results <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">Cell_State =</span> <span class="fu">colnames</span>(proportions_data), </span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>                         <span class="at">Covariate =</span> <span class="fu">colnames</span>(covariables), </span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>                         <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>  compute_spearman <span class="ot">&lt;-</span> <span class="cf">function</span>(cell_state, covariate) {</span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>    test_result <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(proportions_data[, cell_state], covariables[, covariate], </span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>                            <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">exact =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">c</span>(test_result<span class="sc">$</span>estimate, test_result<span class="sc">$</span>p.value))</span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>  cor_pvals <span class="ot">&lt;-</span> <span class="fu">mapply</span>(compute_spearman, results<span class="sc">$</span>Cell_State, results<span class="sc">$</span>Covariate)</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>  results<span class="sc">$</span>Spearman_Correlation <span class="ot">&lt;-</span> cor_pvals[<span class="dv">1</span>, ]</span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>  results<span class="sc">$</span>p_value <span class="ot">&lt;-</span> cor_pvals[<span class="dv">2</span>, ]</span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(results, <span class="st">".../BAYESPRISM_12_3/CORRELATION/C9_spearman_results_true.csv"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a><span class="co"># ARCSIN</span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>CRscores <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">".../BAYESPRISM_12_3/CORRELATION/COVARIABLES/TRANSFORMED_COVARIABLES/ArcSin/C9/ArcSin_FTD_C9_neuropath_SOM.csv"</span>, <span class="at">row.names =</span> <span class="st">"X"</span>)</span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(CRscores) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">"X"</span>, <span class="at">replacement =</span> <span class="st">""</span>, <span class="at">x =</span> <span class="fu">rownames</span>(CRscores))</span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>CRscores <span class="ot">&lt;-</span> CRscores[<span class="fu">rownames</span>(CRscores) <span class="sc">%in%</span> samples,]</span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>proportions_directory <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/TRANSFORMATED_PROPORTIONS/ArcSin/FTLD/C9/Arcsin_Cell_state_proportions.csv"</span></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>proportions_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(proportions_directory, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(proportions_data) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^X"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(proportions_data))</span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>common_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(proportions_data), <span class="fu">rownames</span>(CRscores))</span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>proportions_data <span class="ot">&lt;-</span> proportions_data[common_samples, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>covariables <span class="ot">&lt;-</span> CRscores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">rownames</span>(CRscores) <span class="sc">%in%</span> common_samples) </span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(proportions_data) <span class="sc">&gt;</span> <span class="dv">2</span> <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(covariables) <span class="sc">&gt;</span> <span class="dv">2</span>) {</span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>  results <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">Cell_State =</span> <span class="fu">colnames</span>(proportions_data), </span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a>                         <span class="at">Covariate =</span> <span class="fu">colnames</span>(covariables), </span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>                         <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>  compute_spearman <span class="ot">&lt;-</span> <span class="cf">function</span>(cell_state, covariate) {</span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>    test_result <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(proportions_data[, cell_state], covariables[, covariate], </span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>                            <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">exact =</span> <span class="cn">TRUE</span>) </span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">c</span>(test_result<span class="sc">$</span>estimate, test_result<span class="sc">$</span>p.value))</span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a>  cor_pvals <span class="ot">&lt;-</span> <span class="fu">apply</span>(results, <span class="dv">1</span>, <span class="cf">function</span>(row) {</span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">compute_spearman</span>(row[<span class="st">"Cell_State"</span>], row[<span class="st">"Covariate"</span>])</span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a>  results<span class="sc">$</span>Spearman_Correlation <span class="ot">&lt;-</span> cor_pvals[<span class="dv">1</span>, ]</span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a>  results<span class="sc">$</span>p_value <span class="ot">&lt;-</span> cor_pvals[<span class="dv">2</span>, ]</span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(results, <span class="fu">paste0</span>(outdir, <span class="st">"ArcSin_C9_spearman_results.csv"</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"Seurat"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"RColorBrewer"</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"unikn"</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"cluster"</span>)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"readxl"</span>)  <span class="co"># Asegurar que readxl está cargado</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Load Metadata</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">".../BAYESPRISM_12_3/FTLD_BULK/METADATA/decoder_DeSeq2_FTD_FINAL.xlsx"</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(metadata)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"X"</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> metadata<span class="sc">$</span>X[metadata<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"TDP"</span>]</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> metadata[metadata<span class="sc">$</span>group.ID <span class="sc">==</span> <span class="st">"TDP"</span>,]</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>CRscores <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">".../BAYESPRISM_12_3/CORRELATION/COVARIABLES/FTD_TDP_neuropath_SOM.csv"</span>)</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>CRscores<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">"long"</span>, <span class="at">replacement =</span> <span class="st">""</span>, <span class="at">x =</span> CRscores<span class="sc">$</span>X)</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>CRscores <span class="ot">&lt;-</span> CRscores[CRscores<span class="sc">$</span>X <span class="sc">%in%</span> samples,]</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>proportions_directory <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/BAYESPRISM/FTLD/TDP/theta.state_original.csv"</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>proportions_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(proportions_directory, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(proportions_data) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^X"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(proportions_data))</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>common_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(proportions_data), CRscores<span class="sc">$</span>X)</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>common_samples <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(common_samples, <span class="st">"7BLACK"</span>)  <span class="co"># Remove outlier </span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>proportions_data <span class="ot">&lt;-</span> proportions_data[common_samples, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>covariables <span class="ot">&lt;-</span> CRscores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(X <span class="sc">%in%</span> common_samples) <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">"X"</span>)</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(proportions_data) <span class="sc">&gt;</span> <span class="dv">2</span> <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(covariables) <span class="sc">&gt;</span> <span class="dv">2</span>) {</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>  results <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">Cell_State =</span> <span class="fu">colnames</span>(proportions_data), </span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>                         <span class="at">Covariate =</span> <span class="fu">colnames</span>(covariables), </span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>                         <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>  compute_spearman <span class="ot">&lt;-</span> <span class="cf">function</span>(cell_state, covariate) {</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>    test_result <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(proportions_data[, cell_state], covariables[, covariate], </span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>                            <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">exact =</span> <span class="cn">TRUE</span>) <span class="co"># TRUE </span></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">c</span>(test_result<span class="sc">$</span>estimate, test_result<span class="sc">$</span>p.value))</span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>  cor_pvals <span class="ot">&lt;-</span> <span class="fu">mapply</span>(compute_spearman, results<span class="sc">$</span>Cell_State, results<span class="sc">$</span>Covariate)</span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>  results<span class="sc">$</span>Spearman_Correlation <span class="ot">&lt;-</span> cor_pvals[<span class="dv">1</span>, ]</span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a>  results<span class="sc">$</span>p_value <span class="ot">&lt;-</span> cor_pvals[<span class="dv">2</span>, ]</span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(results, <span class="st">".../BAYESPRISM_12_3/CORRELATION/TDP_spearman_results_true_no_out.csv"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>CRscores <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">".../BAYESPRISM_12_3/CORRELATION/COVARIABLES/TRANSFORMED_COVARIABLES/ArcSin/TDP/ArcSin_FTD_TDP_neuropath_SOM"</span>)</span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>CRscores<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">"long"</span>, <span class="at">replacement =</span> <span class="st">""</span>, <span class="at">x =</span> CRscores<span class="sc">$</span>X)</span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a>CRscores <span class="ot">&lt;-</span> CRscores[CRscores<span class="sc">$</span>X <span class="sc">%in%</span> samples,]</span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a>proportions_directory <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/CELL_PROP/TRANSFORMATED_PROPORTIONS/ArcSin/FTLD/TDP/Arcsin_Cell_state_proportions.csv"</span></span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a>proportions_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(proportions_directory, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(proportions_data) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^X"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(proportions_data))</span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>common_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(proportions_data), CRscores<span class="sc">$</span>X)</span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a>proportions_data <span class="ot">&lt;-</span> proportions_data[common_samples, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a>covariables <span class="ot">&lt;-</span> CRscores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(X <span class="sc">%in%</span> common_samples) <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">"X"</span>)</span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a><span class="co"># ULL AMB AQUEST CANVI DE VALOR</span></span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> covariables[, <span class="st">"STMN2"</span>]</span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a>y[<span class="fu">is.infinite</span>(y)] <span class="ot">&lt;-</span> <span class="fu">min</span>(y[<span class="fu">is.finite</span>(y)]) <span class="sc">-</span> <span class="fl">1e-2</span></span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a>covariables[, <span class="st">"STMN2"</span>] <span class="ot">&lt;-</span> y</span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate STMN2 to remove outlier</span></span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a>cov_stmn2 <span class="ot">&lt;-</span> covariables[<span class="fu">rownames</span>(covariables) <span class="sc">!=</span> <span class="st">"7BLACK"</span>, <span class="st">"STMN2"</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb6-69"><a href="#cb6-69" aria-hidden="true" tabindex="-1"></a>proportions_data_STMN2 <span class="ot">&lt;-</span> proportions_data[<span class="fu">rownames</span>(proportions_data) <span class="sc">!=</span> <span class="st">"7BLACK"</span>, ]</span>
<span id="cb6-70"><a href="#cb6-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-71"><a href="#cb6-71" aria-hidden="true" tabindex="-1"></a>cov_tdp43b <span class="ot">&lt;-</span> covariables[, <span class="st">"TDP43b"</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a><span class="co"># Results dataframe </span></span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">Cell_State =</span> <span class="fu">colnames</span>(proportions_data), </span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a>                       <span class="at">Covariate =</span> <span class="fu">colnames</span>(covariables), </span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a>                       <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-78"><a href="#cb6-78" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Correlations with TDP43b ---</span></span>
<span id="cb6-79"><a href="#cb6-79" aria-hidden="true" tabindex="-1"></a>res_tdp43b <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(<span class="fu">colnames</span>(proportions_data), <span class="cf">function</span>(cs) {</span>
<span id="cb6-80"><a href="#cb6-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">compute_spearman</span>(cs, <span class="st">"TDP43b"</span>, proportions_data, cov_tdp43b)</span>
<span id="cb6-81"><a href="#cb6-81" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb6-82"><a href="#cb6-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-83"><a href="#cb6-83" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Correlations with STMN2 (without outlier) ---</span></span>
<span id="cb6-84"><a href="#cb6-84" aria-hidden="true" tabindex="-1"></a>res_stmn2 <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(<span class="fu">colnames</span>(proportions_data_STMN2), <span class="cf">function</span>(cs) {</span>
<span id="cb6-85"><a href="#cb6-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">compute_spearman</span>(cs, <span class="st">"STMN2"</span>, proportions_data_STMN2, cov_stmn2)</span>
<span id="cb6-86"><a href="#cb6-86" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb6-87"><a href="#cb6-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-88"><a href="#cb6-88" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results</span></span>
<span id="cb6-89"><a href="#cb6-89" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(res_tdp43b, res_stmn2)</span>
<span id="cb6-90"><a href="#cb6-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-91"><a href="#cb6-91" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(results, <span class="fu">paste0</span>(outdir, <span class="st">"ArcSin_TDP_spearman_results.csv"</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/Aumatell\.github\.io\/FTLD_RNAseq_Analysis\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>