<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Single cell analysis – FTLD Analysis Documentation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">FTLD Analysis Documentation</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-introduction" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Introduction</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-introduction">    
        <li>
    <a class="dropdown-item" href="../Introduction/Introduction.html">
 <span class="dropdown-text">Introduction</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Introduction/Background.html">
 <span class="dropdown-text">Background</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Introduction/Cohort.html">
 <span class="dropdown-text">Cohort</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-methods" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Methods</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-methods">    
        <li>
    <a class="dropdown-item" href="../Methods/Bulk_dea.html">
 <span class="dropdown-text">Bulk DEA</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Methods/Bayesprism.html">
 <span class="dropdown-text">Bayesian Deconvolution</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Methods/Cell_prop.html">
 <span class="dropdown-text">Cell proportions analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Methods/Sc_dea.html">
 <span class="dropdown-text">Single-Cell Analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Methods/Hdwgcna.html">
 <span class="dropdown-text">Co-expression Networks analysis</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-results" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Results</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-results">    
        <li>
    <a class="dropdown-item" href="../Results/Bulk_dea_results.html">
 <span class="dropdown-text">Bulk Differential Expression</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Cell_composition.html">
 <span class="dropdown-text">Cell Composition Differences</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Cell_correlation.html">
 <span class="dropdown-text">Cell Composition × Neuropathology</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Sc_dea_results_C9.html">
 <span class="dropdown-text">Single-Cell Differential Expression (C9) 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Sc_dea_results_C9_2.html">
 <span class="dropdown-text">Single-Cell Differential Expression (C9) 2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Sc_dea_results_TDP.html">
 <span class="dropdown-text">Single-Cell Differential Expression (TDP) 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Sc_dea_results_TDP_2.html">
 <span class="dropdown-text">Single-Cell Differential Expression (TDP) 2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Genes_FTLD_summary.html">
 <span class="dropdown-text">Single-Cell FTLD-related genes DE</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Sc_gene_correlation.html">
 <span class="dropdown-text">Single-Cell Gene Correlation</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Hdwgcna_differences.html">
 <span class="dropdown-text">hdWGCNA Differences</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Hdwgcna_C9_correlation.html">
 <span class="dropdown-text">C9 hdWGCNA Correlation</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Results/Hdwgcna_TDP_correlation.html">
 <span class="dropdown-text">TDP hdWGCNA Correlation</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-validation" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Validation</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-validation">    
        <li>
    <a class="dropdown-item" href="../Validation/Bulk_validation.html">
 <span class="dropdown-text">Bulk Analysis Validation</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Validation/Sc_validation.html">
 <span class="dropdown-text">Single-Cell Validation</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Validation/Hdwgcna_validation.html">
 <span class="dropdown-text">hdWGCNA Validation</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-discussion" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Discussion</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-discussion">    
        <li>
    <a class="dropdown-item" href="../Discussion/Discussion.html">
 <span class="dropdown-text">Discussion</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-references" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">References</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-references">    
        <li>
    <a class="dropdown-item" href="../References/References.html">
 <span class="dropdown-text">References</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#single-cell-differencial-expression-analysis" id="toc-single-cell-differencial-expression-analysis" class="nav-link active" data-scroll-target="#single-cell-differencial-expression-analysis"><strong>Single-Cell</strong> differencial expression <strong>Analysis</strong></a>
  <ul class="collapse">
  <li><a href="#analytical-strategy" id="toc-analytical-strategy" class="nav-link" data-scroll-target="#analytical-strategy"><strong>Analytical Strategy</strong></a>
  <ul class="collapse">
  <li><a href="#input-data-bayesprism-derived-cell-state-matrices" id="toc-input-data-bayesprism-derived-cell-state-matrices" class="nav-link" data-scroll-target="#input-data-bayesprism-derived-cell-state-matrices"><strong>1. Input Data: BayesPrism-Derived Cell-State Matrices</strong></a></li>
  <li><a href="#preprocessing-and-gene-filtering" id="toc-preprocessing-and-gene-filtering" class="nav-link" data-scroll-target="#preprocessing-and-gene-filtering"><strong>2. Preprocessing and Gene Filtering</strong></a></li>
  <li><a href="#modeling-latent-sources-of-variation" id="toc-modeling-latent-sources-of-variation" class="nav-link" data-scroll-target="#modeling-latent-sources-of-variation"><strong>3. Modeling Latent Sources of Variation</strong></a></li>
  <li><a href="#differential-expression-testing" id="toc-differential-expression-testing" class="nav-link" data-scroll-target="#differential-expression-testing"><strong>4. Differential Expression Testing</strong></a></li>
  <li><a href="#validation-across-independent-cohorts" id="toc-validation-across-independent-cohorts" class="nav-link" data-scroll-target="#validation-across-independent-cohorts"><strong>5. Validation Across Independent Cohorts</strong></a></li>
  </ul></li>
  <li><a href="#analytical-goals-of-the-single-cellresolved-differential-expression-analysis" id="toc-analytical-goals-of-the-single-cellresolved-differential-expression-analysis" class="nav-link" data-scroll-target="#analytical-goals-of-the-single-cellresolved-differential-expression-analysis"><strong>Analytical Goals of the Single-Cell–Resolved Differential Expression Analysis</strong></a></li>
  </ul></li>
  <li><a href="#correlation-of-genes-of-interest-per-cell-state-with-neuropathology-markers" id="toc-correlation-of-genes-of-interest-per-cell-state-with-neuropathology-markers" class="nav-link" data-scroll-target="#correlation-of-genes-of-interest-per-cell-state-with-neuropathology-markers">Correlation of genes of interest per cell state with neuropathology markers</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Single cell analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="single-cell-differencial-expression-analysis" class="level1">
<h1><strong>Single-Cell</strong> differencial expression <strong>Analysis</strong></h1>
<p>Bulk RNA-seq differential expression identifies tissue-level transcriptional changes but cannot distinguish whether these alterations originate from specific neuronal subclasses, glial populations, or vascular/immune lineages. Because FTLD pathology is known to selectively affect distinct cortical cell populations—such as <strong>RORB-expressing excitatory neurons</strong>, subsets of <strong>inhibitory neurons</strong>, and diverse <strong>astroglial</strong> and <strong>microglial</strong> states—bulk analysis alone provides limited biological resolution.</p>
<p>To overcome this limitation, we leveraged the <strong>cell-state–specific expression matrices</strong> generated by <strong>BayesPrism</strong>, which infer gene-level expression for each of the 44 transcriptional states defined in the Pineda et al.&nbsp;single-nucleus reference. These matrices represent “pseudobulk” expression profiles for each cell state within each sample, enabling differential gene expression analysis at <strong>single-cell resolution using bulk RNA-seq</strong>.</p>
<p>Applying edgeR <span class="citation" data-cites="robinson2010_edger">(<a href="#ref-robinson2010_edger" role="doc-biblioref">Robinson, McCarthy, and Smyth 2010</a>)</span> to these deconvolved matrices allows rigorous detection of transcriptional alterations <strong>within individual cell states</strong>, thereby identifying the precise cellular compartments responsible for disease-associated dysregulation in FTLD-C9 and FTLD-TDP.</p>
<section id="analytical-strategy" class="level2">
<h2 class="anchored" data-anchor-id="analytical-strategy"><strong>Analytical Strategy</strong></h2>
<section id="input-data-bayesprism-derived-cell-state-matrices" class="level3">
<h3 class="anchored" data-anchor-id="input-data-bayesprism-derived-cell-state-matrices"><strong>1. Input Data: BayesPrism-Derived Cell-State Matrices</strong></h3>
<p>For each sample, BayesPrism returned <strong>cell-state–specific gene expression estimates</strong> for all 43–44 states in the Pineda et al.&nbsp;reference. These matrices were:</p>
<ul>
<li><p>aggregated per cell state across samples,</p></li>
<li><p>treated as pseudobulk counts,</p></li>
</ul>
<p>used as direct input to edgeR differential expression workflows.</p>
<p>Only cell states with <strong>sufficient expression and non-zero counts</strong> across samples were retained for statistical modeling.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"edgeR"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"GO.db"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># c9</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>CSV_DECONVOLDED_PATH <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/BAYESPRISM/FTLD/C9/CELL STATE ORIGINAL"</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>CASE_LEGEND <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BayesPrism/FTD/DATASETS/FTD_QUIM/decoder_DeSeq2_FTD_FINAL.xlsx"</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>SV_COVARIATE <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/FTLD_BULK/SURROGATED_VARIABLES/SV1_C9_HC_QUIM.csv"</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>OUTPUT_DIRECTORY <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/EDGER/FTLD/C9/CS_SV"</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>REFERENCE_GROUP <span class="ot">&lt;-</span> <span class="st">"Healthy"</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>covariate_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(SV_COVARIATE)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>covariate_data<span class="sc">$</span>sample.ID <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^long"</span>, <span class="st">""</span>, covariate_data<span class="sc">$</span>sample.ID)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>csv_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(CSV_DECONVOLDED_PATH, <span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.csv$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">endsWith</span>(CASE_LEGEND, <span class="st">".csv"</span>)) {</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>  case_legend <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(CASE_LEGEND)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(case_legend<span class="sc">$</span>group.ID)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> <span class="cf">if</span> (<span class="fu">endsWith</span>(CASE_LEGEND, <span class="st">".xlsx"</span>)) {</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>  case_legend <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(CASE_LEGEND)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(case_legend<span class="sc">$</span>group.ID)</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group2_ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(case_legend<span class="sc">$</span>group2.ID)</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group2.ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Handle other cases or file types</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">"File type not recognized."</span>)</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group2_ID <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(case_legend<span class="sc">$</span>group2_ID)</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">as.character</span>(case_legend<span class="sc">$</span>group_ID)</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(case_legend<span class="sc">$</span>group_ID)</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(case_legend<span class="sc">$</span>group_ID)</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>conditions <span class="ot">&lt;-</span> <span class="fu">levels</span>(case_legend<span class="sc">$</span>group_ID)</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (REFERENCE_GROUP <span class="sc">%in%</span> conditions) {</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>  conditions <span class="ot">&lt;-</span> conditions[conditions <span class="sc">!=</span> REFERENCE_GROUP]</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(OUTPUT_DIRECTORY)) {</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(OUTPUT_DIRECTORY, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>matrix_data <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(conditions)<span class="sc">*</span><span class="fu">length</span>(csv_files), <span class="at">nrow =</span> <span class="fu">length</span>(conditions), <span class="at">ncol =</span> <span class="fu">length</span>(csv_files))</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a DataFrame with the matrix data, specifying column and row names</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(matrix_data, <span class="at">row.names =</span> conditions)</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">lapply</span>(csv_files, <span class="cf">function</span>(file) tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="preprocessing-and-gene-filtering" class="level3">
<h3 class="anchored" data-anchor-id="preprocessing-and-gene-filtering"><strong>2. Preprocessing and Gene Filtering</strong></h3>
<p>For each cell state:</p>
<ul>
<li><p>Genes with zero or near-zero counts were removed.</p></li>
<li><p>Only transcripts expressed in ≥1 sample were kept.</p></li>
<li><p>Mitochondrial, ribosomal, and sex-chromosome genes had already been removed at the BayesPrism stage.</p></li>
<li><p>average CPM</p></li>
<li><p>normalization</p></li>
</ul>
<p>This ensured that differential expression reflected biologically meaningful changes rather than sparsity-related artifacts.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>comparison_data <span class="ot">&lt;-</span> case_legend[case_legend<span class="sc">$</span>group_ID <span class="sc">%in%</span> <span class="fu">c</span>(REFERENCE_GROUP, condition), ]</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>SV1 <span class="ot">&lt;-</span> covariate_data<span class="sc">$</span>SV1[<span class="fu">match</span>(comparison_data<span class="sc">$</span>sample.ID, covariate_data<span class="sc">$</span>sample.ID)]</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">levels =</span> <span class="fu">c</span>(REFERENCE_GROUP, condition))</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">order</span>(comparison_data<span class="sc">$</span>group_ID), ]</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>      common_ids <span class="ot">&lt;-</span> <span class="fu">intersect</span>(comparison_data<span class="sc">$</span>sample.ID, <span class="fu">rownames</span>(data))</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>      data_subset <span class="ot">&lt;-</span> data[<span class="fu">rownames</span>(data) <span class="sc">%in%</span> common_ids, ]</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>      data_ordered <span class="ot">&lt;-</span> data_subset[<span class="fu">match</span>(comparison_data<span class="sc">$</span>sample.ID, <span class="fu">rownames</span>(data_subset)), ]</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>      <span class="co"># CLEAN NA</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>      data_ordered[<span class="sc">!</span><span class="fu">is.finite</span>(<span class="fu">as.matrix</span>(data_ordered))] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>      data_ordered <span class="ot">&lt;-</span> data_ordered[<span class="fu">complete.cases</span>(data_ordered), ]</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">match</span>(<span class="fu">rownames</span>(data_ordered), comparison_data<span class="sc">$</span>sample.ID), ]</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>      group <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID[comparison_data<span class="sc">$</span>group_ID <span class="sc">%in%</span> <span class="fu">c</span>(REFERENCE_GROUP, condition)])</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts=</span><span class="fu">t</span>(data_ordered),<span class="at">group=</span>group)</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> y[, <span class="fu">colSums</span>(y<span class="sc">$</span>counts) <span class="sc">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Keep only samples in comparison_data that are in y</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">match</span>(<span class="fu">colnames</span>(y), comparison_data<span class="sc">$</span>sample.ID), ]</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Average Log CPM histogram</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>      AveLogCPM <span class="ot">&lt;-</span> <span class="fu">aveLogCPM</span>(y)</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">normLibSizes</span>(y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="modeling-latent-sources-of-variation" class="level3">
<h3 class="anchored" data-anchor-id="modeling-latent-sources-of-variation"><strong>3. Modeling Latent Sources of Variation</strong></h3>
<p>To account for unwanted technical and biological variability:</p>
<ul>
<li><strong>Surrogate variables (SVs)</strong> estimated from the bulk RNA-seq data were included as covariates in the design matrix, controlling for batch-like or hidden confounders.</li>
</ul>
<p>This approach improves statistical robustness, especially in modest sample sizes typical of post-mortem FTLD cohorts.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>   group <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">levels =</span> <span class="fu">c</span>(REFERENCE_GROUP, condition))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">relevel</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">ref =</span> REFERENCE_GROUP)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>      design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> comparison_data<span class="sc">$</span>group_ID <span class="sc">+</span> comparison_data<span class="sc">$</span>SV1, <span class="at">data =</span> comparison_data)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">c</span>(REFERENCE_GROUP, condition, <span class="st">"SV1"</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(design) <span class="sc">==</span> <span class="fu">ncol</span>(y)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>      <span class="co"># DISPERSION ESTIMATION</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(y,design)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>      fit <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(y,design)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="differential-expression-testing" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-testing"><strong>4. Differential Expression Testing</strong></h3>
<p>For each of the 43–44 cell states:</p>
<ul>
<li><p>A design matrix modeled diagnostic status (FTLD-C9 or FTLD-TDP) with <strong>Healthy</strong> as the reference level.</p></li>
<li><p>A single contrast per cell state tested the effect of disease.</p></li>
<li><p>The <strong>glmLRT()</strong> framework was applied.</p></li>
</ul>
<p>Multiple-testing correction within each cell state was performed using <strong>Benjamini–Hochberg FDR</strong>, with significance defined as:</p>
<ul>
<li><strong>FDR &lt; 0.05</strong></li>
</ul>
<p>For each significant gene, the following metrics were extracted:</p>
<ul>
<li><p>log2 fold change</p></li>
<li><p>raw p-value and FDR-adjusted p-value</p></li>
<li><p>average CPM per condition</p></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>     <span class="co"># Diferential expression analysis</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>      B.LvsP <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="fu">paste0</span>(condition, <span class="st">" - "</span>, REFERENCE_GROUP), <span class="at">levels=</span>design)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fit, <span class="at">contrast=</span>B.LvsP)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>      is.de <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(res, <span class="at">adjust.method =</span> <span class="st">"fdr"</span>, <span class="at">p.value =</span> <span class="fl">0.05</span>, <span class="at">lfc=</span> <span class="dv">0</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> res<span class="sc">$</span>table</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>      results_table<span class="sc">$</span>adj_pval <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(results_table<span class="sc">$</span>PValue, <span class="at">method =</span> <span class="st">"fdr"</span>, <span class="at">n =</span> <span class="fu">length</span>(results_table<span class="sc">$</span>PValue))</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> <span class="fu">cbind</span>(results_table, <span class="fu">data.frame</span>(is.de))</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>      <span class="co"># SAVING CSV RESULTS</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write.csv</span>(res<span class="sc">$</span>table, <span class="at">file =</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"res_h.csv"</span>))</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write.csv</span>(results_table, <span class="at">file =</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"results_adj_h.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="validation-across-independent-cohorts" class="level3">
<h3 class="anchored" data-anchor-id="validation-across-independent-cohorts"><strong>5. Validation Across Independent Cohorts</strong></h3>
<p>To assess reproducibility:</p>
<ul>
<li><p>Cell-state DEGs from the Sant Pau FTLD-C9 dataset were compared with those from the <strong>Menden et al.</strong> dataset.</p></li>
<li><p>Cell-state DEGs from the Sant Pau FTLD-TDP dataset were compared with the <strong>Pottier et al.</strong> cohort.</p></li>
</ul>
<p>Overlap significance was quantified using:</p>
<ul>
<li><p><strong>GeneOverlap statistics</strong></p></li>
<li><p><strong>Jaccard indices</strong></p></li>
</ul>
<p>This allowed identification of <strong>robust cell-state–specific signatures</strong> conserved across cohorts.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># JACCARD </span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"openxlsx"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"GeneOverlap"</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"VennDiagram"</span>)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"grid"</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Define functions</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Jaccard similarity index</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>jaccard_similarity <span class="ot">&lt;-</span> <span class="cf">function</span>(list1, list2) {</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  intersection_size <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(list1, list2))</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  union_size <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">union</span>(list1, list2))</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(intersection_size <span class="sc">/</span> union_size)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform permutation test for p-value</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>permutation_test <span class="ot">&lt;-</span> <span class="cf">function</span>(list1, list2, <span class="at">n_permutations =</span> <span class="dv">10000</span>) {</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  observed_jaccard <span class="ot">&lt;-</span> <span class="fu">jaccard_similarity</span>(list1, list2)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  permuted_jaccards <span class="ot">&lt;-</span> <span class="fu">replicate</span>(n_permutations, {</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    permuted_list1 <span class="ot">&lt;-</span> <span class="fu">sample</span>(list1)</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>    permuted_list2 <span class="ot">&lt;-</span> <span class="fu">sample</span>(list2)</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">jaccard_similarity</span>(permuted_list1, permuted_list2)</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  p_value <span class="ot">&lt;-</span> <span class="fu">mean</span>(permuted_jaccards <span class="sc">&gt;=</span> observed_jaccard)</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(p_value)</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>hypergeometric_test <span class="ot">&lt;-</span> <span class="cf">function</span>(list1, list2, total_population_size) {</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>  intersection_size <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(list1, list2))</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>  total_genes <span class="ot">&lt;-</span> total_population_size  <span class="co"># Total number of possible genes</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Number of DEGs in list1 and list2</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>  K1 <span class="ot">&lt;-</span> <span class="fu">length</span>(list1)</span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>  K2 <span class="ot">&lt;-</span> <span class="fu">length</span>(list2)</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Hypergeometric test to calculate p-value</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>  p_value <span class="ot">&lt;-</span> <span class="fu">phyper</span>(intersection_size <span class="sc">-</span> <span class="dv">1</span>, K1, total_genes <span class="sc">-</span> K1, K2, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(p_value)</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a><span class="co"># TDP43</span></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Define variables</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>CStates <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Arterial"</span>, <span class="st">"DISC1_RELN"</span>, <span class="st">"Pericyte"</span>, <span class="st">"SMC"</span>, <span class="st">"TLE4_CCBE1"</span>,</span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Capillary"</span>, <span class="st">"GFAP-neg"</span>, <span class="st">"PVALB_CEMIP"</span>, <span class="st">"SST_ADAMTS19"</span>, <span class="st">"TLE4_MEGF11"</span>,</span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CDH4_CCK"</span>, <span class="st">"GFAP-pos"</span>, <span class="st">"PVALB_MYBPC1"</span>, <span class="st">"SST_BRINP3"</span>, <span class="st">"TLE4_SEMA3D"</span>,</span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CDH4_SCGN"</span>, <span class="st">"LAMP5_CA3"</span>, <span class="st">"PVALB_PTHLH"</span>, <span class="st">"SST_GALNT14"</span>, <span class="st">"VAT1L_EYA4"</span>,</span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CLMP_KCNMA1"</span>, <span class="st">"LAMP5_PMEPA1"</span>, <span class="st">"RORB_ADGRL4"</span>, <span class="st">"SST_NPY"</span>, <span class="st">"Venous"</span>,</span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CLMP_PDGFRA"</span>, <span class="st">"Micro"</span>, <span class="st">"RORB_FOXO1"</span>, <span class="st">"VIP_CLSTN2"</span>,</span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CUX2_RASGRF2"</span>, <span class="st">"Oligo"</span>, <span class="st">"RORB_LRRK1"</span>, <span class="st">"T_Cell"</span>, <span class="st">"VIP_HTR2C"</span>,</span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CUX2_RORB"</span>, <span class="st">"OPC"</span>, <span class="st">"RORB_POU3F2"</span>, <span class="st">"THEMIS_NR4A2"</span>, <span class="st">"VIP_LAMA3"</span>,</span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>  <span class="st">"DISC1_CCK"</span>, <span class="st">"PCP4_NXPH2"</span>, <span class="st">"SCN4B_NEFH"</span>, <span class="st">"THEMIS_TMEM233"</span></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>FTLD_TDP <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/EDGER/FTLD/TDP/CS_LRT/"</span></span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>NOU <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/EDGER/NEW/CS_SVcorrect/"</span></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>RESULTS_PATH <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/OVERLAP/TDP_NEW/CS_Correcte"</span></span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate output dataframe</span></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>Results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">ncol =</span> <span class="fu">length</span>(CStates), <span class="at">nrow =</span> <span class="dv">10</span>))</span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Results) <span class="ot">&lt;-</span> CStates</span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Results) <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Upregulated FTLD_TDP"</span>, <span class="st">"Upregulated Nou"</span>, </span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Downregulated FTLD_TDP"</span>, <span class="st">"Downregulated Nou"</span>,</span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CU FTLD – Nou"</span>, <span class="st">"CD FTLD – Nou"</span>, </span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>  <span class="st">"JU FTLD – Nou"</span>, <span class="st">"JD FTLD – Nou"</span>,</span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GU FTLD – Nou"</span>, <span class="st">"GD FTLD – Nou"</span>)</span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate to obtain the results for each case.</span></span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(CStates)) {</span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>  CS <span class="ot">&lt;-</span> CStates[i]</span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Read data</span></span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">file.exists</span>(<span class="fu">paste0</span>(FTLD_TDP, CS ,<span class="st">"/"</span>, <span class="st">"TDP/results_adj_h.csv"</span>))){</span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a>    ftld_tdp <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(FTLD_TDP, CS ,<span class="st">"/"</span>, <span class="st">"TDP/results_adj_h.csv"</span>))</span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a>    ftld_tdp <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">file.exists</span>(<span class="fu">paste0</span>(NOU,CS,<span class="st">"/"</span> ,<span class="st">"TDP/results_adj_h.csv"</span>))){</span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a>    nou <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(NOU,CS,<span class="st">"/"</span> ,<span class="st">"TDP/results_adj_h.csv"</span>))</span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb5-85"><a href="#cb5-85" aria-hidden="true" tabindex="-1"></a>    nou <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb5-86"><a href="#cb5-86" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-87"><a href="#cb5-87" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-88"><a href="#cb5-88" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract DEG</span></span>
<span id="cb5-89"><a href="#cb5-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-90"><a href="#cb5-90" aria-hidden="true" tabindex="-1"></a>  <span class="co"># For U_ftld</span></span>
<span id="cb5-91"><a href="#cb5-91" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(ftld_tdp<span class="sc">$</span>X[ftld_tdp<span class="sc">$</span>X1.TDP..<span class="fl">1.</span>Healthy <span class="sc">==</span> <span class="dv">1</span>])) {</span>
<span id="cb5-92"><a href="#cb5-92" aria-hidden="true" tabindex="-1"></a>    U_ftld <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="dv">0</span>)  <span class="co"># Assign an empty vector if NULL</span></span>
<span id="cb5-93"><a href="#cb5-93" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb5-94"><a href="#cb5-94" aria-hidden="true" tabindex="-1"></a>    U_ftld <span class="ot">&lt;-</span> ftld_tdp<span class="sc">$</span>X[ftld_tdp<span class="sc">$</span>X1.TDP..<span class="fl">1.</span>Healthy <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb5-95"><a href="#cb5-95" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-96"><a href="#cb5-96" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-97"><a href="#cb5-97" aria-hidden="true" tabindex="-1"></a>  <span class="co"># For U_nou</span></span>
<span id="cb5-98"><a href="#cb5-98" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(nou<span class="sc">$</span>X[nou<span class="sc">$</span>X.<span class="fl">1.</span>Control.<span class="fl">1.</span>TDP <span class="sc">==</span> <span class="dv">1</span>])) {</span>
<span id="cb5-99"><a href="#cb5-99" aria-hidden="true" tabindex="-1"></a>    U_nou <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="dv">0</span>)  <span class="co"># Assign an empty vector if NULL</span></span>
<span id="cb5-100"><a href="#cb5-100" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb5-101"><a href="#cb5-101" aria-hidden="true" tabindex="-1"></a>    U_nou <span class="ot">&lt;-</span> nou<span class="sc">$</span>X[nou<span class="sc">$</span>X.<span class="fl">1.</span>Control.<span class="fl">1.</span>TDP <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb5-102"><a href="#cb5-102" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-103"><a href="#cb5-103" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-104"><a href="#cb5-104" aria-hidden="true" tabindex="-1"></a>  <span class="co"># For D_ftld</span></span>
<span id="cb5-105"><a href="#cb5-105" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(ftld_tdp<span class="sc">$</span>X[ftld_tdp<span class="sc">$</span>X1.TDP..<span class="fl">1.</span>Healthy <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>])) {</span>
<span id="cb5-106"><a href="#cb5-106" aria-hidden="true" tabindex="-1"></a>    D_ftld <span class="ot">&lt;-</span><span class="fu">character</span>(<span class="dv">0</span>)  <span class="co"># Assign an empty vector if NULL</span></span>
<span id="cb5-107"><a href="#cb5-107" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb5-108"><a href="#cb5-108" aria-hidden="true" tabindex="-1"></a>    D_ftld <span class="ot">&lt;-</span> ftld_tdp<span class="sc">$</span>X[ftld_tdp<span class="sc">$</span>X1.TDP..<span class="fl">1.</span>Healthy <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb5-109"><a href="#cb5-109" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-110"><a href="#cb5-110" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-111"><a href="#cb5-111" aria-hidden="true" tabindex="-1"></a>  <span class="co"># For D_nou</span></span>
<span id="cb5-112"><a href="#cb5-112" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(nou<span class="sc">$</span>X[nou<span class="sc">$</span>X.<span class="fl">1.</span>Control.<span class="fl">1.</span>TDP <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>])) {</span>
<span id="cb5-113"><a href="#cb5-113" aria-hidden="true" tabindex="-1"></a>    D_nou <span class="ot">&lt;-</span><span class="fu">character</span>(<span class="dv">0</span>) <span class="co"># Assign an empty vector if NULL</span></span>
<span id="cb5-114"><a href="#cb5-114" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb5-115"><a href="#cb5-115" aria-hidden="true" tabindex="-1"></a>    D_nou <span class="ot">&lt;-</span> nou<span class="sc">$</span>X[nou<span class="sc">$</span>X.<span class="fl">1.</span>Control.<span class="fl">1.</span>TDP <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb5-116"><a href="#cb5-116" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-117"><a href="#cb5-117" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-118"><a href="#cb5-118" aria-hidden="true" tabindex="-1"></a>  <span class="co"># OVERLAP GENES</span></span>
<span id="cb5-119"><a href="#cb5-119" aria-hidden="true" tabindex="-1"></a>  C_U_ftld_nou <span class="ot">&lt;-</span> <span class="fu">intersect</span>(U_ftld, U_nou)</span>
<span id="cb5-120"><a href="#cb5-120" aria-hidden="true" tabindex="-1"></a>  C_D_ftld_nou <span class="ot">&lt;-</span> <span class="fu">intersect</span>(D_ftld, D_nou)</span>
<span id="cb5-121"><a href="#cb5-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-122"><a href="#cb5-122" aria-hidden="true" tabindex="-1"></a>  <span class="co"># SIMILARITY INDEX</span></span>
<span id="cb5-123"><a href="#cb5-123" aria-hidden="true" tabindex="-1"></a>  Ujaccard13 <span class="ot">&lt;-</span> <span class="fu">jaccard_similarity</span>(U_ftld, U_nou)</span>
<span id="cb5-124"><a href="#cb5-124" aria-hidden="true" tabindex="-1"></a>  Djaccard13 <span class="ot">&lt;-</span> <span class="fu">jaccard_similarity</span>(D_ftld, D_nou)</span>
<span id="cb5-125"><a href="#cb5-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-126"><a href="#cb5-126" aria-hidden="true" tabindex="-1"></a>  <span class="co"># HIPERGEOMETRIC TESTS</span></span>
<span id="cb5-127"><a href="#cb5-127" aria-hidden="true" tabindex="-1"></a>  total_genes_FN <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">c</span>(ftld_tdp<span class="sc">$</span>X, nou<span class="sc">$</span>X)))</span>
<span id="cb5-128"><a href="#cb5-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-129"><a href="#cb5-129" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Gene Overlap</span></span>
<span id="cb5-130"><a href="#cb5-130" aria-hidden="true" tabindex="-1"></a>  GU_13 <span class="ot">=</span> <span class="fu">testGeneOverlap</span>(<span class="fu">newGeneOverlap</span>(U_ftld, U_nou, total_genes_FN))<span class="sc">@</span>pval</span>
<span id="cb5-131"><a href="#cb5-131" aria-hidden="true" tabindex="-1"></a>  GD_13 <span class="ot">=</span> <span class="fu">testGeneOverlap</span>(<span class="fu">newGeneOverlap</span>(D_ftld, D_nou, total_genes_FN))<span class="sc">@</span>pval</span>
<span id="cb5-132"><a href="#cb5-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-133"><a href="#cb5-133" aria-hidden="true" tabindex="-1"></a>  <span class="co"># UPDATE RESULTS TABLE WITH THE CALCULATED VALUES.</span></span>
<span id="cb5-134"><a href="#cb5-134" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"Upregulated FTLD_TDP"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(U_ftld)</span>
<span id="cb5-135"><a href="#cb5-135" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"Upregulated Nou"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(U_nou)</span>
<span id="cb5-136"><a href="#cb5-136" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"Downregulated FTLD_TDP"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(D_ftld)</span>
<span id="cb5-137"><a href="#cb5-137" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"Downregulated Nou"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(D_nou)</span>
<span id="cb5-138"><a href="#cb5-138" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"CU FTLD – Nou"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(C_U_ftld_nou)</span>
<span id="cb5-139"><a href="#cb5-139" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"CD FTLD – Nou"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(C_D_ftld_nou)</span>
<span id="cb5-140"><a href="#cb5-140" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"JU FTLD – Nou"</span>, CS] <span class="ot">&lt;-</span> Ujaccard13</span>
<span id="cb5-141"><a href="#cb5-141" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"JD FTLD – Nou"</span>, CS] <span class="ot">&lt;-</span> Djaccard13</span>
<span id="cb5-142"><a href="#cb5-142" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"GU FTLD – Nou"</span>, CS] <span class="ot">&lt;-</span> GU_13</span>
<span id="cb5-143"><a href="#cb5-143" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"GD FTLD – Nou"</span>, CS] <span class="ot">&lt;-</span> GD_13</span>
<span id="cb5-144"><a href="#cb5-144" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-145"><a href="#cb5-145" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-146"><a href="#cb5-146" aria-hidden="true" tabindex="-1"></a>Results <span class="ot">&lt;-</span> <span class="fu">t</span>(Results)</span>
<span id="cb5-147"><a href="#cb5-147" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(Results)</span>
<span id="cb5-148"><a href="#cb5-148" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(Results[<span class="fu">rowSums</span>(Results[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]) <span class="sc">&gt;</span> <span class="dv">0</span>, ])</span>
<span id="cb5-149"><a href="#cb5-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-150"><a href="#cb5-150" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(Results, <span class="at">file =</span> <span class="fu">paste0</span>(RESULTS_PATH, <span class="st">"/TDP43_Validation_results_cs.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># C9</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>FTLD_TDP <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/EDGER/FTLD/C9/CS_SV/"</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>NOU <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/EDGER/RIMOD/CS_SV/"</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>RESULTS_PATH <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/OVERLAP/C9_NEW/CS"</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate output dataframe</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>Results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">ncol =</span> <span class="fu">length</span>(CStates), <span class="at">nrow =</span> <span class="dv">10</span>))</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Results) <span class="ot">&lt;-</span> CStates</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Results) <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Upregulated FTLD_TDP"</span>, <span class="st">"Upregulated Rimod"</span>, </span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Downregulated FTLD_TDP"</span>, <span class="st">"Downregulated Rimod"</span>,</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CU FTLD – Rimod"</span>, <span class="st">"CD FTLD – Rimod"</span>, </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">"JU FTLD – Rimod"</span>, <span class="st">"JD FTLD – Rimod"</span>,</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GU FTLD – Rimod"</span>, <span class="st">"GD FTLD – Rimod"</span>)</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate to obtain the results for each case.</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(CStates)) {</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  CS <span class="ot">&lt;-</span> CStates[i]</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read data</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">file.exists</span>(<span class="fu">paste0</span>(FTLD_TDP, CS ,<span class="st">"/"</span>, <span class="st">"C9orf72/results_adj_h.csv"</span>))){</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>    ftld_tdp <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(FTLD_TDP, CS ,<span class="st">"/"</span>, <span class="st">"C9orf72/results_adj_h.csv"</span>))</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>    ftld_tdp <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">file.exists</span>(<span class="fu">paste0</span>(NOU,CS,<span class="st">"/"</span> ,<span class="st">"C9orf72/results_adj_h.csv"</span>))){</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>    nou <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(NOU,CS,<span class="st">"/"</span> ,<span class="st">"C9orf72/results_adj_h.csv"</span>))</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>    nou <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract DEG</span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># For U_ftld</span></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(ftld_tdp<span class="sc">$</span>X[ftld_tdp<span class="sc">$</span>X.<span class="fl">1.</span>Healthy.<span class="fl">1.</span>C9orf72 <span class="sc">==</span> <span class="dv">1</span>])) {</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>    U_ftld <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="dv">0</span>)  <span class="co"># Assign an empty vector if NULL</span></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>    U_ftld <span class="ot">&lt;-</span> ftld_tdp<span class="sc">$</span>X[ftld_tdp<span class="sc">$</span>X.<span class="fl">1.</span>Healthy.<span class="fl">1.</span>C9orf72 <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># For U_nou</span></span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(nou<span class="sc">$</span>X[nou<span class="sc">$</span>X.<span class="fl">1.</span>Healthy.<span class="fl">1.</span>C9orf72 <span class="sc">==</span> <span class="dv">1</span>])) {</span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>    U_nou <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="dv">0</span>)  <span class="co"># Assign an empty vector if NULL</span></span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a>    U_nou <span class="ot">&lt;-</span> nou<span class="sc">$</span>X[nou<span class="sc">$</span>X.<span class="fl">1.</span>Healthy.<span class="fl">1.</span>C9orf72 <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># For D_ftld</span></span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(ftld_tdp<span class="sc">$</span>X[ftld_tdp<span class="sc">$</span>X.<span class="fl">1.</span>Healthy.<span class="fl">1.</span>C9orf72 <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>])) {</span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>    D_ftld <span class="ot">&lt;-</span><span class="fu">character</span>(<span class="dv">0</span>)  <span class="co"># Assign an empty vector if NULL</span></span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a>    D_ftld <span class="ot">&lt;-</span> ftld_tdp<span class="sc">$</span>X[ftld_tdp<span class="sc">$</span>X.<span class="fl">1.</span>Healthy.<span class="fl">1.</span>C9orf72 <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a>  <span class="co"># For D_nou</span></span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(nou<span class="sc">$</span>X[nou<span class="sc">$</span>X.<span class="fl">1.</span>Healthy.<span class="fl">1.</span>c9orf72 <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>])) {</span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>    D_nou <span class="ot">&lt;-</span><span class="fu">character</span>(<span class="dv">0</span>) <span class="co"># Assign an empty vector if NULL</span></span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a>    D_nou <span class="ot">&lt;-</span> nou<span class="sc">$</span>X[nou<span class="sc">$</span>X.<span class="fl">1.</span>Healthy.<span class="fl">1.</span>C9orf72 <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a>  <span class="co"># OVERLAP GENES</span></span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a>  C_U_ftld_nou <span class="ot">&lt;-</span> <span class="fu">intersect</span>(U_ftld, U_nou)</span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a>  C_D_ftld_nou <span class="ot">&lt;-</span> <span class="fu">intersect</span>(D_ftld, D_nou)</span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a>  <span class="co"># SIMILARITY INDEX</span></span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a>  Ujaccard13 <span class="ot">&lt;-</span> <span class="fu">jaccard_similarity</span>(U_ftld, U_nou)</span>
<span id="cb6-69"><a href="#cb6-69" aria-hidden="true" tabindex="-1"></a>  Djaccard13 <span class="ot">&lt;-</span> <span class="fu">jaccard_similarity</span>(D_ftld, D_nou)</span>
<span id="cb6-70"><a href="#cb6-70" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-71"><a href="#cb6-71" aria-hidden="true" tabindex="-1"></a>  <span class="co"># HIPERGEOMETRIC TESTS</span></span>
<span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a>  total_genes_FN <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">c</span>(ftld_tdp<span class="sc">$</span>X, nou<span class="sc">$</span>X)))</span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Gene Overlap</span></span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a>  GU_13 <span class="ot">=</span> <span class="fu">testGeneOverlap</span>(<span class="fu">newGeneOverlap</span>(U_ftld, U_nou, total_genes_FN))<span class="sc">@</span>pval</span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a>  GD_13 <span class="ot">=</span> <span class="fu">testGeneOverlap</span>(<span class="fu">newGeneOverlap</span>(D_ftld, D_nou, total_genes_FN))<span class="sc">@</span>pval</span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-78"><a href="#cb6-78" aria-hidden="true" tabindex="-1"></a>  <span class="co"># UPDATE RESULTS TABLE WITH THE CALCULATED VALUES.</span></span>
<span id="cb6-79"><a href="#cb6-79" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"Upregulated FTLD_TDP"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(U_ftld)</span>
<span id="cb6-80"><a href="#cb6-80" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"Upregulated Rimod"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(U_nou)</span>
<span id="cb6-81"><a href="#cb6-81" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"Downregulated FTLD_TDP"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(D_ftld)</span>
<span id="cb6-82"><a href="#cb6-82" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"Downregulated Rimod"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(D_nou)</span>
<span id="cb6-83"><a href="#cb6-83" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"CU FTLD – Rimod"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(C_U_ftld_nou)</span>
<span id="cb6-84"><a href="#cb6-84" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"CD FTLD – Rimod"</span>, CS] <span class="ot">&lt;-</span> <span class="fu">length</span>(C_D_ftld_nou)</span>
<span id="cb6-85"><a href="#cb6-85" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"JU FTLD – Rimod"</span>, CS] <span class="ot">&lt;-</span> Ujaccard13</span>
<span id="cb6-86"><a href="#cb6-86" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"JD FTLD – Rimod"</span>, CS] <span class="ot">&lt;-</span> Djaccard13</span>
<span id="cb6-87"><a href="#cb6-87" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"GU FTLD – Rimod"</span>, CS] <span class="ot">&lt;-</span> GU_13</span>
<span id="cb6-88"><a href="#cb6-88" aria-hidden="true" tabindex="-1"></a>  Results[<span class="st">"GD FTLD – Rimod"</span>, CS] <span class="ot">&lt;-</span> GD_13</span>
<span id="cb6-89"><a href="#cb6-89" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-90"><a href="#cb6-90" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-91"><a href="#cb6-91" aria-hidden="true" tabindex="-1"></a>Results <span class="ot">&lt;-</span> <span class="fu">t</span>(Results)</span>
<span id="cb6-92"><a href="#cb6-92" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(Results)</span>
<span id="cb6-93"><a href="#cb6-93" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(Results[<span class="fu">rowSums</span>(Results[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]) <span class="sc">&gt;</span> <span class="dv">0</span>, ])</span>
<span id="cb6-94"><a href="#cb6-94" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(Results, <span class="at">file =</span> <span class="fu">paste0</span>(RESULTS_PATH, <span class="st">"/C9_Validation_results_cs.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="analytical-goals-of-the-single-cellresolved-differential-expression-analysis" class="level2">
<h2 class="anchored" data-anchor-id="analytical-goals-of-the-single-cellresolved-differential-expression-analysis"><strong>Analytical Goals of the Single-Cell–Resolved Differential Expression Analysis</strong></h2>
<p>The single-cell differential expression module was designed to:</p>
<ol type="1">
<li><p><strong>Identify the precise cell states driving disease-specific transcriptional alterations</strong> in FTLD-C9 and FTLD-TDP.</p></li>
<li><p><strong>Distinguish neuronal and glial subtypes with high DEG burden</strong>, pointing to selective cellular vulnerability.</p></li>
<li><p><strong>Characterize lineage-specific pathways</strong>, such as synaptic dysfunction in RORB neurons or inflammatory activation in GFAP+ astrocytes.</p></li>
<li><p><strong>Link transcriptional changes to neuropathological markers</strong>, including STMN2 loss, pTDP-43 burden, and DPR/RNA foci density.</p></li>
<li><p><strong>Enable functional enrichment (GO/Metascape)</strong> for each cell state, revealing discrete biological processes altered in disease.</p></li>
<li><p><strong>Provide a unified cellular framework</strong> for comparing FTLD-C9 and FTLD-TDP, facilitating identification of shared and divergent mechanisms.</p></li>
<li><p><strong>Support robust cross-validation</strong> with independent datasets, increasing confidence in the generalizability of cell-state transcriptional signatures.</p></li>
</ol>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"edgeR"</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"GO.db"</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co"># c9</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>CSV_DECONVOLDED_PATH <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/BAYESPRISM/FTLD/C9/CELL STATE ORIGINAL"</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>CASE_LEGEND <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BayesPrism/FTD/DATASETS/FTD_QUIM/decoder_DeSeq2_FTD_FINAL.xlsx"</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>SV_COVARIATE <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/FTLD_BULK/SURROGATED_VARIABLES/SV1_C9_HC_QUIM.csv"</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>OUTPUT_DIRECTORY <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/EDGER/FTLD/C9/CS_SV"</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>REFERENCE_GROUP <span class="ot">&lt;-</span> <span class="st">"Healthy"</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>covariate_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(SV_COVARIATE)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>covariate_data<span class="sc">$</span>sample.ID <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^long"</span>, <span class="st">""</span>, covariate_data<span class="sc">$</span>sample.ID)</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>csv_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(CSV_DECONVOLDED_PATH, <span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.csv$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">endsWith</span>(CASE_LEGEND, <span class="st">".csv"</span>)) {</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  case_legend <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(CASE_LEGEND)</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(case_legend<span class="sc">$</span>group.ID)</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> <span class="cf">if</span> (<span class="fu">endsWith</span>(CASE_LEGEND, <span class="st">".xlsx"</span>)) {</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>  case_legend <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(CASE_LEGEND)</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(case_legend<span class="sc">$</span>group.ID)</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group2_ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(case_legend<span class="sc">$</span>group2.ID)</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group2.ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>  case_legend<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">"File type not recognized."</span>)</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group2_ID <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(case_legend<span class="sc">$</span>group2_ID)</span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">as.character</span>(case_legend<span class="sc">$</span>group_ID)</span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(case_legend<span class="sc">$</span>group_ID)</span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(case_legend<span class="sc">$</span>group_ID)</span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>conditions <span class="ot">&lt;-</span> <span class="fu">levels</span>(case_legend<span class="sc">$</span>group_ID)</span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (REFERENCE_GROUP <span class="sc">%in%</span> conditions) {</span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>  conditions <span class="ot">&lt;-</span> conditions[conditions <span class="sc">!=</span> REFERENCE_GROUP]</span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(OUTPUT_DIRECTORY)) {</span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(OUTPUT_DIRECTORY, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>matrix_data <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(conditions)<span class="sc">*</span><span class="fu">length</span>(csv_files), <span class="at">nrow =</span> <span class="fu">length</span>(conditions), <span class="at">ncol =</span> <span class="fu">length</span>(csv_files))</span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a DataFrame with the matrix data, specifying column and row names</span></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(matrix_data, <span class="at">row.names =</span> conditions)</span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">lapply</span>(csv_files, <span class="cf">function</span>(file) tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file)))</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (file <span class="cf">in</span> csv_files){</span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file))</span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(filename)</span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(file, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^long"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(data))</span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"X"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(data))</span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a>  celltype_folder <span class="ot">&lt;-</span> <span class="fu">file.path</span>(OUTPUT_DIRECTORY, filename)</span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(celltype_folder)) {</span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dir.create</span>(celltype_folder, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a>  conditions <span class="ot">&lt;-</span> <span class="st">"C9orf72"</span></span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (condition <span class="cf">in</span> conditions){</span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(condition)</span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a>    condition_folder <span class="ot">&lt;-</span> <span class="fu">file.path</span>(celltype_folder, condition)</span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(condition_folder)) {</span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dir.create</span>(condition_folder, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tryCatch</span>({</span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> case_legend[case_legend<span class="sc">$</span>group_ID <span class="sc">%in%</span> <span class="fu">c</span>(REFERENCE_GROUP, condition), ]</span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID)</span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>SV1 <span class="ot">&lt;-</span> covariate_data<span class="sc">$</span>SV1[<span class="fu">match</span>(comparison_data<span class="sc">$</span>sample.ID, covariate_data<span class="sc">$</span>sample.ID)]</span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">levels =</span> <span class="fu">c</span>(REFERENCE_GROUP, condition))</span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">order</span>(comparison_data<span class="sc">$</span>group_ID), ]</span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a>      common_ids <span class="ot">&lt;-</span> <span class="fu">intersect</span>(comparison_data<span class="sc">$</span>sample.ID, <span class="fu">rownames</span>(data))</span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a>      data_subset <span class="ot">&lt;-</span> data[<span class="fu">rownames</span>(data) <span class="sc">%in%</span> common_ids, ]</span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a>      data_ordered <span class="ot">&lt;-</span> data_subset[<span class="fu">match</span>(comparison_data<span class="sc">$</span>sample.ID, <span class="fu">rownames</span>(data_subset)), ]</span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a>      <span class="co"># CLEAN NA</span></span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a>      data_ordered[<span class="sc">!</span><span class="fu">is.finite</span>(<span class="fu">as.matrix</span>(data_ordered))] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a>      data_ordered <span class="ot">&lt;-</span> data_ordered[<span class="fu">complete.cases</span>(data_ordered), ]</span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">match</span>(<span class="fu">rownames</span>(data_ordered), comparison_data<span class="sc">$</span>sample.ID), ]</span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a>      group <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID[comparison_data<span class="sc">$</span>group_ID <span class="sc">%in%</span> <span class="fu">c</span>(REFERENCE_GROUP, condition)])</span>
<span id="cb7-95"><a href="#cb7-95" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts=</span><span class="fu">t</span>(data_ordered),<span class="at">group=</span>group)</span>
<span id="cb7-96"><a href="#cb7-96" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> y[, <span class="fu">colSums</span>(y<span class="sc">$</span>counts) <span class="sc">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb7-97"><a href="#cb7-97" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Keep only samples in comparison_data that are in y</span></span>
<span id="cb7-98"><a href="#cb7-98" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">match</span>(<span class="fu">colnames</span>(y), comparison_data<span class="sc">$</span>sample.ID), ]</span>
<span id="cb7-99"><a href="#cb7-99" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-100"><a href="#cb7-100" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Average Log CPM histogram</span></span>
<span id="cb7-101"><a href="#cb7-101" aria-hidden="true" tabindex="-1"></a>      AveLogCPM <span class="ot">&lt;-</span> <span class="fu">aveLogCPM</span>(y)</span>
<span id="cb7-102"><a href="#cb7-102" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">normLibSizes</span>(y)</span>
<span id="cb7-103"><a href="#cb7-103" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-104"><a href="#cb7-104" aria-hidden="true" tabindex="-1"></a>      <span class="co"># MD plot</span></span>
<span id="cb7-105"><a href="#cb7-105" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"MD_plot_h.png"</span>)</span>
<span id="cb7-106"><a href="#cb7-106" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb7-107"><a href="#cb7-107" aria-hidden="true" tabindex="-1"></a>      MD_plot <span class="ot">&lt;-</span> <span class="fu">plotMD</span>(y, <span class="at">column=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb7-108"><a href="#cb7-108" aria-hidden="true" tabindex="-1"></a>        <span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>, <span class="at">col=</span><span class="st">"red"</span>, <span class="at">lty=</span><span class="dv">2</span>, <span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb7-109"><a href="#cb7-109" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb7-110"><a href="#cb7-110" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-111"><a href="#cb7-111" aria-hidden="true" tabindex="-1"></a>      <span class="co"># DESIGN MATRIX</span></span>
<span id="cb7-112"><a href="#cb7-112" aria-hidden="true" tabindex="-1"></a>      group <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">levels =</span> <span class="fu">c</span>(REFERENCE_GROUP, condition))</span>
<span id="cb7-113"><a href="#cb7-113" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">relevel</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">ref =</span> REFERENCE_GROUP)</span>
<span id="cb7-114"><a href="#cb7-114" aria-hidden="true" tabindex="-1"></a>      design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> comparison_data<span class="sc">$</span>group_ID <span class="sc">+</span> comparison_data<span class="sc">$</span>SV1, <span class="at">data =</span> comparison_data)</span>
<span id="cb7-115"><a href="#cb7-115" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">c</span>(REFERENCE_GROUP, condition, <span class="st">"SV1"</span>)</span>
<span id="cb7-116"><a href="#cb7-116" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(design) <span class="sc">==</span> <span class="fu">ncol</span>(y)</span>
<span id="cb7-117"><a href="#cb7-117" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-118"><a href="#cb7-118" aria-hidden="true" tabindex="-1"></a>      <span class="co"># DISPERSION ESTIMATION</span></span>
<span id="cb7-119"><a href="#cb7-119" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(y,design)</span>
<span id="cb7-120"><a href="#cb7-120" aria-hidden="true" tabindex="-1"></a>      fit <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(y,design)</span>
<span id="cb7-121"><a href="#cb7-121" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-122"><a href="#cb7-122" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Diferential expression analysis</span></span>
<span id="cb7-123"><a href="#cb7-123" aria-hidden="true" tabindex="-1"></a>      B.LvsP <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="fu">paste0</span>(condition, <span class="st">" - "</span>, REFERENCE_GROUP), <span class="at">levels=</span>design)</span>
<span id="cb7-124"><a href="#cb7-124" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fit, <span class="at">contrast=</span>B.LvsP)</span>
<span id="cb7-125"><a href="#cb7-125" aria-hidden="true" tabindex="-1"></a>      is.de <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(res, <span class="at">adjust.method =</span> <span class="st">"fdr"</span>, <span class="at">p.value =</span> <span class="fl">0.05</span>, <span class="at">lfc=</span> <span class="dv">0</span>)</span>
<span id="cb7-126"><a href="#cb7-126" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> res<span class="sc">$</span>table</span>
<span id="cb7-127"><a href="#cb7-127" aria-hidden="true" tabindex="-1"></a>      results_table<span class="sc">$</span>adj_pval <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(results_table<span class="sc">$</span>PValue, <span class="at">method =</span> <span class="st">"fdr"</span>, <span class="at">n =</span> <span class="fu">length</span>(results_table<span class="sc">$</span>PValue))</span>
<span id="cb7-128"><a href="#cb7-128" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> <span class="fu">cbind</span>(results_table, <span class="fu">data.frame</span>(is.de))</span>
<span id="cb7-129"><a href="#cb7-129" aria-hidden="true" tabindex="-1"></a>      <span class="co"># SAVING CSV RESULTS</span></span>
<span id="cb7-130"><a href="#cb7-130" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write.csv</span>(res<span class="sc">$</span>table, <span class="at">file =</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"res_h.csv"</span>))</span>
<span id="cb7-131"><a href="#cb7-131" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write.csv</span>(results_table, <span class="at">file =</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"results_adj_h.csv"</span>))</span>
<span id="cb7-132"><a href="#cb7-132" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-133"><a href="#cb7-133" aria-hidden="true" tabindex="-1"></a>      <span class="do">## Visualizations</span></span>
<span id="cb7-134"><a href="#cb7-134" aria-hidden="true" tabindex="-1"></a>      <span class="co"># "Volcano like" plot</span></span>
<span id="cb7-135"><a href="#cb7-135" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"MD_res_plot_h.png"</span>)</span>
<span id="cb7-136"><a href="#cb7-136" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb7-137"><a href="#cb7-137" aria-hidden="true" tabindex="-1"></a>      <span class="fu">plotMD</span>(res, <span class="at">status=</span>is.de)</span>
<span id="cb7-138"><a href="#cb7-138" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb7-139"><a href="#cb7-139" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-140"><a href="#cb7-140" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Heatmap clustering </span></span>
<span id="cb7-141"><a href="#cb7-141" aria-hidden="true" tabindex="-1"></a>      logCPM <span class="ot">&lt;-</span> <span class="fu">cpm</span>(y, <span class="at">prior.count=</span><span class="dv">2</span>, <span class="at">log=</span><span class="cn">TRUE</span>)</span>
<span id="cb7-142"><a href="#cb7-142" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames</span>(logCPM) <span class="ot">&lt;-</span> y<span class="sc">$</span>genes<span class="sc">$</span>Symbol</span>
<span id="cb7-143"><a href="#cb7-143" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(logCPM) <span class="ot">&lt;-</span> <span class="fu">paste</span>(y<span class="sc">$</span>samples<span class="sc">$</span>group, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"-"</span>)</span>
<span id="cb7-144"><a href="#cb7-144" aria-hidden="true" tabindex="-1"></a>      tr <span class="ot">&lt;-</span> <span class="fu">glmTreat</span>(fit, <span class="at">contrast=</span>B.LvsP, <span class="at">lfc=</span><span class="fu">log2</span>(<span class="fl">1.5</span>))</span>
<span id="cb7-145"><a href="#cb7-145" aria-hidden="true" tabindex="-1"></a>      o <span class="ot">&lt;-</span> <span class="fu">order</span>(tr<span class="sc">$</span>table<span class="sc">$</span>PValue)</span>
<span id="cb7-146"><a href="#cb7-146" aria-hidden="true" tabindex="-1"></a>      logCPM <span class="ot">&lt;-</span> logCPM[o[<span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>],]</span>
<span id="cb7-147"><a href="#cb7-147" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-148"><a href="#cb7-148" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"Heatmap_h.png"</span>)</span>
<span id="cb7-149"><a href="#cb7-149" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb7-150"><a href="#cb7-150" aria-hidden="true" tabindex="-1"></a>      <span class="fu">coolmap</span>(logCPM, <span class="at">margins=</span><span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">7</span>), <span class="at">lhei=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span>), <span class="at">lwid=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>))</span>
<span id="cb7-151"><a href="#cb7-151" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb7-152"><a href="#cb7-152" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-153"><a href="#cb7-153" aria-hidden="true" tabindex="-1"></a>      df[condition, tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file))] <span class="ot">&lt;-</span> <span class="fu">nrow</span>(results_table[results_table<span class="sc">$</span>adj_pval<span class="sc">&lt;</span><span class="fl">0.05</span>,])</span>
<span id="cb7-154"><a href="#cb7-154" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-155"><a href="#cb7-155" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e){</span>
<span id="cb7-156"><a href="#cb7-156" aria-hidden="true" tabindex="-1"></a>      df[condition, tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file))] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb7-157"><a href="#cb7-157" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb7-158"><a href="#cb7-158" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-159"><a href="#cb7-159" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-160"><a href="#cb7-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-161"><a href="#cb7-161" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df, <span class="at">file =</span> <span class="fu">file.path</span>(OUTPUT_DIRECTORY, <span class="st">"summary_h.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR, <span class="at">lib.loc =</span> <span class="st">"/home/jaumatell/R/x86_64-pc-linux-gnu-library/4.4/edgeR_4"</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"GO.db"</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"xlsx"</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################################</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Paths and parameters</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>CSV_DECONVOLDED_PATH <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/BAYESPRISM/FTLD/TDP/CELL STATE/"</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>CASE_LEGEND <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BayesPrism/FTD/DATASETS/FTD_QUIM/decoder_DeSeq2_FTD_FINAL.xlsx"</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>OUTPUT_DIRECTORY <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/EDGER/FTLD/TDP/DEG"</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>REFERENCE_GROUP <span class="ot">&lt;-</span> <span class="st">"Healthy"</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################################</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>csv_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(CSV_DECONVOLDED_PATH, <span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.csv$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Read case legend and prepare group IDs</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>case_legend <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(CASE_LEGEND,<span class="at">sheetIndex =</span> <span class="dv">1</span>)</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(case_legend<span class="sc">$</span>group.ID)</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure levels are consistent and drop unused levels</span></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>conditions <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">as.factor</span>(case_legend<span class="sc">$</span>group_ID))</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>conditions <span class="ot">&lt;-</span> <span class="st">"TDP"</span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Create output directory if it doesn't exist</span></span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(OUTPUT_DIRECTORY)) {</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(OUTPUT_DIRECTORY, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize summary matrix</span></span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>matrix_data <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(conditions)<span class="sc">*</span><span class="fu">length</span>(csv_files), <span class="at">nrow =</span> <span class="fu">length</span>(conditions), <span class="at">ncol =</span> <span class="fu">length</span>(csv_files))</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(matrix_data, <span class="at">row.names =</span> conditions)</span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">sapply</span>(csv_files, <span class="cf">function</span>(file) tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file)))</span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Process each CSV file</span></span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (file <span class="cf">in</span> csv_files) {</span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file))</span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(filename)</span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(file, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span><span class="fu">gsub</span>(<span class="st">"X"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(data))</span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a>  celltype_folder <span class="ot">&lt;-</span> <span class="fu">file.path</span>(OUTPUT_DIRECTORY, filename)</span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(celltype_folder)) {</span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dir.create</span>(celltype_folder, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (condition <span class="cf">in</span> conditions) {</span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a>    condition_folder <span class="ot">&lt;-</span> <span class="fu">file.path</span>(celltype_folder, condition)</span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(condition_folder)) {</span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dir.create</span>(condition_folder, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tryCatch</span>({</span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> case_legend[case_legend<span class="sc">$</span>group_ID <span class="sc">%in%</span> <span class="fu">c</span>(REFERENCE_GROUP, condition), ]</span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">levels =</span> <span class="fu">c</span>(REFERENCE_GROUP, condition)), <span class="at">ref =</span> condition)</span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">order</span>(comparison_data<span class="sc">$</span>group_ID), ]</span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a>      common_ids <span class="ot">&lt;-</span> <span class="fu">intersect</span>(comparison_data<span class="sc">$</span>sample.ID, <span class="fu">rownames</span>(data))</span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a>      data_subset <span class="ot">&lt;-</span> data[<span class="fu">rownames</span>(data) <span class="sc">%in%</span> common_ids, ]</span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a>      data_ordered <span class="ot">&lt;-</span> data_subset[<span class="fu">match</span>(comparison_data<span class="sc">$</span>sample.ID, <span class="fu">rownames</span>(data_subset)), ]</span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a>      data_ordered[<span class="fu">is.na</span>(data_ordered)] <span class="ot">&lt;-</span> <span class="fl">10e-8</span></span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a>      group <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID)</span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> <span class="fu">t</span>(data_ordered), <span class="at">group =</span> group)</span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Plot average log CPM</span></span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a>      AveLogCPM <span class="ot">&lt;-</span> <span class="fu">aveLogCPM</span>(y)</span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">normLibSizes</span>(y)</span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"MD_plot_h.png"</span>)</span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a>      <span class="fu">plotMD</span>(y, <span class="at">column =</span> <span class="dv">1</span>)</span>
<span id="cb8-75"><a href="#cb8-75" aria-hidden="true" tabindex="-1"></a>      <span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"red"</span>, <span class="at">lty =</span> <span class="dv">2</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb8-76"><a href="#cb8-76" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb8-77"><a href="#cb8-77" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-78"><a href="#cb8-78" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Create design matrix</span></span>
<span id="cb8-79"><a href="#cb8-79" aria-hidden="true" tabindex="-1"></a>      design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> group)</span>
<span id="cb8-80"><a href="#cb8-80" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">levels</span>(group)</span>
<span id="cb8-81"><a href="#cb8-81" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-82"><a href="#cb8-82" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Estimate dispersion</span></span>
<span id="cb8-83"><a href="#cb8-83" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(y, design)</span>
<span id="cb8-84"><a href="#cb8-84" aria-hidden="true" tabindex="-1"></a>      fit <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(y, design)</span>
<span id="cb8-85"><a href="#cb8-85" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-86"><a href="#cb8-86" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Differential expression analysis</span></span>
<span id="cb8-87"><a href="#cb8-87" aria-hidden="true" tabindex="-1"></a>      contrast <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts =</span> <span class="fu">paste0</span>(<span class="fu">make.names</span>(condition), <span class="st">" - "</span>, <span class="fu">make.names</span>(REFERENCE_GROUP)), <span class="at">levels =</span> design)</span>
<span id="cb8-88"><a href="#cb8-88" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fit, <span class="at">contrast =</span> contrast)</span>
<span id="cb8-89"><a href="#cb8-89" aria-hidden="true" tabindex="-1"></a>      is.de <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(res, <span class="at">adjust.method =</span> <span class="st">"fdr"</span>, <span class="at">p.value =</span> <span class="fl">0.05</span>, <span class="at">lfc =</span> <span class="dv">0</span>)</span>
<span id="cb8-90"><a href="#cb8-90" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> res<span class="sc">$</span>table</span>
<span id="cb8-91"><a href="#cb8-91" aria-hidden="true" tabindex="-1"></a>      results_table<span class="sc">$</span>adj_pval <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(results_table<span class="sc">$</span>PValue, <span class="at">method =</span> <span class="st">"fdr"</span>)</span>
<span id="cb8-92"><a href="#cb8-92" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> <span class="fu">cbind</span>(results_table, <span class="fu">data.frame</span>(is.de))</span>
<span id="cb8-93"><a href="#cb8-93" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-94"><a href="#cb8-94" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Save results</span></span>
<span id="cb8-95"><a href="#cb8-95" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write.csv</span>(res<span class="sc">$</span>table, <span class="at">file =</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"res_h.csv"</span>))</span>
<span id="cb8-96"><a href="#cb8-96" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write.csv</span>(results_table, <span class="at">file =</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"results_adj_h.csv"</span>))</span>
<span id="cb8-97"><a href="#cb8-97" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-98"><a href="#cb8-98" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Histogram</span></span>
<span id="cb8-99"><a href="#cb8-99" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"histogram_plot_h.png"</span>)</span>
<span id="cb8-100"><a href="#cb8-100" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb8-101"><a href="#cb8-101" aria-hidden="true" tabindex="-1"></a>      <span class="fu">hist</span>(AveLogCPM)</span>
<span id="cb8-102"><a href="#cb8-102" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb8-103"><a href="#cb8-103" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-104"><a href="#cb8-104" aria-hidden="true" tabindex="-1"></a>      <span class="co"># BCV plot</span></span>
<span id="cb8-105"><a href="#cb8-105" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"BCV_plot_h.png"</span>)</span>
<span id="cb8-106"><a href="#cb8-106" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb8-107"><a href="#cb8-107" aria-hidden="true" tabindex="-1"></a>      <span class="fu">plotBCV</span>(y)</span>
<span id="cb8-108"><a href="#cb8-108" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb8-109"><a href="#cb8-109" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-110"><a href="#cb8-110" aria-hidden="true" tabindex="-1"></a>      <span class="co"># "Volcano like" plot</span></span>
<span id="cb8-111"><a href="#cb8-111" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"MD_res_plot_h.png"</span>)</span>
<span id="cb8-112"><a href="#cb8-112" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb8-113"><a href="#cb8-113" aria-hidden="true" tabindex="-1"></a>      <span class="fu">plotMD</span>(res, <span class="at">status =</span> is.de)</span>
<span id="cb8-114"><a href="#cb8-114" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb8-115"><a href="#cb8-115" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-116"><a href="#cb8-116" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Heatmap clustering</span></span>
<span id="cb8-117"><a href="#cb8-117" aria-hidden="true" tabindex="-1"></a>      logCPM <span class="ot">&lt;-</span> <span class="fu">cpm</span>(y, <span class="at">prior.count =</span> <span class="dv">2</span>, <span class="at">log =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-118"><a href="#cb8-118" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(logCPM) <span class="ot">&lt;-</span> <span class="fu">paste</span>(y<span class="sc">$</span>samples<span class="sc">$</span>group, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">sep =</span> <span class="st">"-"</span>)</span>
<span id="cb8-119"><a href="#cb8-119" aria-hidden="true" tabindex="-1"></a>      tr <span class="ot">&lt;-</span> <span class="fu">glmTreat</span>(fit, <span class="at">contrast =</span> contrast, <span class="at">lfc =</span> <span class="fu">log2</span>(<span class="fl">1.5</span>))</span>
<span id="cb8-120"><a href="#cb8-120" aria-hidden="true" tabindex="-1"></a>      o <span class="ot">&lt;-</span> <span class="fu">order</span>(tr<span class="sc">$</span>table<span class="sc">$</span>PValue)</span>
<span id="cb8-121"><a href="#cb8-121" aria-hidden="true" tabindex="-1"></a>      logCPM <span class="ot">&lt;-</span> logCPM[o[<span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>], ]</span>
<span id="cb8-122"><a href="#cb8-122" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-123"><a href="#cb8-123" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"Heatmap_h.png"</span>)</span>
<span id="cb8-124"><a href="#cb8-124" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb8-125"><a href="#cb8-125" aria-hidden="true" tabindex="-1"></a>      <span class="fu">coolmap</span>(logCPM, <span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">7</span>, <span class="dv">7</span>), <span class="at">lhei =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">6</span>), <span class="at">lwid =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>))</span>
<span id="cb8-126"><a href="#cb8-126" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb8-127"><a href="#cb8-127" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-128"><a href="#cb8-128" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Update summary DataFrame</span></span>
<span id="cb8-129"><a href="#cb8-129" aria-hidden="true" tabindex="-1"></a>      df[condition, filename] <span class="ot">&lt;-</span> <span class="fu">nrow</span>(results_table[results_table<span class="sc">$</span>adj_pval <span class="sc">&lt;</span> <span class="fl">0.05</span>, ])</span>
<span id="cb8-130"><a href="#cb8-130" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb8-131"><a href="#cb8-131" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb8-132"><a href="#cb8-132" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(e)</span>
<span id="cb8-133"><a href="#cb8-133" aria-hidden="true" tabindex="-1"></a>      df[condition, filename] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb8-134"><a href="#cb8-134" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb8-135"><a href="#cb8-135" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb8-136"><a href="#cb8-136" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-137"><a href="#cb8-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-138"><a href="#cb8-138" aria-hidden="true" tabindex="-1"></a><span class="co"># Save summary DataFrame</span></span>
<span id="cb8-139"><a href="#cb8-139" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df, <span class="at">file =</span> <span class="fu">file.path</span>(OUTPUT_DIRECTORY, <span class="st">"summary_h.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"edgeR"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"GO.db"</span>)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"xlsx"</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################################</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Paths and parameters</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>CSV_DECONVOLDED_PATH <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/BAYESPRISM/RIMOD/CELL STATE ORIGINAL/"</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>CASE_LEGEND <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/RIMOD_BULK/DATA/rimod_ftd_dataset_table_v3.txt"</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>OUTPUT_DIRECTORY <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/EDGER/RIMOD/CS_SV"</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>SV_COVARIATE <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/RIMOD_BULK/METADATA/SVA.csv"</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>REFERENCE_GROUP <span class="ot">&lt;-</span> <span class="st">"Healthy"</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################################</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>covariate_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(SV_COVARIATE, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>covariate_data<span class="sc">$</span>sample.ID <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^long"</span>, <span class="st">""</span>, covariate_data<span class="sc">$</span>ID)</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>csv_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(CSV_DECONVOLDED_PATH, <span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.csv$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Read case legend and prepare group IDs</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>case_legend <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(CASE_LEGEND)</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>case_legend <span class="ot">&lt;-</span> case_legend[case_legend<span class="sc">$</span>DiseaseCode <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"control"</span>, <span class="st">"FTD-C9"</span>), ]</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Renombrar valores dentro de DiseaseCode</span></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>DiseaseCode[case_legend<span class="sc">$</span>DiseaseCode <span class="sc">==</span> <span class="st">"control"</span>] <span class="ot">&lt;-</span> <span class="st">"Healthy"</span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>DiseaseCode[case_legend<span class="sc">$</span>DiseaseCode <span class="sc">==</span> <span class="st">"FTD-C9"</span>] <span class="ot">&lt;-</span> <span class="st">"C9orf72"</span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(case_legend<span class="sc">$</span>DiseaseCode)</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group.ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure levels are consistent and drop unused levels</span></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>conditions <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">as.factor</span>(case_legend<span class="sc">$</span>group_ID))</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>conditions <span class="ot">&lt;-</span> <span class="st">"C9orf72"</span></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Create output directory if it doesn't exist</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(OUTPUT_DIRECTORY)) {</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(OUTPUT_DIRECTORY, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize summary matrix</span></span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>matrix_data <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(conditions)<span class="sc">*</span><span class="fu">length</span>(csv_files), <span class="at">nrow =</span> <span class="fu">length</span>(conditions), <span class="at">ncol =</span> <span class="fu">length</span>(csv_files))</span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(matrix_data, <span class="at">row.names =</span> conditions)</span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">sapply</span>(csv_files, <span class="cf">function</span>(file) tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file)))</span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Process each CSV file</span></span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (file <span class="cf">in</span> csv_files) {</span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file))</span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(filename)</span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(file, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>  celltype_folder <span class="ot">&lt;-</span> <span class="fu">file.path</span>(OUTPUT_DIRECTORY, filename)</span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(celltype_folder)) {</span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dir.create</span>(celltype_folder, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (condition <span class="cf">in</span> conditions) {</span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a>    condition_folder <span class="ot">&lt;-</span> <span class="fu">file.path</span>(celltype_folder, condition)</span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(condition_folder)) {</span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dir.create</span>(condition_folder, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-60"><a href="#cb9-60" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb9-61"><a href="#cb9-61" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-62"><a href="#cb9-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tryCatch</span>({</span>
<span id="cb9-63"><a href="#cb9-63" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb9-64"><a href="#cb9-64" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb9-65"><a href="#cb9-65" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> case_legend[case_legend<span class="sc">$</span>group_ID <span class="sc">%in%</span> <span class="fu">c</span>(REFERENCE_GROUP, condition), ]</span>
<span id="cb9-66"><a href="#cb9-66" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID)</span>
<span id="cb9-67"><a href="#cb9-67" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb9-68"><a href="#cb9-68" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>SV1 <span class="ot">&lt;-</span> covariate_data<span class="sc">$</span>SV1[<span class="fu">match</span>(comparison_data<span class="sc">$</span>RimodID, covariate_data<span class="sc">$</span>ID)]</span>
<span id="cb9-69"><a href="#cb9-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-70"><a href="#cb9-70" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">levels =</span> <span class="fu">c</span>(REFERENCE_GROUP, condition))</span>
<span id="cb9-71"><a href="#cb9-71" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">order</span>(comparison_data<span class="sc">$</span>group_ID), ]</span>
<span id="cb9-72"><a href="#cb9-72" aria-hidden="true" tabindex="-1"></a>      common_ids <span class="ot">&lt;-</span> <span class="fu">intersect</span>(comparison_data<span class="sc">$</span>RimodID, <span class="fu">rownames</span>(data))</span>
<span id="cb9-73"><a href="#cb9-73" aria-hidden="true" tabindex="-1"></a>      data_subset <span class="ot">&lt;-</span> data[<span class="fu">rownames</span>(data) <span class="sc">%in%</span> common_ids, ]</span>
<span id="cb9-74"><a href="#cb9-74" aria-hidden="true" tabindex="-1"></a>      data_ordered <span class="ot">&lt;-</span> data_subset[<span class="fu">match</span>(comparison_data<span class="sc">$</span>RimodID, <span class="fu">rownames</span>(data_subset)), ]</span>
<span id="cb9-75"><a href="#cb9-75" aria-hidden="true" tabindex="-1"></a>      <span class="co"># CLEAN NA</span></span>
<span id="cb9-76"><a href="#cb9-76" aria-hidden="true" tabindex="-1"></a>      data_ordered[<span class="sc">!</span><span class="fu">is.finite</span>(<span class="fu">as.matrix</span>(data_ordered))] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb9-77"><a href="#cb9-77" aria-hidden="true" tabindex="-1"></a>      data_ordered <span class="ot">&lt;-</span> data_ordered[<span class="fu">complete.cases</span>(data_ordered), ]</span>
<span id="cb9-78"><a href="#cb9-78" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">match</span>(<span class="fu">rownames</span>(data_ordered), comparison_data<span class="sc">$</span>RimodID), ]</span>
<span id="cb9-79"><a href="#cb9-79" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb9-80"><a href="#cb9-80" aria-hidden="true" tabindex="-1"></a>      group <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID[comparison_data<span class="sc">$</span>group_ID <span class="sc">%in%</span> <span class="fu">c</span>(REFERENCE_GROUP, condition)])</span>
<span id="cb9-81"><a href="#cb9-81" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts=</span><span class="fu">t</span>(data_ordered),<span class="at">group=</span>group)</span>
<span id="cb9-82"><a href="#cb9-82" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> y[, <span class="fu">colSums</span>(y<span class="sc">$</span>counts) <span class="sc">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb9-83"><a href="#cb9-83" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Keep only samples in comparison_data that are in y</span></span>
<span id="cb9-84"><a href="#cb9-84" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">match</span>(<span class="fu">colnames</span>(y), comparison_data<span class="sc">$</span>RimodID), ]</span>
<span id="cb9-85"><a href="#cb9-85" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb9-86"><a href="#cb9-86" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Average Log CPM histogram</span></span>
<span id="cb9-87"><a href="#cb9-87" aria-hidden="true" tabindex="-1"></a>      AveLogCPM <span class="ot">&lt;-</span> <span class="fu">aveLogCPM</span>(y)</span>
<span id="cb9-88"><a href="#cb9-88" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">normLibSizes</span>(y)</span>
<span id="cb9-89"><a href="#cb9-89" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb9-90"><a href="#cb9-90" aria-hidden="true" tabindex="-1"></a>      <span class="co"># MD plot</span></span>
<span id="cb9-91"><a href="#cb9-91" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"MD_plot_h.png"</span>)</span>
<span id="cb9-92"><a href="#cb9-92" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb9-93"><a href="#cb9-93" aria-hidden="true" tabindex="-1"></a>      MD_plot <span class="ot">&lt;-</span> <span class="fu">plotMD</span>(y, <span class="at">column=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb9-94"><a href="#cb9-94" aria-hidden="true" tabindex="-1"></a>        <span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>, <span class="at">col=</span><span class="st">"red"</span>, <span class="at">lty=</span><span class="dv">2</span>, <span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb9-95"><a href="#cb9-95" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb9-96"><a href="#cb9-96" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb9-97"><a href="#cb9-97" aria-hidden="true" tabindex="-1"></a>      <span class="co"># DESIGN MATRIX</span></span>
<span id="cb9-98"><a href="#cb9-98" aria-hidden="true" tabindex="-1"></a>      group <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">levels =</span> <span class="fu">c</span>(REFERENCE_GROUP, condition))</span>
<span id="cb9-99"><a href="#cb9-99" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">relevel</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">ref =</span> REFERENCE_GROUP)</span>
<span id="cb9-100"><a href="#cb9-100" aria-hidden="true" tabindex="-1"></a>      design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> comparison_data<span class="sc">$</span>group_ID <span class="sc">+</span> comparison_data<span class="sc">$</span>SV1, <span class="at">data =</span> comparison_data)</span>
<span id="cb9-101"><a href="#cb9-101" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">c</span>(REFERENCE_GROUP, condition, <span class="st">"SV1"</span>)</span>
<span id="cb9-102"><a href="#cb9-102" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(design) <span class="sc">==</span> <span class="fu">ncol</span>(y)</span>
<span id="cb9-103"><a href="#cb9-103" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb9-104"><a href="#cb9-104" aria-hidden="true" tabindex="-1"></a>      <span class="co"># DISPERSION ESTIMATION</span></span>
<span id="cb9-105"><a href="#cb9-105" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(y,design)</span>
<span id="cb9-106"><a href="#cb9-106" aria-hidden="true" tabindex="-1"></a>      fit <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(y,design)</span>
<span id="cb9-107"><a href="#cb9-107" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb9-108"><a href="#cb9-108" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Diferential expression analysis</span></span>
<span id="cb9-109"><a href="#cb9-109" aria-hidden="true" tabindex="-1"></a>      B.LvsP <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="fu">paste0</span>(condition, <span class="st">" - "</span>, REFERENCE_GROUP), <span class="at">levels=</span>design)</span>
<span id="cb9-110"><a href="#cb9-110" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fit, <span class="at">contrast=</span>B.LvsP)</span>
<span id="cb9-111"><a href="#cb9-111" aria-hidden="true" tabindex="-1"></a>      is.de <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(res, <span class="at">adjust.method =</span> <span class="st">"fdr"</span>, <span class="at">p.value =</span> <span class="fl">0.05</span>, <span class="at">lfc=</span> <span class="dv">0</span>)</span>
<span id="cb9-112"><a href="#cb9-112" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> res<span class="sc">$</span>table</span>
<span id="cb9-113"><a href="#cb9-113" aria-hidden="true" tabindex="-1"></a>      results_table<span class="sc">$</span>adj_pval <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(results_table<span class="sc">$</span>PValue, <span class="at">method =</span> <span class="st">"fdr"</span>, <span class="at">n =</span> <span class="fu">length</span>(results_table<span class="sc">$</span>PValue))</span>
<span id="cb9-114"><a href="#cb9-114" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> <span class="fu">cbind</span>(results_table, <span class="fu">data.frame</span>(is.de))</span>
<span id="cb9-115"><a href="#cb9-115" aria-hidden="true" tabindex="-1"></a>      <span class="co"># SAVING CSV RESULTS</span></span>
<span id="cb9-116"><a href="#cb9-116" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write.csv</span>(res<span class="sc">$</span>table, <span class="at">file =</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"res_h.csv"</span>))</span>
<span id="cb9-117"><a href="#cb9-117" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write.csv</span>(results_table, <span class="at">file =</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"results_adj_h.csv"</span>))</span>
<span id="cb9-118"><a href="#cb9-118" aria-hidden="true" tabindex="-1"></a>      <span class="do">## Visualizations</span></span>
<span id="cb9-119"><a href="#cb9-119" aria-hidden="true" tabindex="-1"></a>      <span class="co"># "Volcano like" plot</span></span>
<span id="cb9-120"><a href="#cb9-120" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"MD_res_plot_h.png"</span>)</span>
<span id="cb9-121"><a href="#cb9-121" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb9-122"><a href="#cb9-122" aria-hidden="true" tabindex="-1"></a>      <span class="fu">plotMD</span>(res, <span class="at">status=</span>is.de)</span>
<span id="cb9-123"><a href="#cb9-123" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb9-124"><a href="#cb9-124" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb9-125"><a href="#cb9-125" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Heatmap clustering </span></span>
<span id="cb9-126"><a href="#cb9-126" aria-hidden="true" tabindex="-1"></a>      logCPM <span class="ot">&lt;-</span> <span class="fu">cpm</span>(y, <span class="at">prior.count=</span><span class="dv">2</span>, <span class="at">log=</span><span class="cn">TRUE</span>)</span>
<span id="cb9-127"><a href="#cb9-127" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames</span>(logCPM) <span class="ot">&lt;-</span> y<span class="sc">$</span>genes<span class="sc">$</span>Symbol</span>
<span id="cb9-128"><a href="#cb9-128" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(logCPM) <span class="ot">&lt;-</span> <span class="fu">paste</span>(y<span class="sc">$</span>samples<span class="sc">$</span>group, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"-"</span>)</span>
<span id="cb9-129"><a href="#cb9-129" aria-hidden="true" tabindex="-1"></a>      tr <span class="ot">&lt;-</span> <span class="fu">glmTreat</span>(fit, <span class="at">contrast=</span>B.LvsP, <span class="at">lfc=</span><span class="fu">log2</span>(<span class="fl">1.5</span>))</span>
<span id="cb9-130"><a href="#cb9-130" aria-hidden="true" tabindex="-1"></a>      o <span class="ot">&lt;-</span> <span class="fu">order</span>(tr<span class="sc">$</span>table<span class="sc">$</span>PValue)</span>
<span id="cb9-131"><a href="#cb9-131" aria-hidden="true" tabindex="-1"></a>      logCPM <span class="ot">&lt;-</span> logCPM[o[<span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>],]</span>
<span id="cb9-132"><a href="#cb9-132" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb9-133"><a href="#cb9-133" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"Heatmap_h.png"</span>)</span>
<span id="cb9-134"><a href="#cb9-134" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb9-135"><a href="#cb9-135" aria-hidden="true" tabindex="-1"></a>      <span class="fu">coolmap</span>(logCPM, <span class="at">margins=</span><span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">7</span>), <span class="at">lhei=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span>), <span class="at">lwid=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>))</span>
<span id="cb9-136"><a href="#cb9-136" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb9-137"><a href="#cb9-137" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb9-138"><a href="#cb9-138" aria-hidden="true" tabindex="-1"></a>      df[condition, tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file))] <span class="ot">&lt;-</span> <span class="fu">nrow</span>(results_table[results_table<span class="sc">$</span>adj_pval<span class="sc">&lt;</span><span class="fl">0.05</span>,])</span>
<span id="cb9-139"><a href="#cb9-139" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb9-140"><a href="#cb9-140" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e){</span>
<span id="cb9-141"><a href="#cb9-141" aria-hidden="true" tabindex="-1"></a>      df[condition, tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file))] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb9-142"><a href="#cb9-142" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb9-143"><a href="#cb9-143" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb9-144"><a href="#cb9-144" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-145"><a href="#cb9-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-146"><a href="#cb9-146" aria-hidden="true" tabindex="-1"></a><span class="co"># Save summary DataFrame</span></span>
<span id="cb9-147"><a href="#cb9-147" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df, <span class="at">file =</span> <span class="fu">file.path</span>(OUTPUT_DIRECTORY, <span class="st">"summary_h.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"edgeR"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"GO.db"</span>)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># FTD</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co">#CT</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>CSV_DECONVOLDED_PATH <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/BAYESPRISM/NEW/CELL STATE ORIGINAL"</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>CASE_LEGEND <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/NEW_BULK/METADATA/Sample_info.txt"</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>SV_COVARIATE <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/NEW_BULK/METADATA/SVA_correcte.csv"</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>OUTPUT_DIRECTORY <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/EDGER/NEW/CS_SVcorrect"</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>REFERENCE_GROUP <span class="ot">&lt;-</span> <span class="st">"Control"</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>covariate_data <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(SV_COVARIATE)</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>covariate_data<span class="sc">$</span>sample.ID <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^long"</span>, <span class="st">""</span>, covariate_data<span class="sc">$</span>ID)</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>csv_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(CSV_DECONVOLDED_PATH, <span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.csv$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>case_legend <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(CASE_LEGEND)</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>case_legend <span class="ot">&lt;-</span> case_legend[case_legend<span class="sc">$</span>GROUP <span class="sc">!=</span> <span class="st">"FTLD-TDP-C"</span>,]</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>GROUP[case_legend<span class="sc">$</span>GROUP <span class="sc">!=</span> <span class="st">"Control"</span>] <span class="ot">&lt;-</span> <span class="st">"TDP"</span></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>case_legend<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(case_legend<span class="sc">$</span>GROUP)</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>conditions <span class="ot">&lt;-</span> <span class="fu">levels</span>(case_legend<span class="sc">$</span>group_ID)</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (REFERENCE_GROUP <span class="sc">%in%</span> conditions) {</span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>  conditions <span class="ot">&lt;-</span> conditions[conditions <span class="sc">!=</span> REFERENCE_GROUP]</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(OUTPUT_DIRECTORY)) {</span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(OUTPUT_DIRECTORY, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a>matrix_data <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(conditions)<span class="sc">*</span><span class="fu">length</span>(csv_files), <span class="at">nrow =</span> <span class="fu">length</span>(conditions), <span class="at">ncol =</span> <span class="fu">length</span>(csv_files))</span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a DataFrame with the matrix data, specifying column and row names</span></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(matrix_data, <span class="at">row.names =</span> conditions)</span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">lapply</span>(csv_files, <span class="cf">function</span>(file) tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file)))</span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (file <span class="cf">in</span> csv_files){</span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file))</span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(filename)</span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(file, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^long"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(data))</span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"X"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(data))</span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a>  celltype_folder <span class="ot">&lt;-</span> <span class="fu">file.path</span>(OUTPUT_DIRECTORY, filename)</span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(celltype_folder)) {</span>
<span id="cb10-52"><a href="#cb10-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dir.create</span>(celltype_folder, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-53"><a href="#cb10-53" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb10-54"><a href="#cb10-54" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-55"><a href="#cb10-55" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (condition <span class="cf">in</span> conditions){</span>
<span id="cb10-56"><a href="#cb10-56" aria-hidden="true" tabindex="-1"></a>    condition <span class="ot">&lt;-</span> <span class="st">"TDP"</span></span>
<span id="cb10-57"><a href="#cb10-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(condition)</span>
<span id="cb10-58"><a href="#cb10-58" aria-hidden="true" tabindex="-1"></a>    condition_folder <span class="ot">&lt;-</span> <span class="fu">file.path</span>(celltype_folder, condition)</span>
<span id="cb10-59"><a href="#cb10-59" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(condition_folder)) {</span>
<span id="cb10-60"><a href="#cb10-60" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dir.create</span>(condition_folder, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-61"><a href="#cb10-61" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-62"><a href="#cb10-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-63"><a href="#cb10-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tryCatch</span>({</span>
<span id="cb10-64"><a href="#cb10-64" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb10-65"><a href="#cb10-65" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb10-66"><a href="#cb10-66" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> case_legend[case_legend<span class="sc">$</span>group_ID <span class="sc">%in%</span> <span class="fu">c</span>(REFERENCE_GROUP, condition), ]</span>
<span id="cb10-67"><a href="#cb10-67" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID)</span>
<span id="cb10-68"><a href="#cb10-68" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-69"><a href="#cb10-69" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>SV1 <span class="ot">&lt;-</span> covariate_data<span class="sc">$</span>SV1[<span class="fu">match</span>(comparison_data<span class="sc">$</span>FCX_ID, covariate_data<span class="sc">$</span>ID)]</span>
<span id="cb10-70"><a href="#cb10-70" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-71"><a href="#cb10-71" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">levels =</span> <span class="fu">c</span>(REFERENCE_GROUP, condition))</span>
<span id="cb10-72"><a href="#cb10-72" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">order</span>(comparison_data<span class="sc">$</span>group_ID), ]</span>
<span id="cb10-73"><a href="#cb10-73" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>FCX_ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-"</span>, <span class="st">"."</span>, comparison_data<span class="sc">$</span>FCX_ID)</span>
<span id="cb10-74"><a href="#cb10-74" aria-hidden="true" tabindex="-1"></a>      common_ids <span class="ot">&lt;-</span> <span class="fu">intersect</span>(comparison_data<span class="sc">$</span>FCX_ID, <span class="fu">rownames</span>(data))</span>
<span id="cb10-75"><a href="#cb10-75" aria-hidden="true" tabindex="-1"></a>      data_subset <span class="ot">&lt;-</span> data[<span class="fu">rownames</span>(data) <span class="sc">%in%</span> common_ids, ]</span>
<span id="cb10-76"><a href="#cb10-76" aria-hidden="true" tabindex="-1"></a>      data_ordered <span class="ot">&lt;-</span> data_subset[<span class="fu">match</span>(comparison_data<span class="sc">$</span>FCX_ID, <span class="fu">rownames</span>(data_subset)), ]</span>
<span id="cb10-77"><a href="#cb10-77" aria-hidden="true" tabindex="-1"></a>      <span class="co"># CLEAN NA</span></span>
<span id="cb10-78"><a href="#cb10-78" aria-hidden="true" tabindex="-1"></a>      data_ordered[<span class="sc">!</span><span class="fu">is.finite</span>(<span class="fu">as.matrix</span>(data_ordered))] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb10-79"><a href="#cb10-79" aria-hidden="true" tabindex="-1"></a>      data_ordered <span class="ot">&lt;-</span> data_ordered[<span class="fu">complete.cases</span>(data_ordered), ]</span>
<span id="cb10-80"><a href="#cb10-80" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">match</span>(<span class="fu">rownames</span>(data_ordered), comparison_data<span class="sc">$</span>FCX_ID), ]</span>
<span id="cb10-81"><a href="#cb10-81" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-82"><a href="#cb10-82" aria-hidden="true" tabindex="-1"></a>      group <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID[comparison_data<span class="sc">$</span>group_ID <span class="sc">%in%</span> <span class="fu">c</span>(REFERENCE_GROUP, condition)])</span>
<span id="cb10-83"><a href="#cb10-83" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts=</span><span class="fu">t</span>(data_ordered),<span class="at">group=</span>group)</span>
<span id="cb10-84"><a href="#cb10-84" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> y[, <span class="fu">colSums</span>(y<span class="sc">$</span>counts) <span class="sc">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb10-85"><a href="#cb10-85" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Keep only samples in comparison_data that are in y</span></span>
<span id="cb10-86"><a href="#cb10-86" aria-hidden="true" tabindex="-1"></a>      comparison_data <span class="ot">&lt;-</span> comparison_data[<span class="fu">match</span>(<span class="fu">colnames</span>(y), comparison_data<span class="sc">$</span>FCX_ID), ]</span>
<span id="cb10-87"><a href="#cb10-87" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-88"><a href="#cb10-88" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Average Log CPM histogram</span></span>
<span id="cb10-89"><a href="#cb10-89" aria-hidden="true" tabindex="-1"></a>      AveLogCPM <span class="ot">&lt;-</span> <span class="fu">aveLogCPM</span>(y)</span>
<span id="cb10-90"><a href="#cb10-90" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">normLibSizes</span>(y)</span>
<span id="cb10-91"><a href="#cb10-91" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-92"><a href="#cb10-92" aria-hidden="true" tabindex="-1"></a>      <span class="co"># MD plot</span></span>
<span id="cb10-93"><a href="#cb10-93" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"MD_plot_h.png"</span>)</span>
<span id="cb10-94"><a href="#cb10-94" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb10-95"><a href="#cb10-95" aria-hidden="true" tabindex="-1"></a>      MD_plot <span class="ot">&lt;-</span> <span class="fu">plotMD</span>(y, <span class="at">column=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-96"><a href="#cb10-96" aria-hidden="true" tabindex="-1"></a>        <span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>, <span class="at">col=</span><span class="st">"red"</span>, <span class="at">lty=</span><span class="dv">2</span>, <span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb10-97"><a href="#cb10-97" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb10-98"><a href="#cb10-98" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-99"><a href="#cb10-99" aria-hidden="true" tabindex="-1"></a>      <span class="co"># DESIGN MATRIX</span></span>
<span id="cb10-100"><a href="#cb10-100" aria-hidden="true" tabindex="-1"></a>      group <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">levels =</span> <span class="fu">c</span>(REFERENCE_GROUP, condition))</span>
<span id="cb10-101"><a href="#cb10-101" aria-hidden="true" tabindex="-1"></a>      comparison_data<span class="sc">$</span>group_ID <span class="ot">&lt;-</span> <span class="fu">relevel</span>(comparison_data<span class="sc">$</span>group_ID, <span class="at">ref =</span> REFERENCE_GROUP)</span>
<span id="cb10-102"><a href="#cb10-102" aria-hidden="true" tabindex="-1"></a>      design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> comparison_data<span class="sc">$</span>group_ID <span class="sc">+</span> comparison_data<span class="sc">$</span>SV1, <span class="at">data =</span> comparison_data)</span>
<span id="cb10-103"><a href="#cb10-103" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">c</span>(REFERENCE_GROUP, condition, <span class="st">"SV1"</span>)</span>
<span id="cb10-104"><a href="#cb10-104" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(design) <span class="sc">==</span> <span class="fu">ncol</span>(y)</span>
<span id="cb10-105"><a href="#cb10-105" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-106"><a href="#cb10-106" aria-hidden="true" tabindex="-1"></a>      <span class="co"># DISPERSION ESTIMATION</span></span>
<span id="cb10-107"><a href="#cb10-107" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(y,design)</span>
<span id="cb10-108"><a href="#cb10-108" aria-hidden="true" tabindex="-1"></a>      fit <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(y,design)</span>
<span id="cb10-109"><a href="#cb10-109" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-110"><a href="#cb10-110" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Diferential expression analysis</span></span>
<span id="cb10-111"><a href="#cb10-111" aria-hidden="true" tabindex="-1"></a>      B.LvsP <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="fu">paste0</span>(condition, <span class="st">" - "</span>, REFERENCE_GROUP), <span class="at">levels=</span>design)</span>
<span id="cb10-112"><a href="#cb10-112" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fit, <span class="at">contrast=</span>B.LvsP)</span>
<span id="cb10-113"><a href="#cb10-113" aria-hidden="true" tabindex="-1"></a>      is.de <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(res, <span class="at">adjust.method =</span> <span class="st">"fdr"</span>, <span class="at">p.value =</span> <span class="fl">0.05</span>, <span class="at">lfc=</span> <span class="dv">0</span>)</span>
<span id="cb10-114"><a href="#cb10-114" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> res<span class="sc">$</span>table</span>
<span id="cb10-115"><a href="#cb10-115" aria-hidden="true" tabindex="-1"></a>      results_table<span class="sc">$</span>adj_pval <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(results_table<span class="sc">$</span>PValue, <span class="at">method =</span> <span class="st">"fdr"</span>, <span class="at">n =</span> <span class="fu">length</span>(results_table<span class="sc">$</span>PValue))</span>
<span id="cb10-116"><a href="#cb10-116" aria-hidden="true" tabindex="-1"></a>      results_table <span class="ot">&lt;-</span> <span class="fu">cbind</span>(results_table, <span class="fu">data.frame</span>(is.de))</span>
<span id="cb10-117"><a href="#cb10-117" aria-hidden="true" tabindex="-1"></a>      <span class="co"># SAVING CSV RESULTS</span></span>
<span id="cb10-118"><a href="#cb10-118" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write.csv</span>(res<span class="sc">$</span>table, <span class="at">file =</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"res_h.csv"</span>))</span>
<span id="cb10-119"><a href="#cb10-119" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write.csv</span>(results_table, <span class="at">file =</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"results_adj_h.csv"</span>))</span>
<span id="cb10-120"><a href="#cb10-120" aria-hidden="true" tabindex="-1"></a>      <span class="do">## Visualizations</span></span>
<span id="cb10-121"><a href="#cb10-121" aria-hidden="true" tabindex="-1"></a>      <span class="co"># "Volcano like" plot</span></span>
<span id="cb10-122"><a href="#cb10-122" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"MD_res_plot_h.png"</span>)</span>
<span id="cb10-123"><a href="#cb10-123" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb10-124"><a href="#cb10-124" aria-hidden="true" tabindex="-1"></a>      <span class="fu">plotMD</span>(res, <span class="at">status=</span>is.de)</span>
<span id="cb10-125"><a href="#cb10-125" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb10-126"><a href="#cb10-126" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-127"><a href="#cb10-127" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Heatmap clustering </span></span>
<span id="cb10-128"><a href="#cb10-128" aria-hidden="true" tabindex="-1"></a>      logCPM <span class="ot">&lt;-</span> <span class="fu">cpm</span>(y, <span class="at">prior.count=</span><span class="dv">2</span>, <span class="at">log=</span><span class="cn">TRUE</span>)</span>
<span id="cb10-129"><a href="#cb10-129" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames</span>(logCPM) <span class="ot">&lt;-</span> y<span class="sc">$</span>genes<span class="sc">$</span>Symbol</span>
<span id="cb10-130"><a href="#cb10-130" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(logCPM) <span class="ot">&lt;-</span> <span class="fu">paste</span>(y<span class="sc">$</span>samples<span class="sc">$</span>group, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"-"</span>)</span>
<span id="cb10-131"><a href="#cb10-131" aria-hidden="true" tabindex="-1"></a>      tr <span class="ot">&lt;-</span> <span class="fu">glmTreat</span>(fit, <span class="at">contrast=</span>B.LvsP, <span class="at">lfc=</span><span class="fu">log2</span>(<span class="fl">1.5</span>))</span>
<span id="cb10-132"><a href="#cb10-132" aria-hidden="true" tabindex="-1"></a>      o <span class="ot">&lt;-</span> <span class="fu">order</span>(tr<span class="sc">$</span>table<span class="sc">$</span>PValue)</span>
<span id="cb10-133"><a href="#cb10-133" aria-hidden="true" tabindex="-1"></a>      logCPM <span class="ot">&lt;-</span> logCPM[o[<span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>],]</span>
<span id="cb10-134"><a href="#cb10-134" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-135"><a href="#cb10-135" aria-hidden="true" tabindex="-1"></a>      plot_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(condition_folder, <span class="st">"Heatmap_h.png"</span>)</span>
<span id="cb10-136"><a href="#cb10-136" aria-hidden="true" tabindex="-1"></a>      <span class="fu">png</span>(<span class="at">file =</span> plot_filename)</span>
<span id="cb10-137"><a href="#cb10-137" aria-hidden="true" tabindex="-1"></a>      <span class="fu">coolmap</span>(logCPM, <span class="at">margins=</span><span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">7</span>), <span class="at">lhei=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span>), <span class="at">lwid=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>))</span>
<span id="cb10-138"><a href="#cb10-138" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dev.off</span>()</span>
<span id="cb10-139"><a href="#cb10-139" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-140"><a href="#cb10-140" aria-hidden="true" tabindex="-1"></a>      df[condition, tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file))] <span class="ot">&lt;-</span> <span class="fu">nrow</span>(results_table[results_table<span class="sc">$</span>adj_pval<span class="sc">&lt;</span><span class="fl">0.05</span>,])</span>
<span id="cb10-141"><a href="#cb10-141" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb10-142"><a href="#cb10-142" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e){</span>
<span id="cb10-143"><a href="#cb10-143" aria-hidden="true" tabindex="-1"></a>      df[condition, tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file))] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb10-144"><a href="#cb10-144" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb10-145"><a href="#cb10-145" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb10-146"><a href="#cb10-146" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-147"><a href="#cb10-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-148"><a href="#cb10-148" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df, <span class="at">file =</span> <span class="fu">file.path</span>(OUTPUT_DIRECTORY, <span class="st">"summary_h.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="correlation-of-genes-of-interest-per-cell-state-with-neuropathology-markers" class="level1">
<h1>Correlation of genes of interest per cell state with neuropathology markers</h1>
<p>Evaluate correlations between selected genes of interest and transformed cell-type proportions across FTLD cohorts.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(car)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(unikn)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cluster)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>safe_cor <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y){</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">all</span>(<span class="fu">is.na</span>(x)) <span class="sc">|</span> <span class="fu">all</span>(<span class="fu">is.na</span>(y))) <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">estimate =</span> <span class="cn">NA</span>, <span class="at">p.value =</span> <span class="cn">NA</span>))</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(x, y, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(<span class="at">estimate =</span> res<span class="sc">$</span>estimate, <span class="at">p.value =</span> res<span class="sc">$</span>p.value)</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) <span class="fu">list</span>(<span class="at">estimate =</span> <span class="cn">NA</span>, <span class="at">p.value =</span> <span class="cn">NA</span>))</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="co"># C9</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>Expression_per_cell_directory <span class="ot">&lt;-</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/BAYESPRISM/FTLD/C9/CELL STATE ORIGINAL/"</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>OG_Covariables <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/CORRELATION/COVARIABLES/FTD_C9_neuropath_SOM.csv"</span>)</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>OG_Covariables_log    <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/CORRELATION/COVARIABLES/TRANSFORMED_COVARIABLES/Log/C9/Log_FTD_C9_neuropath_SOM.csv"</span>)</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>OG_Covariables_log2   <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/CORRELATION/COVARIABLES/TRANSFORMED_COVARIABLES/Log2/C9/Log2_FTD_C9_neuropath_SOM.csv"</span>)</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>OG_Covariables_logit  <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/CORRELATION/COVARIABLES/TRANSFORMED_COVARIABLES/Logit/C9/Logit_FTD_C9_neuropath_SOM.csv"</span>)</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>OG_Covariables_arcsin <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/CORRELATION/COVARIABLES/TRANSFORMED_COVARIABLES/ArcSin/C9/ArcSin_FTD_C9_neuropath_SOM.csv"</span>)</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Excitatory neurons</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>cells <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"CUX2_RASGRF2"</span>, <span class="st">"CUX2_RORB"</span>, <span class="st">"SCN4B_NEFH"</span>, <span class="st">"RORB_FOXO1"</span>, </span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>           <span class="st">"RORB_POU3F2"</span>, <span class="st">"RORB_ADGRL4"</span>, <span class="st">"RORB_LRRK1"</span>, <span class="st">"PCP4_NXPH2"</span>, </span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>           <span class="st">"VAT1L_EYA4"</span>, <span class="st">"THEMIS_NR4A2"</span>, <span class="st">"THEMIS_TMEM233"</span>, </span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>           <span class="st">"TLE4_CCBE1"</span>, <span class="st">"TLE4_MEGF11"</span>, <span class="st">"TLE4_SEMA3D"</span>, <span class="st">"GFAP-neg"</span>, </span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>           <span class="st">"GFAP-pos"</span>, <span class="st">"Micro"</span>, <span class="st">"Oligo"</span>, <span class="st">"OPC"</span>)</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"STMN2"</span>, <span class="st">"NPTX2"</span>, <span class="st">"C9orf72"</span>, <span class="st">"CXCL10"</span>, <span class="st">"SPP1"</span>, <span class="st">"CX3CR1"</span>) </span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>covariables <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"STMN2"</span>, <span class="st">"pTDP43"</span>, <span class="st">"lncRNA"</span>, <span class="st">"polyGA"</span>, <span class="st">"polyGP"</span>, <span class="st">"polyGR"</span>, <span class="st">"fociSENSE"</span>, <span class="st">"fociANTI"</span>,<span class="st">"ACSL3"</span>)</span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Cell =</span> <span class="fu">character</span>(),</span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Gene =</span> <span class="fu">character</span>(),</span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Covariable =</span> <span class="fu">character</span>(),</span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Gene_transformation =</span> <span class="fu">character</span>(),</span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Cov_transformation =</span> <span class="fu">character</span>(),</span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Spearman =</span> <span class="fu">numeric</span>(),</span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>                      <span class="at">p_value =</span> <span class="fu">numeric</span>(),</span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>                      <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cell <span class="cf">in</span> cells){</span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(Expression_per_cell_directory, cell, <span class="st">".csv"</span>), <span class="at">row.names =</span> <span class="st">"X"</span>)</span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (gene <span class="cf">in</span> genes){</span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>    common_ids <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(data), OG_Covariables_log<span class="sc">$</span>X)</span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a>    subset_gene_subset <span class="ot">&lt;-</span> data[common_ids, gene]</span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a>    subset_Covariables  <span class="ot">&lt;-</span> OG_Covariables[<span class="fu">match</span>(common_ids, OG_Covariables<span class="sc">$</span>X), ]</span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a>    subset_Covariables_log  <span class="ot">&lt;-</span> OG_Covariables_log[<span class="fu">match</span>(common_ids, OG_Covariables_log<span class="sc">$</span>X),]</span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a>    subset_Covariables_log2  <span class="ot">&lt;-</span> OG_Covariables_log2[<span class="fu">match</span>(common_ids, OG_Covariables_log2<span class="sc">$</span>X), ]</span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a>    subset_Covariables_logit  <span class="ot">&lt;-</span> OG_Covariables_logit[<span class="fu">match</span>(common_ids, OG_Covariables_logit<span class="sc">$</span>X), ]</span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a>    subset_Covariables_arcsin  <span class="ot">&lt;-</span> OG_Covariables_arcsin[<span class="fu">match</span>(common_ids, OG_Covariables_arcsin<span class="sc">$</span>X), ]</span>
<span id="cb11-57"><a href="#cb11-57" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-58"><a href="#cb11-58" aria-hidden="true" tabindex="-1"></a>    <span class="co"># gene transformations</span></span>
<span id="cb11-59"><a href="#cb11-59" aria-hidden="true" tabindex="-1"></a>    subset_log_data    <span class="ot">&lt;-</span> <span class="fu">log1p</span>(subset_gene_subset)</span>
<span id="cb11-60"><a href="#cb11-60" aria-hidden="true" tabindex="-1"></a>    subset_log2_data   <span class="ot">&lt;-</span> <span class="fu">log</span>(subset_gene_subset, <span class="at">base =</span> <span class="dv">2</span>)</span>
<span id="cb11-61"><a href="#cb11-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># protect logit and arcsin</span></span>
<span id="cb11-62"><a href="#cb11-62" aria-hidden="true" tabindex="-1"></a>    subset_logit_data <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>({</span>
<span id="cb11-63"><a href="#cb11-63" aria-hidden="true" tabindex="-1"></a>      <span class="fu">logit</span>(gene_subset, <span class="at">percents =</span> <span class="cn">FALSE</span>, <span class="at">adjust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-64"><a href="#cb11-64" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e) <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(subset_gene_subset)))</span>
<span id="cb11-65"><a href="#cb11-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-66"><a href="#cb11-66" aria-hidden="true" tabindex="-1"></a>    subset_norm_logit_data <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>({</span>
<span id="cb11-67"><a href="#cb11-67" aria-hidden="true" tabindex="-1"></a>      <span class="fu">logit</span>(subset_gene_subset<span class="sc">/</span><span class="fu">max</span>(subset_gene_subset), <span class="at">percents =</span> <span class="cn">FALSE</span>, <span class="at">adjust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-68"><a href="#cb11-68" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e) <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(subset_gene_subset)))</span>
<span id="cb11-69"><a href="#cb11-69" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-70"><a href="#cb11-70" aria-hidden="true" tabindex="-1"></a>    subset_arcsin_data <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>({</span>
<span id="cb11-71"><a href="#cb11-71" aria-hidden="true" tabindex="-1"></a>      <span class="fu">asin</span>(<span class="fu">sqrt</span>(subset_gene_subset))</span>
<span id="cb11-72"><a href="#cb11-72" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e) <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(subset_gene_subset)))</span>
<span id="cb11-73"><a href="#cb11-73" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-74"><a href="#cb11-74" aria-hidden="true" tabindex="-1"></a>    subset_norm_arcsin_data <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>({</span>
<span id="cb11-75"><a href="#cb11-75" aria-hidden="true" tabindex="-1"></a>      <span class="fu">asin</span>(<span class="fu">sqrt</span>(subset_gene_subset<span class="sc">/</span><span class="fu">max</span>(subset_gene_subset)))</span>
<span id="cb11-76"><a href="#cb11-76" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e) <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(subset_gene_subset)))</span>
<span id="cb11-77"><a href="#cb11-77" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-78"><a href="#cb11-78" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-79"><a href="#cb11-79" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (covariable <span class="cf">in</span> covariables){</span>
<span id="cb11-80"><a href="#cb11-80" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb11-81"><a href="#cb11-81" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (covariable <span class="sc">==</span> <span class="st">"AAA"</span>){</span>
<span id="cb11-82"><a href="#cb11-82" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-83"><a href="#cb11-83" aria-hidden="true" tabindex="-1"></a>        outlier <span class="ot">&lt;-</span> <span class="st">"7BLACK"</span></span>
<span id="cb11-84"><a href="#cb11-84" aria-hidden="true" tabindex="-1"></a>        no_outlier_ids <span class="ot">&lt;-</span> common_ids[common_ids <span class="sc">!=</span> <span class="st">"7BLACK"</span>] </span>
<span id="cb11-85"><a href="#cb11-85" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-86"><a href="#cb11-86" aria-hidden="true" tabindex="-1"></a>        gene_subset <span class="ot">&lt;-</span> subset_gene_subset[no_outlier_ids,]</span>
<span id="cb11-87"><a href="#cb11-87" aria-hidden="true" tabindex="-1"></a>        Covariables  <span class="ot">&lt;-</span> subset_Covariables[<span class="fu">match</span>(no_outlier_ids, subset_Covariables<span class="sc">$</span>X), ]</span>
<span id="cb11-88"><a href="#cb11-88" aria-hidden="true" tabindex="-1"></a>        Covariables_log  <span class="ot">&lt;-</span> subset_Covariables_log[<span class="fu">match</span>(no_outlier_ids, subset_Covariables_log<span class="sc">$</span>X),]</span>
<span id="cb11-89"><a href="#cb11-89" aria-hidden="true" tabindex="-1"></a>        Covariables_log2  <span class="ot">&lt;-</span> subset_Covariables_log2[<span class="fu">match</span>(no_outlier_ids, subset_Covariables_log2<span class="sc">$</span>X), ]</span>
<span id="cb11-90"><a href="#cb11-90" aria-hidden="true" tabindex="-1"></a>        Covariables_logit  <span class="ot">&lt;-</span> subset_Covariables_logit[<span class="fu">match</span>(no_outlier_ids, subset_Covariables_logit<span class="sc">$</span>X), ]</span>
<span id="cb11-91"><a href="#cb11-91" aria-hidden="true" tabindex="-1"></a>        Covariables_arcsin  <span class="ot">&lt;-</span> subset_Covariables_arcsin[<span class="fu">match</span>(no_outlier_ids, subset_Covariables_arcsin<span class="sc">$</span>X), ]</span>
<span id="cb11-92"><a href="#cb11-92" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-93"><a href="#cb11-93" aria-hidden="true" tabindex="-1"></a>        <span class="co"># gene transformations</span></span>
<span id="cb11-94"><a href="#cb11-94" aria-hidden="true" tabindex="-1"></a>        log_data    <span class="ot">&lt;-</span> subset_log_data</span>
<span id="cb11-95"><a href="#cb11-95" aria-hidden="true" tabindex="-1"></a>        log2_data   <span class="ot">&lt;-</span> subset_log2_data</span>
<span id="cb11-96"><a href="#cb11-96" aria-hidden="true" tabindex="-1"></a>        logit_data <span class="ot">&lt;-</span> subset_logit_data</span>
<span id="cb11-97"><a href="#cb11-97" aria-hidden="true" tabindex="-1"></a>        norm_logit_data <span class="ot">&lt;-</span> subset_norm_logit_data</span>
<span id="cb11-98"><a href="#cb11-98" aria-hidden="true" tabindex="-1"></a>        arcsin_data <span class="ot">&lt;-</span> subset_arcsin_data</span>
<span id="cb11-99"><a href="#cb11-99" aria-hidden="true" tabindex="-1"></a>        norm_arcsin_data <span class="ot">&lt;-</span> subset_norm_arcsin_data</span>
<span id="cb11-100"><a href="#cb11-100" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-101"><a href="#cb11-101" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb11-102"><a href="#cb11-102" aria-hidden="true" tabindex="-1"></a>        gene_subset <span class="ot">&lt;-</span> subset_gene_subset</span>
<span id="cb11-103"><a href="#cb11-103" aria-hidden="true" tabindex="-1"></a>        Covariables  <span class="ot">&lt;-</span> subset_Covariables</span>
<span id="cb11-104"><a href="#cb11-104" aria-hidden="true" tabindex="-1"></a>        Covariables_log  <span class="ot">&lt;-</span> subset_Covariables_log</span>
<span id="cb11-105"><a href="#cb11-105" aria-hidden="true" tabindex="-1"></a>        Covariables_log2  <span class="ot">&lt;-</span> subset_Covariables_log2</span>
<span id="cb11-106"><a href="#cb11-106" aria-hidden="true" tabindex="-1"></a>        Covariables_logit  <span class="ot">&lt;-</span> subset_Covariables_logit</span>
<span id="cb11-107"><a href="#cb11-107" aria-hidden="true" tabindex="-1"></a>        Covariables_arcsin  <span class="ot">&lt;-</span> subset_Covariables_arcsin</span>
<span id="cb11-108"><a href="#cb11-108" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-109"><a href="#cb11-109" aria-hidden="true" tabindex="-1"></a>        <span class="co"># gene transformations</span></span>
<span id="cb11-110"><a href="#cb11-110" aria-hidden="true" tabindex="-1"></a>        log_data    <span class="ot">&lt;-</span> subset_log_data</span>
<span id="cb11-111"><a href="#cb11-111" aria-hidden="true" tabindex="-1"></a>        log2_data   <span class="ot">&lt;-</span> subset_log2_data</span>
<span id="cb11-112"><a href="#cb11-112" aria-hidden="true" tabindex="-1"></a>        logit_data <span class="ot">&lt;-</span> subset_logit_data</span>
<span id="cb11-113"><a href="#cb11-113" aria-hidden="true" tabindex="-1"></a>        norm_logit_data <span class="ot">&lt;-</span> subset_norm_logit_data</span>
<span id="cb11-114"><a href="#cb11-114" aria-hidden="true" tabindex="-1"></a>        arcsin_data <span class="ot">&lt;-</span> subset_arcsin_data</span>
<span id="cb11-115"><a href="#cb11-115" aria-hidden="true" tabindex="-1"></a>        norm_arcsin_data <span class="ot">&lt;-</span> subset_norm_arcsin_data</span>
<span id="cb11-116"><a href="#cb11-116" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb11-117"><a href="#cb11-117" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb11-118"><a href="#cb11-118" aria-hidden="true" tabindex="-1"></a>      result_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(gene_subset, Covariables[, covariable])</span>
<span id="cb11-119"><a href="#cb11-119" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb11-120"><a href="#cb11-120" aria-hidden="true" tabindex="-1"></a>      result_log_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(log_data, Covariables[, covariable])</span>
<span id="cb11-121"><a href="#cb11-121" aria-hidden="true" tabindex="-1"></a>      result_log_log  <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(log_data, Covariables_log[,covariable])</span>
<span id="cb11-122"><a href="#cb11-122" aria-hidden="true" tabindex="-1"></a>      result_gene_log <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(gene_subset, Covariables_log[,covariable])</span>
<span id="cb11-123"><a href="#cb11-123" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb11-124"><a href="#cb11-124" aria-hidden="true" tabindex="-1"></a>      result_log2_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(log2_data, Covariables[, covariable])</span>
<span id="cb11-125"><a href="#cb11-125" aria-hidden="true" tabindex="-1"></a>      result_log2_log2 <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(log2_data, Covariables_log2[,covariable])</span>
<span id="cb11-126"><a href="#cb11-126" aria-hidden="true" tabindex="-1"></a>      result_gene_log2 <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(gene_subset, Covariables_log2[,covariable])</span>
<span id="cb11-127"><a href="#cb11-127" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb11-128"><a href="#cb11-128" aria-hidden="true" tabindex="-1"></a>      result_logit_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(logit_data, Covariables[, covariable])</span>
<span id="cb11-129"><a href="#cb11-129" aria-hidden="true" tabindex="-1"></a>      result_logit_logit <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(logit_data, Covariables_logit[,covariable])</span>
<span id="cb11-130"><a href="#cb11-130" aria-hidden="true" tabindex="-1"></a>      result_gene_logit <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(gene_subset, Covariables_logit[, covariable])</span>
<span id="cb11-131"><a href="#cb11-131" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb11-132"><a href="#cb11-132" aria-hidden="true" tabindex="-1"></a>      result_norm_logit_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(norm_logit_data, Covariables[, covariable])</span>
<span id="cb11-133"><a href="#cb11-133" aria-hidden="true" tabindex="-1"></a>      result_norm_logit_logit <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(norm_logit_data,  Covariables_logit[,covariable])</span>
<span id="cb11-134"><a href="#cb11-134" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb11-135"><a href="#cb11-135" aria-hidden="true" tabindex="-1"></a>      result_arcsin_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(arcsin_data, Covariables[, covariable])</span>
<span id="cb11-136"><a href="#cb11-136" aria-hidden="true" tabindex="-1"></a>      result_arcsin_arcsin <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(arcsin_data, Covariables_arcsin[,covariable])</span>
<span id="cb11-137"><a href="#cb11-137" aria-hidden="true" tabindex="-1"></a>      result_gene_arcsin <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(gene_subset, Covariables_arcsin[,covariable])</span>
<span id="cb11-138"><a href="#cb11-138" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb11-139"><a href="#cb11-139" aria-hidden="true" tabindex="-1"></a>      result_norm_arcsin_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(norm_arcsin_data, Covariables[, covariable])</span>
<span id="cb11-140"><a href="#cb11-140" aria-hidden="true" tabindex="-1"></a>      result_norm_arcsin_arcsin <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(norm_arcsin_data, Covariables_arcsin[,covariable])</span>
<span id="cb11-141"><a href="#cb11-141" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb11-142"><a href="#cb11-142" aria-hidden="true" tabindex="-1"></a>      results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results,</span>
<span id="cb11-143"><a href="#cb11-143" aria-hidden="true" tabindex="-1"></a>                       <span class="co">#Cor - Cor</span></span>
<span id="cb11-144"><a href="#cb11-144" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-145"><a href="#cb11-145" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-146"><a href="#cb11-146" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">""</span>,</span>
<span id="cb11-147"><a href="#cb11-147" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>, </span>
<span id="cb11-148"><a href="#cb11-148" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_corr<span class="sc">$</span>estimate, </span>
<span id="cb11-149"><a href="#cb11-149" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_corr<span class="sc">$</span>p.value),</span>
<span id="cb11-150"><a href="#cb11-150" aria-hidden="true" tabindex="-1"></a>                       <span class="co">#Log - Cor</span></span>
<span id="cb11-151"><a href="#cb11-151" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-152"><a href="#cb11-152" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-153"><a href="#cb11-153" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Log"</span>,</span>
<span id="cb11-154"><a href="#cb11-154" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>, </span>
<span id="cb11-155"><a href="#cb11-155" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_log_corr<span class="sc">$</span>estimate, </span>
<span id="cb11-156"><a href="#cb11-156" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_log_corr<span class="sc">$</span>p.value),</span>
<span id="cb11-157"><a href="#cb11-157" aria-hidden="true" tabindex="-1"></a>                       <span class="co">#Log - Log</span></span>
<span id="cb11-158"><a href="#cb11-158" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-159"><a href="#cb11-159" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-160"><a href="#cb11-160" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Log"</span>,</span>
<span id="cb11-161"><a href="#cb11-161" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Log"</span>, </span>
<span id="cb11-162"><a href="#cb11-162" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_log_log<span class="sc">$</span>estimate, </span>
<span id="cb11-163"><a href="#cb11-163" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_log_log<span class="sc">$</span>p.value),</span>
<span id="cb11-164"><a href="#cb11-164" aria-hidden="true" tabindex="-1"></a>                       <span class="co">#Gene - Log</span></span>
<span id="cb11-165"><a href="#cb11-165" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-166"><a href="#cb11-166" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-167"><a href="#cb11-167" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">""</span>,</span>
<span id="cb11-168"><a href="#cb11-168" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Log"</span>, </span>
<span id="cb11-169"><a href="#cb11-169" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_gene_log<span class="sc">$</span>estimate, </span>
<span id="cb11-170"><a href="#cb11-170" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_gene_log<span class="sc">$</span>p.value),</span>
<span id="cb11-171"><a href="#cb11-171" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Log2 - Corr</span></span>
<span id="cb11-172"><a href="#cb11-172" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-173"><a href="#cb11-173" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-174"><a href="#cb11-174" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Log2"</span>,</span>
<span id="cb11-175"><a href="#cb11-175" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>, </span>
<span id="cb11-176"><a href="#cb11-176" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_log2_corr<span class="sc">$</span>estimate, </span>
<span id="cb11-177"><a href="#cb11-177" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_log2_corr<span class="sc">$</span>p.value),</span>
<span id="cb11-178"><a href="#cb11-178" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Log2 - Log2</span></span>
<span id="cb11-179"><a href="#cb11-179" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-180"><a href="#cb11-180" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-181"><a href="#cb11-181" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Log2"</span>,</span>
<span id="cb11-182"><a href="#cb11-182" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Log2"</span>, </span>
<span id="cb11-183"><a href="#cb11-183" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_log2_log2<span class="sc">$</span>estimate, </span>
<span id="cb11-184"><a href="#cb11-184" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_log2_log2<span class="sc">$</span>p.value),</span>
<span id="cb11-185"><a href="#cb11-185" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Gene - Log2</span></span>
<span id="cb11-186"><a href="#cb11-186" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-187"><a href="#cb11-187" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-188"><a href="#cb11-188" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">""</span>,</span>
<span id="cb11-189"><a href="#cb11-189" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Log2"</span>, </span>
<span id="cb11-190"><a href="#cb11-190" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_gene_log2<span class="sc">$</span>estimate, </span>
<span id="cb11-191"><a href="#cb11-191" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_gene_log2<span class="sc">$</span>p.value),</span>
<span id="cb11-192"><a href="#cb11-192" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Logit - Corr</span></span>
<span id="cb11-193"><a href="#cb11-193" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-194"><a href="#cb11-194" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-195"><a href="#cb11-195" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Logit"</span>,</span>
<span id="cb11-196"><a href="#cb11-196" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>,  </span>
<span id="cb11-197"><a href="#cb11-197" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_logit_corr<span class="sc">$</span>estimate, </span>
<span id="cb11-198"><a href="#cb11-198" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_logit_corr<span class="sc">$</span>p.value),</span>
<span id="cb11-199"><a href="#cb11-199" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Logit - Logit</span></span>
<span id="cb11-200"><a href="#cb11-200" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-201"><a href="#cb11-201" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-202"><a href="#cb11-202" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Logit"</span>,</span>
<span id="cb11-203"><a href="#cb11-203" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Logit"</span>,  </span>
<span id="cb11-204"><a href="#cb11-204" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_logit_logit<span class="sc">$</span>estimate, </span>
<span id="cb11-205"><a href="#cb11-205" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_logit_logit<span class="sc">$</span>p.value),</span>
<span id="cb11-206"><a href="#cb11-206" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Gene - Logit</span></span>
<span id="cb11-207"><a href="#cb11-207" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-208"><a href="#cb11-208" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-209"><a href="#cb11-209" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">""</span>,</span>
<span id="cb11-210"><a href="#cb11-210" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Logit"</span>,  </span>
<span id="cb11-211"><a href="#cb11-211" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_gene_logit<span class="sc">$</span>estimate, </span>
<span id="cb11-212"><a href="#cb11-212" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_gene_logit<span class="sc">$</span>p.value),</span>
<span id="cb11-213"><a href="#cb11-213" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Norm+Logit - Corr</span></span>
<span id="cb11-214"><a href="#cb11-214" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-215"><a href="#cb11-215" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-216"><a href="#cb11-216" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Norm + Logit"</span>,</span>
<span id="cb11-217"><a href="#cb11-217" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>,  </span>
<span id="cb11-218"><a href="#cb11-218" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_norm_logit_corr<span class="sc">$</span>estimate, </span>
<span id="cb11-219"><a href="#cb11-219" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_norm_logit_corr<span class="sc">$</span>p.value),</span>
<span id="cb11-220"><a href="#cb11-220" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Norm+Logit - Logit</span></span>
<span id="cb11-221"><a href="#cb11-221" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-222"><a href="#cb11-222" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-223"><a href="#cb11-223" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Norm + Logit"</span>,</span>
<span id="cb11-224"><a href="#cb11-224" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Logit"</span>,  </span>
<span id="cb11-225"><a href="#cb11-225" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_norm_logit_logit<span class="sc">$</span>estimate, </span>
<span id="cb11-226"><a href="#cb11-226" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_norm_logit_logit<span class="sc">$</span>p.value),</span>
<span id="cb11-227"><a href="#cb11-227" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Arcsin - Corr</span></span>
<span id="cb11-228"><a href="#cb11-228" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-229"><a href="#cb11-229" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-230"><a href="#cb11-230" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Arcsin"</span>,</span>
<span id="cb11-231"><a href="#cb11-231" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>,  </span>
<span id="cb11-232"><a href="#cb11-232" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_arcsin_corr<span class="sc">$</span>estimate, </span>
<span id="cb11-233"><a href="#cb11-233" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_arcsin_corr<span class="sc">$</span>p.value),</span>
<span id="cb11-234"><a href="#cb11-234" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Arcsin - Arcsin</span></span>
<span id="cb11-235"><a href="#cb11-235" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-236"><a href="#cb11-236" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-237"><a href="#cb11-237" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Arcsin"</span>,</span>
<span id="cb11-238"><a href="#cb11-238" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Arcsin"</span>,  </span>
<span id="cb11-239"><a href="#cb11-239" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_arcsin_arcsin<span class="sc">$</span>estimate, </span>
<span id="cb11-240"><a href="#cb11-240" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_arcsin_arcsin<span class="sc">$</span>p.value),</span>
<span id="cb11-241"><a href="#cb11-241" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Gene - Arcsin</span></span>
<span id="cb11-242"><a href="#cb11-242" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-243"><a href="#cb11-243" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-244"><a href="#cb11-244" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">""</span>,</span>
<span id="cb11-245"><a href="#cb11-245" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Arcsin"</span>,  </span>
<span id="cb11-246"><a href="#cb11-246" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_gene_arcsin<span class="sc">$</span>estimate, </span>
<span id="cb11-247"><a href="#cb11-247" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_gene_arcsin<span class="sc">$</span>p.value),</span>
<span id="cb11-248"><a href="#cb11-248" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Norm+Arcsin - Corr</span></span>
<span id="cb11-249"><a href="#cb11-249" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-250"><a href="#cb11-250" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-251"><a href="#cb11-251" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Norm + Arcsin"</span>,</span>
<span id="cb11-252"><a href="#cb11-252" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>,  </span>
<span id="cb11-253"><a href="#cb11-253" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_norm_arcsin_corr<span class="sc">$</span>estimate, </span>
<span id="cb11-254"><a href="#cb11-254" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_norm_arcsin_corr<span class="sc">$</span>p.value),</span>
<span id="cb11-255"><a href="#cb11-255" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Norm+Arcsin - Arcsin</span></span>
<span id="cb11-256"><a href="#cb11-256" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb11-257"><a href="#cb11-257" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb11-258"><a href="#cb11-258" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Norm + Arcsin"</span>,</span>
<span id="cb11-259"><a href="#cb11-259" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Arcsin"</span>,  </span>
<span id="cb11-260"><a href="#cb11-260" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_norm_arcsin_arcsin<span class="sc">$</span>estimate, </span>
<span id="cb11-261"><a href="#cb11-261" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_norm_arcsin_arcsin<span class="sc">$</span>p.value))</span>
<span id="cb11-262"><a href="#cb11-262" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb11-263"><a href="#cb11-263" aria-hidden="true" tabindex="-1"></a>      result_log_corr <span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb11-264"><a href="#cb11-264" aria-hidden="true" tabindex="-1"></a>      result_log_log  <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-265"><a href="#cb11-265" aria-hidden="true" tabindex="-1"></a>      result_gene_log <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-266"><a href="#cb11-266" aria-hidden="true" tabindex="-1"></a>      result_log2_corr <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-267"><a href="#cb11-267" aria-hidden="true" tabindex="-1"></a>      result_log2_log2 <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-268"><a href="#cb11-268" aria-hidden="true" tabindex="-1"></a>      result_gene_log2 <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-269"><a href="#cb11-269" aria-hidden="true" tabindex="-1"></a>      result_logit_corr <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-270"><a href="#cb11-270" aria-hidden="true" tabindex="-1"></a>      result_logit_logit <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-271"><a href="#cb11-271" aria-hidden="true" tabindex="-1"></a>      result_gene_logit <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-272"><a href="#cb11-272" aria-hidden="true" tabindex="-1"></a>      result_norm_logit_corr <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-273"><a href="#cb11-273" aria-hidden="true" tabindex="-1"></a>      result_norm_logit_logit <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-274"><a href="#cb11-274" aria-hidden="true" tabindex="-1"></a>      result_arcsin_corr <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-275"><a href="#cb11-275" aria-hidden="true" tabindex="-1"></a>      result_arcsin_arcsin <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-276"><a href="#cb11-276" aria-hidden="true" tabindex="-1"></a>      result_gene_arcsin <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-277"><a href="#cb11-277" aria-hidden="true" tabindex="-1"></a>      result_norm_arcsin_corr <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-278"><a href="#cb11-278" aria-hidden="true" tabindex="-1"></a>      result_norm_arcsin_arcsin <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-279"><a href="#cb11-279" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb11-280"><a href="#cb11-280" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb11-281"><a href="#cb11-281" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-282"><a href="#cb11-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-283"><a href="#cb11-283" aria-hidden="true" tabindex="-1"></a>results</span>
<span id="cb11-284"><a href="#cb11-284" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(results, <span class="at">file =</span> <span class="st">"/media/jaumatell/datos/URI/BAYESPRISM_12_3/CORRELATION/GENES_CORR/C9_genes_correlation_allCovs.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(car)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(unikn)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cluster)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>safe_cor <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y){</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">all</span>(<span class="fu">is.na</span>(x)) <span class="sc">|</span> <span class="fu">all</span>(<span class="fu">is.na</span>(y))) <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">estimate =</span> <span class="cn">NA</span>, <span class="at">p.value =</span> <span class="cn">NA</span>))</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(x, y, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(<span class="at">estimate =</span> res<span class="sc">$</span>estimate, <span class="at">p.value =</span> res<span class="sc">$</span>p.value)</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) <span class="fu">list</span>(<span class="at">estimate =</span> <span class="cn">NA</span>, <span class="at">p.value =</span> <span class="cn">NA</span>))</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="co"># tdp</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>Expression_per_cell_directory <span class="ot">&lt;-</span> <span class="st">".../BAYESPRISM_12_3/BAYESPRISM/FTLD/TDP/CELL STATE ORIGINAL/"</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>OG_Covariables <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">".../BAYESPRISM_12_3/CORRELATION/COVARIABLES/FTD_TDP_neuropath_SOM.csv"</span>)</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>OG_Covariables_log    <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">".../BAYESPRISM_12_3/CORRELATION/COVARIABLES/TRANSFORMED_COVARIABLES/Log/C9/Log_FTD_TDP_neuropath_SOM.csv"</span>)</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>OG_Covariables_log2   <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">".../BAYESPRISM_12_3/CORRELATION/COVARIABLES/TRANSFORMED_COVARIABLES/Log2/C9/Log2_FTD_TDP_neuropath_SOM.csv"</span>)</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>OG_Covariables_logit  <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">".../BAYESPRISM_12_3/CORRELATION/COVARIABLES/TRANSFORMED_COVARIABLES/Logit/C9/Logit_FTD_TDP_neuropath_SOM.csv"</span>)</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>OG_Covariables_arcsin <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">".../BAYESPRISM_12_3/CORRELATION/COVARIABLES/TRANSFORMED_COVARIABLES/ArcSin/C9/ArcSin_FTD_TDP_neuropath_SOM.csv"</span>)</span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Excitatory neurons</span></span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>cells <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"CUX2_RASGRF2"</span>, <span class="st">"CUX2_RORB"</span>, <span class="st">"SCN4B_NEFH"</span>, <span class="st">"RORB_FOXO1"</span>, </span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>           <span class="st">"RORB_POU3F2"</span>, <span class="st">"RORB_ADGRL4"</span>, <span class="st">"RORB_LRRK1"</span>, <span class="st">"PCP4_NXPH2"</span>, </span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>           <span class="st">"VAT1L_EYA4"</span>, <span class="st">"THEMIS_NR4A2"</span>, <span class="st">"THEMIS_TMEM233"</span>, </span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a>           <span class="st">"TLE4_CCBE1"</span>, <span class="st">"TLE4_MEGF11"</span>, <span class="st">"TLE4_SEMA3D"</span>, <span class="st">"GFAP-neg"</span>, </span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>           <span class="st">"GFAP-pos"</span>, <span class="st">"Micro"</span>, <span class="st">"Oligo"</span>, <span class="st">"OPC"</span>)</span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"STMN2"</span>, <span class="st">"NPTX2"</span>, <span class="st">"C9orf72"</span>, <span class="st">"CXCL10"</span>, <span class="st">"SPP1"</span>, <span class="st">"CX3CR1"</span>) </span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a>covariables <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"STMN2"</span>, <span class="st">"TDP43b"</span>)</span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Cell =</span> <span class="fu">character</span>(),</span>
<span id="cb12-38"><a href="#cb12-38" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Gene =</span> <span class="fu">character</span>(),</span>
<span id="cb12-39"><a href="#cb12-39" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Covariable =</span> <span class="fu">character</span>(),</span>
<span id="cb12-40"><a href="#cb12-40" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Gene_transformation =</span> <span class="fu">character</span>(),</span>
<span id="cb12-41"><a href="#cb12-41" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Cov_transformation =</span> <span class="fu">character</span>(),</span>
<span id="cb12-42"><a href="#cb12-42" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Spearman =</span> <span class="fu">numeric</span>(),</span>
<span id="cb12-43"><a href="#cb12-43" aria-hidden="true" tabindex="-1"></a>                      <span class="at">p_value =</span> <span class="fu">numeric</span>(),</span>
<span id="cb12-44"><a href="#cb12-44" aria-hidden="true" tabindex="-1"></a>                      <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb12-45"><a href="#cb12-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-46"><a href="#cb12-46" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cell <span class="cf">in</span> cells){</span>
<span id="cb12-47"><a href="#cb12-47" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(Expression_per_cell_directory, cell, <span class="st">".csv"</span>), <span class="at">row.names =</span> <span class="st">"X"</span>)</span>
<span id="cb12-48"><a href="#cb12-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-49"><a href="#cb12-49" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (gene <span class="cf">in</span> genes){</span>
<span id="cb12-50"><a href="#cb12-50" aria-hidden="true" tabindex="-1"></a>    common_ids <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(data), OG_Covariables_log<span class="sc">$</span>X)</span>
<span id="cb12-51"><a href="#cb12-51" aria-hidden="true" tabindex="-1"></a>    subset_gene_subset <span class="ot">&lt;-</span> data[common_ids, gene]</span>
<span id="cb12-52"><a href="#cb12-52" aria-hidden="true" tabindex="-1"></a>    subset_Covariables  <span class="ot">&lt;-</span> OG_Covariables[<span class="fu">match</span>(common_ids, OG_Covariables<span class="sc">$</span>X), ]</span>
<span id="cb12-53"><a href="#cb12-53" aria-hidden="true" tabindex="-1"></a>    subset_Covariables_log  <span class="ot">&lt;-</span> OG_Covariables_log[<span class="fu">match</span>(common_ids, OG_Covariables_log<span class="sc">$</span>X),]</span>
<span id="cb12-54"><a href="#cb12-54" aria-hidden="true" tabindex="-1"></a>    subset_Covariables_log2  <span class="ot">&lt;-</span> OG_Covariables_log2[<span class="fu">match</span>(common_ids, OG_Covariables_log2<span class="sc">$</span>X), ]</span>
<span id="cb12-55"><a href="#cb12-55" aria-hidden="true" tabindex="-1"></a>    subset_Covariables_logit  <span class="ot">&lt;-</span> OG_Covariables_logit[<span class="fu">match</span>(common_ids, OG_Covariables_logit<span class="sc">$</span>X), ]</span>
<span id="cb12-56"><a href="#cb12-56" aria-hidden="true" tabindex="-1"></a>    subset_Covariables_arcsin  <span class="ot">&lt;-</span> OG_Covariables_arcsin[<span class="fu">match</span>(common_ids, OG_Covariables_arcsin<span class="sc">$</span>X), ]</span>
<span id="cb12-57"><a href="#cb12-57" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-58"><a href="#cb12-58" aria-hidden="true" tabindex="-1"></a>    <span class="co"># gene transformations</span></span>
<span id="cb12-59"><a href="#cb12-59" aria-hidden="true" tabindex="-1"></a>    subset_log_data    <span class="ot">&lt;-</span> <span class="fu">log1p</span>(subset_gene_subset)</span>
<span id="cb12-60"><a href="#cb12-60" aria-hidden="true" tabindex="-1"></a>    subset_log2_data   <span class="ot">&lt;-</span> <span class="fu">log</span>(subset_gene_subset, <span class="at">base =</span> <span class="dv">2</span>)</span>
<span id="cb12-61"><a href="#cb12-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># protect logit and arcsin</span></span>
<span id="cb12-62"><a href="#cb12-62" aria-hidden="true" tabindex="-1"></a>    subset_logit_data <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>({</span>
<span id="cb12-63"><a href="#cb12-63" aria-hidden="true" tabindex="-1"></a>      <span class="fu">logit</span>(gene_subset, <span class="at">percents =</span> <span class="cn">FALSE</span>, <span class="at">adjust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-64"><a href="#cb12-64" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e) <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(subset_gene_subset)))</span>
<span id="cb12-65"><a href="#cb12-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-66"><a href="#cb12-66" aria-hidden="true" tabindex="-1"></a>    subset_norm_logit_data <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>({</span>
<span id="cb12-67"><a href="#cb12-67" aria-hidden="true" tabindex="-1"></a>      <span class="fu">logit</span>(subset_gene_subset<span class="sc">/</span><span class="fu">max</span>(subset_gene_subset), <span class="at">percents =</span> <span class="cn">FALSE</span>, <span class="at">adjust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-68"><a href="#cb12-68" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e) <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(subset_gene_subset)))</span>
<span id="cb12-69"><a href="#cb12-69" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-70"><a href="#cb12-70" aria-hidden="true" tabindex="-1"></a>    subset_arcsin_data <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>({</span>
<span id="cb12-71"><a href="#cb12-71" aria-hidden="true" tabindex="-1"></a>      <span class="fu">asin</span>(<span class="fu">sqrt</span>(subset_gene_subset))</span>
<span id="cb12-72"><a href="#cb12-72" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e) <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(subset_gene_subset)))</span>
<span id="cb12-73"><a href="#cb12-73" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-74"><a href="#cb12-74" aria-hidden="true" tabindex="-1"></a>    subset_norm_arcsin_data <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>({</span>
<span id="cb12-75"><a href="#cb12-75" aria-hidden="true" tabindex="-1"></a>      <span class="fu">asin</span>(<span class="fu">sqrt</span>(subset_gene_subset<span class="sc">/</span><span class="fu">max</span>(subset_gene_subset)))</span>
<span id="cb12-76"><a href="#cb12-76" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">error =</span> <span class="cf">function</span>(e) <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(subset_gene_subset)))</span>
<span id="cb12-77"><a href="#cb12-77" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-78"><a href="#cb12-78" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb12-79"><a href="#cb12-79" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (covariable <span class="cf">in</span> covariables){</span>
<span id="cb12-80"><a href="#cb12-80" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb12-81"><a href="#cb12-81" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (covariable <span class="sc">==</span> <span class="st">"AAA"</span>){</span>
<span id="cb12-82"><a href="#cb12-82" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb12-83"><a href="#cb12-83" aria-hidden="true" tabindex="-1"></a>        outlier <span class="ot">&lt;-</span> <span class="st">"7BLACK"</span></span>
<span id="cb12-84"><a href="#cb12-84" aria-hidden="true" tabindex="-1"></a>        no_outlier_ids <span class="ot">&lt;-</span> common_ids[common_ids <span class="sc">!=</span> <span class="st">"7BLACK"</span>] </span>
<span id="cb12-85"><a href="#cb12-85" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb12-86"><a href="#cb12-86" aria-hidden="true" tabindex="-1"></a>        gene_subset <span class="ot">&lt;-</span> subset_gene_subset[no_outlier_ids,]</span>
<span id="cb12-87"><a href="#cb12-87" aria-hidden="true" tabindex="-1"></a>        Covariables  <span class="ot">&lt;-</span> subset_Covariables[<span class="fu">match</span>(no_outlier_ids, subset_Covariables<span class="sc">$</span>X), ]</span>
<span id="cb12-88"><a href="#cb12-88" aria-hidden="true" tabindex="-1"></a>        Covariables_log  <span class="ot">&lt;-</span> subset_Covariables_log[<span class="fu">match</span>(no_outlier_ids, subset_Covariables_log<span class="sc">$</span>X),]</span>
<span id="cb12-89"><a href="#cb12-89" aria-hidden="true" tabindex="-1"></a>        Covariables_log2  <span class="ot">&lt;-</span> subset_Covariables_log2[<span class="fu">match</span>(no_outlier_ids, subset_Covariables_log2<span class="sc">$</span>X), ]</span>
<span id="cb12-90"><a href="#cb12-90" aria-hidden="true" tabindex="-1"></a>        Covariables_logit  <span class="ot">&lt;-</span> subset_Covariables_logit[<span class="fu">match</span>(no_outlier_ids, subset_Covariables_logit<span class="sc">$</span>X), ]</span>
<span id="cb12-91"><a href="#cb12-91" aria-hidden="true" tabindex="-1"></a>        Covariables_arcsin  <span class="ot">&lt;-</span> subset_Covariables_arcsin[<span class="fu">match</span>(no_outlier_ids, subset_Covariables_arcsin<span class="sc">$</span>X), ]</span>
<span id="cb12-92"><a href="#cb12-92" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb12-93"><a href="#cb12-93" aria-hidden="true" tabindex="-1"></a>        <span class="co"># gene transformations</span></span>
<span id="cb12-94"><a href="#cb12-94" aria-hidden="true" tabindex="-1"></a>        log_data    <span class="ot">&lt;-</span> subset_log_data</span>
<span id="cb12-95"><a href="#cb12-95" aria-hidden="true" tabindex="-1"></a>        log2_data   <span class="ot">&lt;-</span> subset_log2_data</span>
<span id="cb12-96"><a href="#cb12-96" aria-hidden="true" tabindex="-1"></a>        logit_data <span class="ot">&lt;-</span> subset_logit_data</span>
<span id="cb12-97"><a href="#cb12-97" aria-hidden="true" tabindex="-1"></a>        norm_logit_data <span class="ot">&lt;-</span> subset_norm_logit_data</span>
<span id="cb12-98"><a href="#cb12-98" aria-hidden="true" tabindex="-1"></a>        arcsin_data <span class="ot">&lt;-</span> subset_arcsin_data</span>
<span id="cb12-99"><a href="#cb12-99" aria-hidden="true" tabindex="-1"></a>        norm_arcsin_data <span class="ot">&lt;-</span> subset_norm_arcsin_data</span>
<span id="cb12-100"><a href="#cb12-100" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb12-101"><a href="#cb12-101" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb12-102"><a href="#cb12-102" aria-hidden="true" tabindex="-1"></a>        gene_subset <span class="ot">&lt;-</span> subset_gene_subset</span>
<span id="cb12-103"><a href="#cb12-103" aria-hidden="true" tabindex="-1"></a>        Covariables  <span class="ot">&lt;-</span> subset_Covariables</span>
<span id="cb12-104"><a href="#cb12-104" aria-hidden="true" tabindex="-1"></a>        Covariables_log  <span class="ot">&lt;-</span> subset_Covariables_log</span>
<span id="cb12-105"><a href="#cb12-105" aria-hidden="true" tabindex="-1"></a>        Covariables_log2  <span class="ot">&lt;-</span> subset_Covariables_log2</span>
<span id="cb12-106"><a href="#cb12-106" aria-hidden="true" tabindex="-1"></a>        Covariables_logit  <span class="ot">&lt;-</span> subset_Covariables_logit</span>
<span id="cb12-107"><a href="#cb12-107" aria-hidden="true" tabindex="-1"></a>        Covariables_arcsin  <span class="ot">&lt;-</span> subset_Covariables_arcsin</span>
<span id="cb12-108"><a href="#cb12-108" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb12-109"><a href="#cb12-109" aria-hidden="true" tabindex="-1"></a>        <span class="co"># gene transformations</span></span>
<span id="cb12-110"><a href="#cb12-110" aria-hidden="true" tabindex="-1"></a>        log_data    <span class="ot">&lt;-</span> subset_log_data</span>
<span id="cb12-111"><a href="#cb12-111" aria-hidden="true" tabindex="-1"></a>        log2_data   <span class="ot">&lt;-</span> subset_log2_data</span>
<span id="cb12-112"><a href="#cb12-112" aria-hidden="true" tabindex="-1"></a>        logit_data <span class="ot">&lt;-</span> subset_logit_data</span>
<span id="cb12-113"><a href="#cb12-113" aria-hidden="true" tabindex="-1"></a>        norm_logit_data <span class="ot">&lt;-</span> subset_norm_logit_data</span>
<span id="cb12-114"><a href="#cb12-114" aria-hidden="true" tabindex="-1"></a>        arcsin_data <span class="ot">&lt;-</span> subset_arcsin_data</span>
<span id="cb12-115"><a href="#cb12-115" aria-hidden="true" tabindex="-1"></a>        norm_arcsin_data <span class="ot">&lt;-</span> subset_norm_arcsin_data</span>
<span id="cb12-116"><a href="#cb12-116" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb12-117"><a href="#cb12-117" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb12-118"><a href="#cb12-118" aria-hidden="true" tabindex="-1"></a>      result_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(gene_subset, Covariables[, covariable])</span>
<span id="cb12-119"><a href="#cb12-119" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb12-120"><a href="#cb12-120" aria-hidden="true" tabindex="-1"></a>      result_log_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(log_data, Covariables[, covariable])</span>
<span id="cb12-121"><a href="#cb12-121" aria-hidden="true" tabindex="-1"></a>      result_log_log  <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(log_data, Covariables_log[,covariable])</span>
<span id="cb12-122"><a href="#cb12-122" aria-hidden="true" tabindex="-1"></a>      result_gene_log <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(gene_subset, Covariables_log[,covariable])</span>
<span id="cb12-123"><a href="#cb12-123" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb12-124"><a href="#cb12-124" aria-hidden="true" tabindex="-1"></a>      result_log2_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(log2_data, Covariables[, covariable])</span>
<span id="cb12-125"><a href="#cb12-125" aria-hidden="true" tabindex="-1"></a>      result_log2_log2 <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(log2_data, Covariables_log2[,covariable])</span>
<span id="cb12-126"><a href="#cb12-126" aria-hidden="true" tabindex="-1"></a>      result_gene_log2 <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(gene_subset, Covariables_log2[,covariable])</span>
<span id="cb12-127"><a href="#cb12-127" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb12-128"><a href="#cb12-128" aria-hidden="true" tabindex="-1"></a>      result_logit_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(logit_data, Covariables[, covariable])</span>
<span id="cb12-129"><a href="#cb12-129" aria-hidden="true" tabindex="-1"></a>      result_logit_logit <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(logit_data, Covariables_logit[,covariable])</span>
<span id="cb12-130"><a href="#cb12-130" aria-hidden="true" tabindex="-1"></a>      result_gene_logit <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(gene_subset, Covariables_logit[, covariable])</span>
<span id="cb12-131"><a href="#cb12-131" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb12-132"><a href="#cb12-132" aria-hidden="true" tabindex="-1"></a>      result_norm_logit_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(norm_logit_data, Covariables[, covariable])</span>
<span id="cb12-133"><a href="#cb12-133" aria-hidden="true" tabindex="-1"></a>      result_norm_logit_logit <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(norm_logit_data,  Covariables_logit[,covariable])</span>
<span id="cb12-134"><a href="#cb12-134" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb12-135"><a href="#cb12-135" aria-hidden="true" tabindex="-1"></a>      result_arcsin_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(arcsin_data, Covariables[, covariable])</span>
<span id="cb12-136"><a href="#cb12-136" aria-hidden="true" tabindex="-1"></a>      result_arcsin_arcsin <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(arcsin_data, Covariables_arcsin[,covariable])</span>
<span id="cb12-137"><a href="#cb12-137" aria-hidden="true" tabindex="-1"></a>      result_gene_arcsin <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(gene_subset, Covariables_arcsin[,covariable])</span>
<span id="cb12-138"><a href="#cb12-138" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb12-139"><a href="#cb12-139" aria-hidden="true" tabindex="-1"></a>      result_norm_arcsin_corr <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(norm_arcsin_data, Covariables[, covariable])</span>
<span id="cb12-140"><a href="#cb12-140" aria-hidden="true" tabindex="-1"></a>      result_norm_arcsin_arcsin <span class="ot">&lt;-</span> <span class="fu">safe_cor</span>(norm_arcsin_data, Covariables_arcsin[,covariable])</span>
<span id="cb12-141"><a href="#cb12-141" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb12-142"><a href="#cb12-142" aria-hidden="true" tabindex="-1"></a>      results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results,</span>
<span id="cb12-143"><a href="#cb12-143" aria-hidden="true" tabindex="-1"></a>                       <span class="co">#Cor - Cor</span></span>
<span id="cb12-144"><a href="#cb12-144" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-145"><a href="#cb12-145" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-146"><a href="#cb12-146" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">""</span>,</span>
<span id="cb12-147"><a href="#cb12-147" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>, </span>
<span id="cb12-148"><a href="#cb12-148" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_corr<span class="sc">$</span>estimate, </span>
<span id="cb12-149"><a href="#cb12-149" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_corr<span class="sc">$</span>p.value),</span>
<span id="cb12-150"><a href="#cb12-150" aria-hidden="true" tabindex="-1"></a>                       <span class="co">#Log - Cor</span></span>
<span id="cb12-151"><a href="#cb12-151" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-152"><a href="#cb12-152" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-153"><a href="#cb12-153" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Log"</span>,</span>
<span id="cb12-154"><a href="#cb12-154" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>, </span>
<span id="cb12-155"><a href="#cb12-155" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_log_corr<span class="sc">$</span>estimate, </span>
<span id="cb12-156"><a href="#cb12-156" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_log_corr<span class="sc">$</span>p.value),</span>
<span id="cb12-157"><a href="#cb12-157" aria-hidden="true" tabindex="-1"></a>                       <span class="co">#Log - Log</span></span>
<span id="cb12-158"><a href="#cb12-158" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-159"><a href="#cb12-159" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-160"><a href="#cb12-160" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Log"</span>,</span>
<span id="cb12-161"><a href="#cb12-161" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Log"</span>, </span>
<span id="cb12-162"><a href="#cb12-162" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_log_log<span class="sc">$</span>estimate, </span>
<span id="cb12-163"><a href="#cb12-163" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_log_log<span class="sc">$</span>p.value),</span>
<span id="cb12-164"><a href="#cb12-164" aria-hidden="true" tabindex="-1"></a>                       <span class="co">#Gene - Log</span></span>
<span id="cb12-165"><a href="#cb12-165" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-166"><a href="#cb12-166" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-167"><a href="#cb12-167" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">""</span>,</span>
<span id="cb12-168"><a href="#cb12-168" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Log"</span>, </span>
<span id="cb12-169"><a href="#cb12-169" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_gene_log<span class="sc">$</span>estimate, </span>
<span id="cb12-170"><a href="#cb12-170" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_gene_log<span class="sc">$</span>p.value),</span>
<span id="cb12-171"><a href="#cb12-171" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Log2 - Corr</span></span>
<span id="cb12-172"><a href="#cb12-172" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-173"><a href="#cb12-173" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-174"><a href="#cb12-174" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Log2"</span>,</span>
<span id="cb12-175"><a href="#cb12-175" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>, </span>
<span id="cb12-176"><a href="#cb12-176" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_log2_corr<span class="sc">$</span>estimate, </span>
<span id="cb12-177"><a href="#cb12-177" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_log2_corr<span class="sc">$</span>p.value),</span>
<span id="cb12-178"><a href="#cb12-178" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Log2 - Log2</span></span>
<span id="cb12-179"><a href="#cb12-179" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-180"><a href="#cb12-180" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-181"><a href="#cb12-181" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Log2"</span>,</span>
<span id="cb12-182"><a href="#cb12-182" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Log2"</span>, </span>
<span id="cb12-183"><a href="#cb12-183" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_log2_log2<span class="sc">$</span>estimate, </span>
<span id="cb12-184"><a href="#cb12-184" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_log2_log2<span class="sc">$</span>p.value),</span>
<span id="cb12-185"><a href="#cb12-185" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Gene - Log2</span></span>
<span id="cb12-186"><a href="#cb12-186" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-187"><a href="#cb12-187" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-188"><a href="#cb12-188" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">""</span>,</span>
<span id="cb12-189"><a href="#cb12-189" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Log2"</span>, </span>
<span id="cb12-190"><a href="#cb12-190" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_gene_log2<span class="sc">$</span>estimate, </span>
<span id="cb12-191"><a href="#cb12-191" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_gene_log2<span class="sc">$</span>p.value),</span>
<span id="cb12-192"><a href="#cb12-192" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Logit - Corr</span></span>
<span id="cb12-193"><a href="#cb12-193" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-194"><a href="#cb12-194" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-195"><a href="#cb12-195" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Logit"</span>,</span>
<span id="cb12-196"><a href="#cb12-196" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>,  </span>
<span id="cb12-197"><a href="#cb12-197" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_logit_corr<span class="sc">$</span>estimate, </span>
<span id="cb12-198"><a href="#cb12-198" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_logit_corr<span class="sc">$</span>p.value),</span>
<span id="cb12-199"><a href="#cb12-199" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Logit - Logit</span></span>
<span id="cb12-200"><a href="#cb12-200" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-201"><a href="#cb12-201" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-202"><a href="#cb12-202" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Logit"</span>,</span>
<span id="cb12-203"><a href="#cb12-203" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Logit"</span>,  </span>
<span id="cb12-204"><a href="#cb12-204" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_logit_logit<span class="sc">$</span>estimate, </span>
<span id="cb12-205"><a href="#cb12-205" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_logit_logit<span class="sc">$</span>p.value),</span>
<span id="cb12-206"><a href="#cb12-206" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Gene - Logit</span></span>
<span id="cb12-207"><a href="#cb12-207" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-208"><a href="#cb12-208" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-209"><a href="#cb12-209" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">""</span>,</span>
<span id="cb12-210"><a href="#cb12-210" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Logit"</span>,  </span>
<span id="cb12-211"><a href="#cb12-211" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_gene_logit<span class="sc">$</span>estimate, </span>
<span id="cb12-212"><a href="#cb12-212" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_gene_logit<span class="sc">$</span>p.value),</span>
<span id="cb12-213"><a href="#cb12-213" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Norm+Logit - Corr</span></span>
<span id="cb12-214"><a href="#cb12-214" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-215"><a href="#cb12-215" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-216"><a href="#cb12-216" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Norm + Logit"</span>,</span>
<span id="cb12-217"><a href="#cb12-217" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>,  </span>
<span id="cb12-218"><a href="#cb12-218" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_norm_logit_corr<span class="sc">$</span>estimate, </span>
<span id="cb12-219"><a href="#cb12-219" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_norm_logit_corr<span class="sc">$</span>p.value),</span>
<span id="cb12-220"><a href="#cb12-220" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Norm+Logit - Logit</span></span>
<span id="cb12-221"><a href="#cb12-221" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-222"><a href="#cb12-222" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-223"><a href="#cb12-223" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Norm + Logit"</span>,</span>
<span id="cb12-224"><a href="#cb12-224" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Logit"</span>,  </span>
<span id="cb12-225"><a href="#cb12-225" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_norm_logit_logit<span class="sc">$</span>estimate, </span>
<span id="cb12-226"><a href="#cb12-226" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_norm_logit_logit<span class="sc">$</span>p.value),</span>
<span id="cb12-227"><a href="#cb12-227" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Arcsin - Corr</span></span>
<span id="cb12-228"><a href="#cb12-228" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-229"><a href="#cb12-229" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-230"><a href="#cb12-230" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Arcsin"</span>,</span>
<span id="cb12-231"><a href="#cb12-231" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>,  </span>
<span id="cb12-232"><a href="#cb12-232" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_arcsin_corr<span class="sc">$</span>estimate, </span>
<span id="cb12-233"><a href="#cb12-233" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_arcsin_corr<span class="sc">$</span>p.value),</span>
<span id="cb12-234"><a href="#cb12-234" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Arcsin - Arcsin</span></span>
<span id="cb12-235"><a href="#cb12-235" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-236"><a href="#cb12-236" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-237"><a href="#cb12-237" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Arcsin"</span>,</span>
<span id="cb12-238"><a href="#cb12-238" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Arcsin"</span>,  </span>
<span id="cb12-239"><a href="#cb12-239" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_arcsin_arcsin<span class="sc">$</span>estimate, </span>
<span id="cb12-240"><a href="#cb12-240" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_arcsin_arcsin<span class="sc">$</span>p.value),</span>
<span id="cb12-241"><a href="#cb12-241" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Gene - Arcsin</span></span>
<span id="cb12-242"><a href="#cb12-242" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-243"><a href="#cb12-243" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-244"><a href="#cb12-244" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">""</span>,</span>
<span id="cb12-245"><a href="#cb12-245" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Arcsin"</span>,  </span>
<span id="cb12-246"><a href="#cb12-246" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_gene_arcsin<span class="sc">$</span>estimate, </span>
<span id="cb12-247"><a href="#cb12-247" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_gene_arcsin<span class="sc">$</span>p.value),</span>
<span id="cb12-248"><a href="#cb12-248" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Norm+Arcsin - Corr</span></span>
<span id="cb12-249"><a href="#cb12-249" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-250"><a href="#cb12-250" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-251"><a href="#cb12-251" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Norm + Arcsin"</span>,</span>
<span id="cb12-252"><a href="#cb12-252" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">""</span>,  </span>
<span id="cb12-253"><a href="#cb12-253" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_norm_arcsin_corr<span class="sc">$</span>estimate, </span>
<span id="cb12-254"><a href="#cb12-254" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_norm_arcsin_corr<span class="sc">$</span>p.value),</span>
<span id="cb12-255"><a href="#cb12-255" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># Norm+Arcsin - Arcsin</span></span>
<span id="cb12-256"><a href="#cb12-256" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">data.frame</span>(<span class="at">Cell =</span> cell, <span class="at">Gene =</span> gene, </span>
<span id="cb12-257"><a href="#cb12-257" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Covariable =</span> covariable,</span>
<span id="cb12-258"><a href="#cb12-258" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Gene_transformation =</span> <span class="st">"Norm + Arcsin"</span>,</span>
<span id="cb12-259"><a href="#cb12-259" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Cov_transformation =</span> <span class="st">"Arcsin"</span>,  </span>
<span id="cb12-260"><a href="#cb12-260" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Spearman =</span> result_norm_arcsin_arcsin<span class="sc">$</span>estimate, </span>
<span id="cb12-261"><a href="#cb12-261" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_value =</span> result_norm_arcsin_arcsin<span class="sc">$</span>p.value))</span>
<span id="cb12-262"><a href="#cb12-262" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb12-263"><a href="#cb12-263" aria-hidden="true" tabindex="-1"></a>      result_log_corr <span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb12-264"><a href="#cb12-264" aria-hidden="true" tabindex="-1"></a>      result_log_log  <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-265"><a href="#cb12-265" aria-hidden="true" tabindex="-1"></a>      result_gene_log <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-266"><a href="#cb12-266" aria-hidden="true" tabindex="-1"></a>      result_log2_corr <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-267"><a href="#cb12-267" aria-hidden="true" tabindex="-1"></a>      result_log2_log2 <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-268"><a href="#cb12-268" aria-hidden="true" tabindex="-1"></a>      result_gene_log2 <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-269"><a href="#cb12-269" aria-hidden="true" tabindex="-1"></a>      result_logit_corr <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-270"><a href="#cb12-270" aria-hidden="true" tabindex="-1"></a>      result_logit_logit <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-271"><a href="#cb12-271" aria-hidden="true" tabindex="-1"></a>      result_gene_logit <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-272"><a href="#cb12-272" aria-hidden="true" tabindex="-1"></a>      result_norm_logit_corr <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-273"><a href="#cb12-273" aria-hidden="true" tabindex="-1"></a>      result_norm_logit_logit <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-274"><a href="#cb12-274" aria-hidden="true" tabindex="-1"></a>      result_arcsin_corr <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-275"><a href="#cb12-275" aria-hidden="true" tabindex="-1"></a>      result_arcsin_arcsin <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-276"><a href="#cb12-276" aria-hidden="true" tabindex="-1"></a>      result_gene_arcsin <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-277"><a href="#cb12-277" aria-hidden="true" tabindex="-1"></a>      result_norm_arcsin_corr <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-278"><a href="#cb12-278" aria-hidden="true" tabindex="-1"></a>      result_norm_arcsin_arcsin <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-279"><a href="#cb12-279" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb12-280"><a href="#cb12-280" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb12-281"><a href="#cb12-281" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb12-282"><a href="#cb12-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-283"><a href="#cb12-283" aria-hidden="true" tabindex="-1"></a>results</span>
<span id="cb12-284"><a href="#cb12-284" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(results, <span class="at">file =</span> <span class="st">".../BAYESPRISM_12_3/CORRELATION/GENES_CORR/TDP_genes_correlation.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>



</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-robinson2010_edger" class="csl-entry" role="listitem">
Robinson, Mark D., Davis J. McCarthy, and Gordon K. Smyth. 2010. <span>“edgeR: A Bioconductor Package for Differential Expression Analysis of Digital Gene Expression Data.”</span> <em>Bioinformatics</em> 26 (1): 139–40. <a href="https://doi.org/10.1093/bioinformatics/btp616">https://doi.org/10.1093/bioinformatics/btp616</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/Aumatell\.github\.io\/FTLD_RNAseq_Analysis\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>